// @libriciel/ls-composants/esm5/lib/pages/login/forgot-password/forgot-password.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/forgot-password/forgot-password.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
var ForgotPasswordComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ForgotPasswordComponent, _super);
    function ForgotPasswordComponent(_DomSanitizationService) {
        var _this = _super.call(this, _DomSanitizationService) || this;
        _this._DomSanitizationService = _DomSanitizationService;
        _this.oneFieldForgot = false;
        _this.usernameormailInputName = 'usernameormail';
        _this.usernameInputName = 'username';
        _this.emailInputName = 'email';
        _this.backToLogin = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    ForgotPasswordComponent.prototype.backToLoginPage = /**
     * @return {?}
     */
    function () {
        this.backToLogin.emit();
    };
    /**
     * @param {?} event
     * @return {?}
     */
    ForgotPasswordComponent.prototype.preventFormSubmit = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        _super.prototype.preventFormSubmit.call(this, event);
    };
    /**
     * @return {?}
     */
    ForgotPasswordComponent.prototype.submitForm = /**
     * @return {?}
     */
    function () {
        this.submit.emit({
            username: this.username,
            email: this.email,
            usernameormail: this.usernameormail
        });
    };
    ForgotPasswordComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-forgot-password',
                    template: "<ls-login-model [logo]=\"logo\" [visualConfiguration]=\"_visualConfiguration\">\n  <section id=\"forgot-page\">\n    <h1 class=\"page-title\">Mot de passe oubli\u00E9 ?</h1>\n    <form [action]=\"formAction\" id=\"ls-forgot-form\" method=\"post\" ngNoForm (submit)=\"preventFormSubmit($event)\">\n\n      <div *ngIf=\"!oneFieldForgot\" class=\"usernameormail\">\n        <label [attr.for]=\"usernameInputName\">\n          Identifiant\n        </label>\n        <input [attr.name]=\"usernameInputName\" [attr.id]=\"usernameInputName\" [(ngModel)]=\"username\" class=\"form-control\" placeholder=\"Mon identifiant\"\n               type=\"text\" required>\n      </div>\n      <div *ngIf=\"!oneFieldForgot\" class=\"usernameormail\">\n        <label [attr.for]=\"emailInputName\">\n          Adresse mail\n        </label>\n        <input [attr.id]=\"emailInputName\" [attr.name]=\"emailInputName\" [(ngModel)]=\"email\" class=\"form-control\" placeholder=\"adresse@mail.com\" type=\"text\" required>\n      </div>\n\n\n      <div *ngIf=\"oneFieldForgot\" class=\"usernameormail\">\n        <label [attr.for]=\"usernameormailInputName\">\n          Saisissez votre nom d'utilisateur ou votre adresse mail\n        </label>\n        <input [attr.name]=\"usernameormailInputName\"  [attr.id]=\"usernameormailInputName\" [(ngModel)]=\"usernameormail\" class=\"form-control\"\n               placeholder=\"Nom d'utilisateur ou adresse mail\"\n               type=\"text\" required>\n      </div>\n\n      <div class=\"alert alert-info\">\n        <div class=\"fa fa-lg fa-info-circle\"></div>\n        <div>\n          Si vos informations sont correctes, un e-mail vous sera envoy\u00E9.<br>\n          Vous pourrez ainsi cr\u00E9er un nouveau mot de passe.\n        </div>\n      </div>\n\n      <ng-content></ng-content>\n\n      <div class=\"float-left\">\n        <span (click)=\"backToLoginPage()\" class=\"btn btn-link forgot-passwd-lnk\">\n          <i class=\"fa fa-angle-double-left\"></i>\n          Revenir \u00E0 l'\u00E9cran de connexion\n        </span>\n      </div>\n\n\n      <div class=\"float-right\">\n        <button class=\"btn btn-primary\" (click)=\"submitForm()\" type=\"submit\" form=\"ls-forgot-form\" [class.disabled]=\"submitted\">\n          <i *ngIf=\"!submitted\" class=\"fa fa-paper-plane fa-fw\"></i>\n          <i *ngIf=\"submitted\" class=\"fa fa-spinner fa-spin fa-fw\"></i>\n          Soumettre\n        </button>\n      </div>\n    </form>\n  </section>\n</ls-login-model>\n",
                    styles: [".alert{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.alert .fa{width:2rem}#ls-forgot-form>*{margin-bottom:1rem}span.btn{cursor:pointer}.page-title{font-size:1rem;font-weight:400;padding-top:3rem;padding-bottom:1.5rem;text-align:center}"]
                }] }
    ];
    /** @nocollapse */
    ForgotPasswordComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    ForgotPasswordComponent.propDecorators = {
        oneFieldForgot: [{ type: Input }],
        usernameormailInputName: [{ type: Input }],
        usernameInputName: [{ type: Input }],
        emailInputName: [{ type: Input }],
        backToLogin: [{ type: Output }]
    };
    return ForgotPasswordComponent;
}(LoginPageModelClass));
export { ForgotPasswordComponent };
if (false) {
    /** @type {?} */
    ForgotPasswordComponent.prototype.username;
    /** @type {?} */
    ForgotPasswordComponent.prototype.email;
    /** @type {?} */
    ForgotPasswordComponent.prototype.usernameormail;
    /** @type {?} */
    ForgotPasswordComponent.prototype.oneFieldForgot;
    /** @type {?} */
    ForgotPasswordComponent.prototype.usernameormailInputName;
    /** @type {?} */
    ForgotPasswordComponent.prototype.usernameInputName;
    /** @type {?} */
    ForgotPasswordComponent.prototype.emailInputName;
    /** @type {?} */
    ForgotPasswordComponent.prototype.backToLogin;
    /** @type {?} */
    ForgotPasswordComponent.prototype._DomSanitizationService;
}
