// @libriciel/ls-composants/esm5/lib/component/password/password.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/password/password.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, forwardRef, Input, Renderer2, ViewChild } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var PasswordComponent = /** @class */ (function () {
    function PasswordComponent(element, renderer2) {
        this.element = element;
        this.renderer2 = renderer2;
        this.showIcon = true;
        this.trim = false;
        this.isShown = false;
        this.onChange = (/**
         * @return {?}
         */
        function () {
            // tslint:disable-next-line:semicolon
        });
        this.onTouched = (/**
         * @return {?}
         */
        function () {
            // tslint:disable-next-line:semicolon
        });
    }
    Object.defineProperty(PasswordComponent.prototype, "value", {
        get: /**
         * @return {?}
         */
        function () {
            return this._value;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            if (this.trim) {
                this._value = value.trim();
            }
            else {
                this._value = value;
            }
            this.onChange(this._value);
            this.onTouched();
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    PasswordComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var id = this.element.nativeElement.getAttribute('id');
        this.renderer2.removeAttribute(this.element.nativeElement, 'id');
        this.renderer2.setAttribute(this.input.nativeElement, 'id', id);
    };
    /**
     * @return {?}
     */
    PasswordComponent.prototype.togglePassword = /**
     * @return {?}
     */
    function () {
        this.isShown = !this.isShown;
    };
    /**
     * @return {?}
     */
    PasswordComponent.prototype.onChangeSelection = /**
     * @return {?}
     */
    function () {
        this.onChange(this._value);
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    PasswordComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    PasswordComponent.prototype.registerOnTouched = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} value
     * @return {?}
     */
    PasswordComponent.prototype.writeValue = /**
     * @param {?} value
     * @return {?}
     */
    function (value) {
        if (value) {
            this.value = value;
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    PasswordComponent.prototype.trimValue = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.trim) {
            event.target.value = event.target.value.trim();
        }
    };
    PasswordComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-password',
                    template: "<div class=\"form-group input-group-front-icon\" [class.input-group-front-icon]=\"showIcon\">\n  <span *ngIf=\"showIcon\">\n    <i class=\"fa fa-lock\"></i>\n  </span>\n  <input #input [attr.name]=\"name\" [(ngModel)]=\"value\" (ngModelChange)=\"onChangeSelection()\" [required]=\"required\"\n         class=\"form-control\" [type]=\"isShown ? 'text' : 'password'\" (keyup)=\"trimValue($event)\">\n\n  <span class=\"input-icon\" (click)=\"togglePassword()\">\n    <i class=\"fa\" [ngClass]=\"isShown ? 'fa-eye-slash' : 'fa-eye'\"></i>\n  </span>\n</div>\n\n\n\n\n\n",
                    providers: [
                        {
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return PasswordComponent; })),
                            multi: true
                        }
                    ],
                    styles: [".input-icon{position:absolute;top:0;right:0;background-color:transparent;color:var(--ls-grey-500);cursor:pointer;height:99%;width:40px;border-radius:5px 0 0 5px;display:-webkit-inline-box;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center}.input-icon:hover{color:var(--ls-grey-700)}.input-group-end-icon,:host.input-group-end-icon{position:relative;display:-webkit-box;display:-ms-flexbox;display:flex}.input-group-end-icon input,:host.input-group-end-icon input{padding-right:36px}input.ls-action-input{cursor:pointer}.input-group.input-group-end-icon .input-icon,:host.input-group.input-group-end-icon .input-icon{z-index:3}.form-group{position:relative}input::-ms-clear,input::-ms-reveal{display:none}"]
                }] }
    ];
    /** @nocollapse */
    PasswordComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    PasswordComponent.propDecorators = {
        showIcon: [{ type: Input }],
        required: [{ type: Input }],
        trim: [{ type: Input }],
        input: [{ type: ViewChild, args: ['input',] }],
        name: [{ type: Input }],
        _value: [{ type: Input }]
    };
    return PasswordComponent;
}());
export { PasswordComponent };
if (false) {
    /** @type {?} */
    PasswordComponent.prototype.showIcon;
    /** @type {?} */
    PasswordComponent.prototype.required;
    /** @type {?} */
    PasswordComponent.prototype.trim;
    /** @type {?} */
    PasswordComponent.prototype.input;
    /** @type {?} */
    PasswordComponent.prototype.name;
    /** @type {?} */
    PasswordComponent.prototype.isShown;
    /** @type {?} */
    PasswordComponent.prototype._value;
    /** @type {?} */
    PasswordComponent.prototype.onChange;
    /** @type {?} */
    PasswordComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    PasswordComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    PasswordComponent.prototype.renderer2;
}
