// @libriciel/ls-composants/esm5/workflows/workflows/ls-workflow-editor/ls-workflow-editor.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflow-editor/ls-workflow-editor.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChild, EventEmitter, Input, Output, QueryList, TemplateRef, ViewChild, ViewChildren } from '@angular/core';
import { NgbPopover } from '@ng-bootstrap/ng-bootstrap';
import { WorkflowModel } from '../../shared/model/workflows/workflow-model';
import { StepModel } from '../../shared/model/workflows/step-model';
import { moveItemInArray } from '@angular/cdk/drag-drop';
import { WorkflowModelLocalizedMessages, WorkflowModelOptions } from '../../shared/model/workflows/workflow-model-options';
var LsWorkflowEditorComponent = /** @class */ (function () {
    function LsWorkflowEditorComponent() {
        this.messages = {
            CREATE_STEP_TITLE: 'Ajouter une étape',
            EDIT_STEP_TITLE: 'Modification de l\'étape',
            ADD_STEP_BTN_TXT: 'Ajouter une étape',
            FIRST_STEP_TXT: 'Début',
            LAST_STEP_TXT: 'Fin',
        };
        this.activeOptions = {
            startIconClass: 'far fa-flag',
            endIconClass: 'fas fa-flag-checkered',
            editorGlobalLayoutClass: 'ls-workflow-editor-default-layout',
            additionalFinalEndpointClasses: ''
        };
        this.endpointIsEditable = false;
        this.workflowChange = new EventEmitter();
    }
    Object.defineProperty(LsWorkflowEditorComponent.prototype, "localizedMessages", {
        get: /**
         * @return {?}
         */
        function () {
            return this.messages;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.messages = Object.assign(this.messages, value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LsWorkflowEditorComponent.prototype, "options", {
        get: /**
         * @return {?}
         */
        function () {
            return this.activeOptions;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.activeOptions = Object.assign(this.activeOptions, value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LsWorkflowEditorComponent.prototype, "workflow", {
        get: /**
         * @return {?}
         */
        function () {
            return this.workflowData;
        },
        set: /**
         * @param {?} newValue
         * @return {?}
         */
        function (newValue) {
            this.workflowData = newValue;
            this.workflowChange.emit(newValue);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        if (!this.workflowData) {
            console.error('Missing input workflow data');
            this.workflowData = new WorkflowModel();
        }
        if (!this.messages) {
            throw new Error('Missing localised messages for LsWorkflowViewerComponent!');
        }
    };
    /**
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.workflowDataChanged = /**
     * @return {?}
     */
    function () {
        this.workflowChange.emit(this.workflowData);
    };
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.askForStepDeletion = /**
     * @param {?} stepIdx
     * @return {?}
     */
    function (stepIdx) {
        var _this = this;
        console.debug('delete step');
        if (this.confirmStepDeletionFct) {
            this.confirmStepDeletionFct(this.workflowData.steps[stepIdx]).then((/**
             * @param {?} result
             * @return {?}
             */
            function (result) {
                if (result) {
                    _this.deleteStep(stepIdx);
                }
            }));
        }
        else {
            this.deleteStep(stepIdx);
        }
    };
    /**
     * @param {?} stepData
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.confirmStepCreation = /**
     * @param {?} stepData
     * @param {?} idx
     * @return {?}
     */
    function (stepData, idx) {
        this.addStep(stepData, idx);
        this.closeCurrentPopover();
    };
    /**
     * @param {?} stepData
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.confirmStepEdition = /**
     * @param {?} stepData
     * @param {?} idx
     * @return {?}
     */
    function (stepData, idx) {
        this.replaceStep(stepData, idx);
        this.closeCurrentPopover();
    };
    /**
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.confirmFinalStepEdition = /**
     * @return {?}
     */
    function () {
        this.closeCurrentPopover();
    };
    /**
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.cancelStepEdition = /**
     * @return {?}
     */
    function () {
        this.closeCurrentPopover();
    };
    /**
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.cancelStepCreation = /**
     * @param {?} idx
     * @return {?}
     */
    function (idx) {
        this.closeCurrentPopover();
    };
    /**
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.isFirstStep = /**
     * @return {?}
     */
    function () {
        return (!this.workflowData || this.workflowData.steps.length === 0);
    };
    /**
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.getAdditionalStepClass = /**
     * @param {?} idx
     * @return {?}
     */
    function (idx) {
        if (this.currentEditedStepIdx !== null && this.currentEditedStepIdx === idx) {
            return 'selected';
        }
        return '';
    };
    /**
     * @param {?} step
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.getStepIconClass = /**
     * @param {?} step
     * @return {?}
     */
    function (step) {
        if (this.getStepIconClassFct) {
            return this.getStepIconClassFct(step);
        }
        return '';
    };
    /**
     * @param {?} step
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.getStepTooltip = /**
     * @param {?} step
     * @return {?}
     */
    function (step) {
        if (this.getStepTooltipFct) {
            return this.getStepTooltipFct(step);
        }
        return '';
    };
    /**
     * @param {?} src
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.cloneStep = /**
     * @param {?} src
     * @return {?}
     */
    function (src) {
        return src.clone();
    };
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.createStep = /**
     * @param {?} stepIdx
     * @return {?}
     */
    function (stepIdx) {
        var _this = this;
        /** @type {?} */
        var newStep = new StepModel();
        this.currentPopover = this.getCreationPopoverForIdx(stepIdx);
        this.currentPopover.hidden.subscribe((/**
         * @return {?}
         */
        function () { return _this.cancelStepCreation(stepIdx); }));
        this.currentPopover.open({
            stepData: newStep,
            idx: stepIdx,
            cancelStepCreation: this.cancelStepCreation.bind(this),
            confirmStepCreation: this.confirmStepCreation.bind(this)
        });
    };
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.editStep = /**
     * @param {?} stepIdx
     * @return {?}
     */
    function (stepIdx) {
        var _this = this;
        this.currentEditedStepIdx = stepIdx;
        /** @type {?} */
        var editingStep = this.cloneStep(this.workflowData.steps[stepIdx]);
        this.currentPopover = this.getEditionPopoverForIdx(stepIdx);
        this.currentPopover.hidden.subscribe((/**
         * @return {?}
         */
        function () { return _this.cancelStepEdition(); }));
        this.currentPopover.open({
            stepData: editingStep,
            idx: stepIdx,
            cancelStepEdition: this.cancelStepEdition.bind(this),
            confirmStepEdition: this.confirmStepEdition.bind(this)
        });
    };
    /**
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.editFinalStep = /**
     * @return {?}
     */
    function () {
        var _this = this;
        /** @type {?} */
        var editingStep = this.cloneStep(this.lastStepData);
        this.currentPopover = this.endpointPopoverElementRef;
        this.currentPopover.hidden.subscribe((/**
         * @return {?}
         */
        function () { return _this.cancelStepEdition(); }));
        this.currentPopover.open({
            stepData: editingStep,
            cancelStepEdition: this.cancelStepEdition.bind(this),
            confirmStepEdition: this.confirmFinalStepEdition.bind(this)
        });
    };
    /**
     * @param {?} event
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.drop = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        moveItemInArray(this.workflowData.steps, event.previousIndex, event.currentIndex);
        this.workflowDataChanged();
    };
    /**
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.dragDisabledForIdx = /**
     * @param {?} idx
     * @return {?}
     */
    function (idx) {
        return (this.currentEditedStepIdx !== null && this.currentEditedStepIdx === idx);
    };
    /**
     * @private
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.closeCurrentPopover = /**
     * @private
     * @return {?}
     */
    function () {
        if (this.currentPopover && this.currentPopover.isOpen()) {
            this.currentPopover.close();
        }
        this.currentPopover = null;
        this.currentEditedStepIdx = null;
    };
    /**
     * @private
     * @param {?} step
     * @param {?} pos
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.addStep = /**
     * @private
     * @param {?} step
     * @param {?} pos
     * @return {?}
     */
    function (step, pos) {
        this.workflowData.steps.splice(pos, 0, step);
        this.workflowDataChanged();
    };
    /**
     * @private
     * @param {?} newStep
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.replaceStep = /**
     * @private
     * @param {?} newStep
     * @param {?} stepIdx
     * @return {?}
     */
    function (newStep, stepIdx) {
        this.workflowData.steps.splice(stepIdx, 1, newStep);
        this.workflowDataChanged();
    };
    /**
     * @private
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.deleteStep = /**
     * @private
     * @param {?} stepIdx
     * @return {?}
     */
    function (stepIdx) {
        this.workflowData.steps.splice(stepIdx, 1);
        this.workflowDataChanged();
    };
    /**
     * @private
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.getCreationPopoverForIdx = /**
     * @private
     * @param {?} idx
     * @return {?}
     */
    function (idx) {
        /** @type {?} */
        var filteredList = this.allCircuitEditionPopover.filter((/**
         * @param {?} item
         * @return {?}
         */
        function (item) {
            /** @type {?} */
            var elem = ((/** @type {?} */ (item)))._elementRef;
            if (elem && elem.nativeElement) {
                return elem.nativeElement.hasAttribute('create-btn');
            }
            return true;
        }));
        return filteredList[idx];
    };
    /**
     * @private
     * @param {?} idx
     * @return {?}
     */
    LsWorkflowEditorComponent.prototype.getEditionPopoverForIdx = /**
     * @private
     * @param {?} idx
     * @return {?}
     */
    function (idx) {
        /** @type {?} */
        var filteredList = this.allCircuitEditionPopover.filter((/**
         * @param {?} item
         * @return {?}
         */
        function (item) {
            /** @type {?} */
            var elem = ((/** @type {?} */ (item)))._elementRef;
            if (elem && elem.nativeElement) {
                return !(elem.nativeElement.hasAttribute('create-btn'));
            }
            return true;
        }));
        return filteredList[idx];
    };
    LsWorkflowEditorComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-workflow-editor',
                    template: "<div class=\"ls-workflow-editor\" [ngClass]=\"options.editorGlobalLayoutClass\">\n  <div class=\"ls-workflow-canevas\">\n    <div class=\"ls-workflow-start ls-workflow-endpoint\">\n      <div class=\"ls-workflow-endpoint-content\">\n        <div class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.startIconClass\"></div>\n        <div class=\"ls-step-bounds-txt\">{{messages.FIRST_STEP_TXT}}</div>\n      </div>\n    </div>\n    <div class=\"ls-workflow-stepList-container\" cdkDropList (cdkDropListDropped)=\"drop($event)\">\n      <div class=\"ls-workflow-link-container\">\n        <div class=\"ls-workflow-link\"></div>\n        <div [ngClass]=\"isFirstStep() ? 'firstStep' : ''\" class=\"ls-link-button-container\">\n          <div (click)=\"createStep(0)\" [autoClose]=\"'outside'\" [ngbPopover]=\"popCreateContent\"\n               [popoverTitle]=\"messages.popCreateTitle\" class=\"add-step-btn\" create-btn\n               placement=\"right\" popoverClass=\"ls-step-edition-popover\"\n               triggers=\"manual\"\n               autoClose=\"false\">\n            <span class=\"fa fa-plus-circle\"></span>\n            <span class=\"add-step-btn-txt\">{{messages.ADD_STEP_BTN_TXT}}</span>\n          </div>\n        </div>\n      </div>\n\n      <div *ngFor=\"let step of workflowData.steps; index as i\">\n        <div class=\"ls-workflow-step-container\" cdkDrag [cdkDragDisabled]=\"dragDisabledForIdx(i)\">\n          <div [ngClass]=\"getAdditionalStepClass(i)\" class=\"ls-workflow-step\">\n            <div class=\"ls-step-top-line\">\n              <div class=\"step-number\">{{i + 1}}.</div>\n              <div class=\"step-name\">{{step.name}}</div>\n              <div [ngClass]=\"getStepIconClass(step)\" class=\"step-icn fa\" ngbTooltip=\"{{getStepTooltip(step)}}\"></div>\n            </div>\n            <div class=\"ls-step-validator-list\">\n              <ng-template\n                [ngTemplateOutlet]=\"stepDisplayContentRef\"\n                [ngTemplateOutletContext]=\"{step: step}\">\n              </ng-template>\n            </div>\n          </div>\n          <div class=\"ls-workflow-step-external-buttons\">\n            <div (click)=\"editStep(i)\" [autoClose]=\"'outside'\" [ngbPopover]=\"popEditContent\"\n                 [popoverTitle]=\"messages.popEditTitle\" class=\"fa fa-pencil-alt fa-pencil edit-step-btn\"\n                 placement=\"right-top\"\n                 popoverClass=\"ls-step-edition-popover\"\n                 triggers=\"manual\"></div>\n            <div (click)=\"askForStepDeletion(i)\" class=\"fa fa-trash delete-step-btn\"></div>\n          </div>\n        </div>\n\n        <div class=\"ls-workflow-link-container\">\n          <div class=\"ls-workflow-link\"></div>\n          <div class=\"ls-link-button-container\">\n            <div (click)=\"createStep(i + 1)\" [autoClose]=\"'outside'\" [ngbPopover]=\"popCreateContent\"\n                 [popoverTitle]=\"messages.popCreateTitle\" class=\"add-step-btn\" create-btn\n                 placement=\"right\" popoverClass=\"ls-step-edition-popover\"\n                 triggers=\"manual\"\n                 autoClose=\"false\">\n              <span class=\"fa fa-plus-circle\"></span>\n              <span class=\"add-step-btn-txt\">{{messages.ADD_STEP_BTN_TXT}}</span>\n            </div>\n          </div>\n        </div>\n      </div>\n\n    </div>\n    <div class=\"ls-workflow-end ls-workflow-endpoint\" [ngClass]=\"options.additionalFinalEndpointClasses\">\n      <ng-container *ngIf=\"endpointIsEditable\">\n        <div\n          #endpointPopoverHandle\n          (click)=\"editFinalStep()\"\n          [autoClose]=\"'outside'\"\n          [ngbPopover]=\"popEditFinalStepContent\"\n          [popoverTitle]=\"messages.popEditTitle\"\n          class=\"ls-workflow-endpoint-overlay\"\n          placement=\"right\"\n          popoverClass=\"ls-step-edition-popover\"\n          triggers=\"manual\">\n\n        </div>\n      </ng-container>\n      <div class=\"ls-workflow-endpoint-content\">\n        <div class=\"ls-workflow-endpoint-icn-group\">\n          <div class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.endIconClass\"></div>\n          <div class=\"ls-step-bounds-txt\">{{messages.LAST_STEP_TXT}}</div>\n        </div>\n        <div *ngIf=\"endpointIsEditable\" class=\"ls-step-validator-list\">\n          <ng-template [ngTemplateOutlet]=\"lastStepDisplayContentRef\">\n          </ng-template>\n        </div>\n      </div>\n\n    </div>\n  </div>\n</div>\n\n<ng-template\n  #popEditContent\n  let-idx=\"idx\"\n  let-stepData=\"stepData\"\n  let-cancelStepEdition=\"cancelStepEdition\"\n  let-confirmStepEdition=\"confirmStepEdition\">\n  <ng-template\n    [ngTemplateOutlet]=\"popEditContentTemplateRef\"\n    [ngTemplateOutletContext]=\"{idx:idx, stepData:stepData, cancelStepEdition:cancelStepEdition, confirmStepEdition:confirmStepEdition}\">\n  </ng-template>\n</ng-template>\n\n<ng-template\n  #popCreateContent\n  let-idx=\"idx\"\n  let-stepData=\"stepData\"\n  let-cancelStepCreation=\"cancelStepCreation\"\n  let-confirmStepCreation=\"confirmStepCreation\">\n  <ng-template\n    [ngTemplateOutlet]=\"popCreateContentTemplateRef\"\n    [ngTemplateOutletContext]=\"{idx:idx, stepData:stepData, cancelStepCreation:cancelStepCreation, confirmStepCreation:confirmStepCreation}\">\n  </ng-template>\n</ng-template>\n\n\n<ng-template\n  #popEditFinalStepContent\n  let-stepData=\"stepData\"\n  let-cancelStepEdition=\"cancelStepEdition\"\n  let-confirmStepEdition=\"confirmStepEdition\">\n  <ng-template\n    [ngTemplateOutlet]=\"endpointPopoverEditContentRef\"\n    [ngTemplateOutletContext]=\"{stepData:stepData, cancelStepEdition:cancelStepEdition, confirmStepEdition:confirmStepEdition, isLastStep:true}\">\n  </ng-template>\n</ng-template>\n",
                    styles: ["body{background-color:var(--ls-grey-50)}.active-btn{color:#42a1e8}.ls-workflow-canevas{-ms-grid-column:1;grid-column:1;height:100%;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start}.ls-workflow-endpoint.with-content,.ls-workflow-step{background-color:#fff;-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);box-shadow:0 0 .25rem rgba(0,0,0,.15);border:none;width:var(--ls-workflow-step-width,20rem);min-height:6rem}.ls-workflow-endpoint.with-content .ls-step-top-line,.ls-workflow-step .ls-step-top-line{padding:.5rem;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-align:center;-ms-flex-align:center;align-items:center;font-weight:700}.ls-workflow-endpoint.with-content .ls-display-step-separator,.ls-workflow-step .ls-display-step-separator{width:100%;border-bottom:1px solid #d3d3d3}.ls-workflow-endpoint.with-content .ls-step-validator-list,.ls-workflow-step .ls-step-validator-list{padding:.5rem .5rem .5rem 1.5rem;color:#5b5b5b}.ls-workflow-step.selected,.selected.ls-workflow-endpoint.with-content{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.ls-workflow-endpoint.with-content .step-icn,.ls-workflow-step .step-icn{width:1.7rem;height:1.7rem}.ls-workflow-endpoint-content{margin:auto;text-align:center}.ls-workflow-endpoint-content .ls-step-bounds-txt{color:var(--primary);font-weight:700}.ls-workflow-endpoint-content .ls-workflow-endpoint-icn{color:var(--primary)}.ls-workflow-endpoint-content.not-active .ls-step-bounds-txt,.ls-workflow-endpoint-content.not-active .ls-workflow-endpoint-icn{color:#5b5b5b}.ls-workflow-endpoint-content .ls-step-validator-list{text-align:left}.ls-horizontal-separation-line{margin-top:2rem;border-bottom:1px solid #5b5b5b}.ls-workflow-endpoint{background-color:#fff;-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);box-shadow:0 0 .25rem rgba(0,0,0,.15);border:none;display:-ms-grid;display:grid;width:6rem;height:6rem;border-radius:50%;position:relative}.ls-workflow-endpoint.with-content{display:block;width:var(--ls-workflow-step-width,20rem);border-radius:0}.ls-workflow-endpoint.with-content .ls-workflow-endpoint-content{width:100%;height:100%;padding:.5em}.ls-workflow-endpoint.with-content .ls-workflow-endpoint-content .ls-workflow-endpoint-icn-group{width:100%;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:reverse;-ms-flex-direction:row-reverse;flex-direction:row-reverse;-webkit-box-pack:justify;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-align:baseline;-ms-flex-align:baseline;align-items:baseline;padding:0 .5em}.ls-workflow-endpoint.with-content .ls-workflow-endpoint-content .ls-workflow-endpoint-icn-group:after{content:'';display:block;width:2em}.ls-workflow-endpoint .ls-workflow-endpoint-overlay{width:100%;height:100%;position:absolute;left:0;bottom:0;z-index:2;cursor:pointer}.ls-workflow-link{width:0;border:1px solid #222}.cdk-drag-preview .ls-workflow-endpoint.with-content,.cdk-drag-preview .ls-workflow-step{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.cdk-drag-placeholder{min-height:6rem}.cdk-drag-placeholder .ls-workflow-endpoint.with-content,.cdk-drag-placeholder .ls-workflow-step,.cdk-drag-placeholder .ls-workflow-step-external-buttons{display:none}.cdk-drop-list-dragging .cdk-drag{-webkit-transition:-webkit-transform 250ms cubic-bezier(0,0,.2,1);transition:transform 250ms cubic-bezier(0,0,.2,1);transition:transform 250ms cubic-bezier(0,0,.2,1),-webkit-transform 250ms cubic-bezier(0,0,.2,1)}.cdk-drag-animating{-webkit-transition:-webkit-transform .3s cubic-bezier(0,0,.2,1);transition:transform .3s cubic-bezier(0,0,.2,1);transition:transform .3s cubic-bezier(0,0,.2,1),-webkit-transform .3s cubic-bezier(0,0,.2,1)}.ls-workflow-link-container{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;min-height:4rem;min-width:20rem}.ls-workflow-link-container::before{content:\"\";-webkit-box-flex:0;-ms-flex:0 0 calc(50% - 2px);flex:0 0 calc(50% - 2px)}.ls-workflow-link-container .ls-link-button-container{-webkit-box-flex:0;-ms-flex:0 0 calc(50% - 2px);flex:0 0 calc(50% - 2px);-ms-flex-item-align:center;-ms-grid-row-align:center;align-self:center;height:-webkit-min-content;height:-moz-min-content;height:min-content;width:-webkit-min-content;width:-moz-min-content;width:min-content;padding-left:1rem}.ls-workflow-stepList-container{width:100%}.ls-workflow-step-container{-ms-flex-item-align:center;align-self:center;margin-left:auto;margin-right:auto;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center}.ls-workflow-step-container::before{content:\"\";-webkit-box-flex:0;-ms-flex:0 0 5rem;flex:0 0 5rem}.ls-workflow-step-container .ls-workflow-endpoint.with-content,.ls-workflow-step-container .ls-workflow-step{-ms-flex-item-align:center;-ms-grid-row-align:center;align-self:center}.ls-workflow-step-container .ls-workflow-step-external-buttons{padding-left:1rem;height:-webkit-min-content;height:-moz-min-content;height:min-content;-webkit-box-flex:0;-ms-flex:0 0 5rem;flex:0 0 5rem}.ls-workflow-step-external-buttons{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start}.ls-workflow-step-external-buttons .edit-step-btn{margin-right:1rem;cursor:pointer}.ls-workflow-step-external-buttons .edit-step-btn:hover{color:var(--primary)}.ls-workflow-step-external-buttons .delete-step-btn{cursor:pointer}.ls-workflow-step-external-buttons .delete-step-btn:hover{color:var(--danger)}.ls-workflow-editor{display:-ms-grid;display:grid;height:100%}.ls-workflow-editor .ls-link-button-container .add-step-btn{width:-webkit-max-content;width:-moz-max-content;width:max-content;cursor:pointer}.ls-workflow-editor .ls-link-button-container .add-step-btn .add-step-btn-txt{margin-left:.5em;opacity:0;-webkit-transition:opacity .2s ease-in-out;transition:opacity .2s ease-in-out}.ls-workflow-editor .ls-link-button-container.firstStep .add-step-btn .add-step-btn-txt,.ls-workflow-editor .ls-link-button-container:hover .add-step-btn .add-step-btn-txt{opacity:100}.ls-workflow-editor-default-layout{-ms-grid-columns:66% auto;grid-template-columns:66% auto;-ms-grid-rows:100%;grid-template-rows:100%}@media (min-width:1400px){.container{max-width:1340px}}@media (min-width:1600px){.container{max-width:1540px}}"]
                }] }
    ];
    /** @nocollapse */
    LsWorkflowEditorComponent.ctorParameters = function () { return []; };
    LsWorkflowEditorComponent.propDecorators = {
        endpointPopoverElementRef: [{ type: ViewChild, args: ['endpointPopoverHandle', { read: NgbPopover },] }],
        allCircuitEditionPopover: [{ type: ViewChildren, args: [NgbPopover,] }],
        popEditContentTemplateRef: [{ type: ContentChild, args: ['popEditContent',] }],
        popCreateContentTemplateRef: [{ type: ContentChild, args: ['popCreateContent',] }],
        endpointPopoverEditContentRef: [{ type: ContentChild, args: ['endpointPopoverEditContent',] }],
        stepDisplayContentRef: [{ type: ContentChild, args: ['stepDisplayContent',] }],
        lastStepDisplayContentRef: [{ type: ContentChild, args: ['lastStepDisplayContent',] }],
        endpointIsEditable: [{ type: Input }],
        getStepIconClassFct: [{ type: Input }],
        getStepTooltipFct: [{ type: Input }],
        confirmStepDeletionFct: [{ type: Input }],
        lastStepData: [{ type: Input }],
        workflowChange: [{ type: Output }],
        localizedMessages: [{ type: Input }],
        options: [{ type: Input }],
        workflow: [{ type: Input }]
    };
    return LsWorkflowEditorComponent;
}());
export { LsWorkflowEditorComponent };
if (false) {
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.currentEditedStepIdx;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.endpointPopoverElementRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.allCircuitEditionPopover;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.currentPopover;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.popEditContentTemplateRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.popCreateContentTemplateRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.endpointPopoverEditContentRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.stepDisplayContentRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.lastStepDisplayContentRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.messages;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.activeOptions;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.endpointIsEditable;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.getStepIconClassFct;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.getStepTooltipFct;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.confirmStepDeletionFct;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.lastStepData;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.workflowData;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.workflowChange;
}
