// @libriciel/ls-composants/esm2015/lib/component/password/password.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/password/password.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, forwardRef, Input, Renderer2, ViewChild } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
export class PasswordComponent {
    /**
     * @param {?} element
     * @param {?} renderer2
     */
    constructor(element, renderer2) {
        this.element = element;
        this.renderer2 = renderer2;
        this.showIcon = true;
        this.trim = false;
        this.isShown = false;
        this.onChange = (/**
         * @return {?}
         */
        () => {
            // tslint:disable-next-line:semicolon
        });
        this.onTouched = (/**
         * @return {?}
         */
        () => {
            // tslint:disable-next-line:semicolon
        });
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set value(value) {
        if (this.trim) {
            this._value = value.trim();
        }
        else {
            this._value = value;
        }
        this.onChange(this._value);
        this.onTouched();
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        /** @type {?} */
        const id = this.element.nativeElement.getAttribute('id');
        this.renderer2.removeAttribute(this.element.nativeElement, 'id');
        this.renderer2.setAttribute(this.input.nativeElement, 'id', id);
    }
    /**
     * @return {?}
     */
    togglePassword() {
        this.isShown = !this.isShown;
    }
    /**
     * @return {?}
     */
    onChangeSelection() {
        this.onChange(this._value);
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.onChange = fn;
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value) {
            this.value = value;
        }
    }
    /**
     * @param {?} event
     * @return {?}
     */
    trimValue(event) {
        if (this.trim) {
            event.target.value = event.target.value.trim();
        }
    }
}
PasswordComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-password',
                template: "<div class=\"form-group input-group-front-icon\" [class.input-group-front-icon]=\"showIcon\">\n  <span *ngIf=\"showIcon\">\n    <i class=\"fa fa-lock\"></i>\n  </span>\n  <input #input [attr.name]=\"name\" [(ngModel)]=\"value\" (ngModelChange)=\"onChangeSelection()\" [required]=\"required\"\n         class=\"form-control\" [type]=\"isShown ? 'text' : 'password'\" (keyup)=\"trimValue($event)\">\n\n  <span class=\"input-icon\" (click)=\"togglePassword()\">\n    <i class=\"fa\" [ngClass]=\"isShown ? 'fa-eye-slash' : 'fa-eye'\"></i>\n  </span>\n</div>\n\n\n\n\n\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => PasswordComponent)),
                        multi: true
                    }
                ],
                styles: [".input-icon{-ms-flex-align:center;-ms-flex-pack:center;-webkit-box-align:center;-webkit-box-pack:center;align-items:center;background-color:transparent;border-radius:5px 0 0 5px;color:var(--ls-grey-500);cursor:pointer;display:-webkit-inline-box;display:-ms-inline-flexbox;display:inline-flex;height:99%;justify-content:center;position:absolute;right:0;top:0;width:40px}.input-icon:hover{color:var(--ls-grey-700)}.input-group-end-icon,:host.input-group-end-icon{display:-webkit-box;display:-ms-flexbox;display:flex;position:relative}.input-group-end-icon input,:host.input-group-end-icon input{padding-right:36px}input.ls-action-input{cursor:pointer}.input-group.input-group-end-icon .input-icon,:host.input-group.input-group-end-icon .input-icon{z-index:3}.form-group{position:relative}input::-ms-clear,input::-ms-reveal{display:none}"]
            }] }
];
/** @nocollapse */
PasswordComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
PasswordComponent.propDecorators = {
    showIcon: [{ type: Input }],
    required: [{ type: Input }],
    trim: [{ type: Input }],
    input: [{ type: ViewChild, args: ['input',] }],
    name: [{ type: Input }],
    _value: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    PasswordComponent.prototype.showIcon;
    /** @type {?} */
    PasswordComponent.prototype.required;
    /** @type {?} */
    PasswordComponent.prototype.trim;
    /** @type {?} */
    PasswordComponent.prototype.input;
    /** @type {?} */
    PasswordComponent.prototype.name;
    /** @type {?} */
    PasswordComponent.prototype.isShown;
    /** @type {?} */
    PasswordComponent.prototype._value;
    /** @type {?} */
    PasswordComponent.prototype.onChange;
    /** @type {?} */
    PasswordComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    PasswordComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    PasswordComponent.prototype.renderer2;
}
