// @libriciel/ls-composants/esm2015/lib/pages/login/login-form/login-form.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/login-form/login-form.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
export class LoginFormComponent extends LoginPageModelClass {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        super(_DomSanitizationService);
        this._DomSanitizationService = _DomSanitizationService;
        this.showRememberMe = false;
        this.usernameInputName = 'username';
        this.passwordInputName = 'password';
        this.rememberMeInputName = 'rememberme';
        this.passwordReset = false;
        this.successMessage = '';
        this.forgotPassword = new EventEmitter();
    }
    /**
     * @return {?}
     */
    askForResetPassword() {
        this.forgotPassword.emit();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    preventFormSubmit(event) {
        super.preventFormSubmit(event);
    }
    /**
     * @return {?}
     */
    submitForm() {
        this.submit.emit({
            username: this.username,
            password: this.password,
            rememberMe: this.rememberMe
        });
    }
}
LoginFormComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-login-form',
                template: "<ls-login-model [logo]=\"logo\" [visualConfiguration]=\"_visualConfiguration\">\n  <section>\n    <h1 class=\"page-title\">Veuillez saisir vos identifiants de connexion</h1>\n    <form [action]=\"formAction\" method=\"post\" id=\"ls-login-form\" ngNoForm (submit)=\"preventFormSubmit($event)\">\n\n      <div *ngIf=\"showError\" class=\"alert alert-danger\">\n        <i class=\"fa fa-exclamation-triangle\"></i>\n        Identifiant ou mot de passe erron\u00E9\n      </div>\n\n      <div *ngIf=\"successMessage\" class=\"alert alert-success\">\n        <i class=\"fa fa-check\"></i>\n        {{successMessage}}\n      </div>\n\n      <div *ngIf=\"passwordReset\" class=\"alert alert-success\">\n        <i class=\"fa fa-check\"></i>\n        Vous pouvez desormais vous connecter avec votre identifiant et votre nouveau mot de passe\n      </div>\n\n      <div class=\"login-input-row\">\n        <label [attr.for]=\"usernameInputName\">\n          Identifiant\n        </label>\n        <div class=\"form-group input-group-front-icon\">\n                <span>\n                  <i class=\"fa fa-user\"></i>\n                </span>\n          <input [(ngModel)]=\"username\" [attr.name]=\"usernameInputName\" [attr.id]=\"usernameInputName\" class=\"form-control\" required>\n        </div>\n      </div>\n\n      <div class=\"login-input-row\">\n        <label [attr.for]=\"passwordInputName\">\n          Mot de passe\n        </label>\n        <div>\n          <ls-password [(ngModel)]=\"password\" [name]=\"passwordInputName\" [attr.id]=\"passwordInputName\" required></ls-password>\n        </div>\n      </div>\n\n      <div *ngIf=\"showRememberMe\" class=\"login-input-row\">\n        <div class=\"form-check\">\n          <input [(ngModel)]=\"rememberMe\" [attr.name]=\"rememberMeInputName\" [attr.id]=\"rememberMeInputName\" class=\"form-check-input\"\n                 type=\"checkbox\">\n          <label class=\"form-check-label\" [attr.for]=\"rememberMeInputName\">Se souvenir de moi</label>\n        </div>\n      </div>\n\n      <ng-content></ng-content>\n\n      <div class=\"float-right\">\n        <span (click)=\"askForResetPassword()\" class=\"btn btn-link forgot-passwd-lnk\">\n          Mot de passe oubli\u00E9 ?\n        </span>\n\n        <button (click)=\"submitForm()\" class=\"btn btn-primary\" form=\"ls-login-form\" type=\"submit\" [class.disabled]=\"submitted\">\n          <i *ngIf=\"!submitted\" class=\"fa fa-sign-in fa-sign-in-alt fa-fw\"></i>\n          <i *ngIf=\"submitted\" class=\"fa fa-spinner fa-spin fa-fw\"></i>\n          Se connecter\n        </button>\n      </div>\n    </form>\n  </section>\n</ls-login-model>\n",
                styles: [".page-title{font-size:1rem;font-weight:400;padding-top:3rem;padding-bottom:1.5rem;text-align:center}.alert{text-align:center}.forgot-passwd-lnk{cursor:pointer;margin-right:1rem}.float-right{padding-top:1rem}.login-input-row{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.login-input-row>label{width:33%;margin-right:1rem}.login-input-row>div{display:inline-block;width:67%}.login-input-row>.form-check{width:100%;padding-left:calc(33% + 35px)}"]
            }] }
];
/** @nocollapse */
LoginFormComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
LoginFormComponent.propDecorators = {
    showRememberMe: [{ type: Input }],
    usernameInputName: [{ type: Input }],
    passwordInputName: [{ type: Input }],
    rememberMeInputName: [{ type: Input }],
    passwordReset: [{ type: Input }],
    successMessage: [{ type: Input }],
    forgotPassword: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    LoginFormComponent.prototype.username;
    /** @type {?} */
    LoginFormComponent.prototype.password;
    /** @type {?} */
    LoginFormComponent.prototype.rememberMe;
    /** @type {?} */
    LoginFormComponent.prototype.showRememberMe;
    /** @type {?} */
    LoginFormComponent.prototype.usernameInputName;
    /** @type {?} */
    LoginFormComponent.prototype.passwordInputName;
    /** @type {?} */
    LoginFormComponent.prototype.rememberMeInputName;
    /** @type {?} */
    LoginFormComponent.prototype.passwordReset;
    /** @type {?} */
    LoginFormComponent.prototype.successMessage;
    /** @type {?} */
    LoginFormComponent.prototype.forgotPassword;
    /** @type {?} */
    LoginFormComponent.prototype._DomSanitizationService;
}
