// @libriciel/ls-composants/esm2015/lib/pages/login/reset-password/reset-password.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/reset-password/reset-password.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, Input } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
export class ResetPasswordComponent extends LoginPageModelClass {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        super(_DomSanitizationService);
        this._DomSanitizationService = _DomSanitizationService;
        this.samePassword = true;
        this.passwordInputName = 'password';
        this.passwordConfirmInputName = 'passwordconfirm';
    }
    /**
     * @param {?} event
     * @return {?}
     */
    preventFormSubmit(event) {
        super.preventFormSubmit(event);
    }
    /**
     * @return {?}
     */
    submitForm() {
        if (this.newpassword === this.newpasswordconfirm) {
            this.submit.emit({ password: this.newpassword });
        }
        else {
            this.samePassword = false;
        }
    }
}
ResetPasswordComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-reset-password',
                template: "<ls-login-model [visualConfiguration]=\"_visualConfiguration\" [logo]=\"logo\">\n  <section id=\"reset-page\">\n    <h1 class=\"page-title\">Veuillez saisir un nouveau mot de passe</h1>\n    <form ngNoForm action=\"\" id=\"reset-form\" method=\"post\" [action]=\"formAction\" (submit)=\"preventFormSubmit($event)\">\n\n      <div class=\"alert alert-danger\" *ngIf=\"!samePassword\">\n        <i class=\"fa fa-exclamation-triangle\"></i>\n        Les deux mots de passe ne sont pas identiques\n      </div>\n\n      <div>\n        <label [attr.for]=\"passwordInputName\">\n          Nouveau mot de passe\n        </label>\n        <div>\n          <ls-password [showIcon]=\"false\" [(ngModel)]=\"newpassword\" [name]=\"passwordInputName\" [attr.id]=\"passwordInputName\" required></ls-password>\n        </div>\n      </div>\n\n      <div>\n        <label [attr.for]=\"passwordConfirmInputName\">\n          Confirmation du nouveau mot de passe\n        </label>\n        <div>\n          <ls-password [showIcon]=\"false\" [attr.id]=\"passwordConfirmInputName\" [(ngModel)]=\"newpasswordconfirm\" [name]=\"passwordConfirmInputName\" required></ls-password>\n        </div>\n      </div>\n\n      <ng-content></ng-content>\n\n      <div class=\"float-right\">\n        <button type=\"submit\" (click)=\"submitForm()\" class=\"btn btn-primary\" [class.disabled]=\"submitted\">\n          <i *ngIf=\"!submitted\" class=\"fa fa-save fa-fw\"></i>\n          <i *ngIf=\"submitted\" class=\"fa fa-spinner fa-spin fa-fw\"></i>\n          Enregistrer\n        </button>\n      </div>\n    </form>\n  </section>\n</ls-login-model>\n",
                styles: ["span.btn{cursor:pointer}#reset-form .alert-soft-warning,.page-title{text-align:center}.page-title{font-size:1rem;font-weight:400;padding-bottom:1.5rem;padding-top:3rem}.login-input-row{-ms-flex-align:center;-webkit-box-align:center;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex}.login-input-row>label{margin-right:1rem;width:33%}.login-input-row>div{display:inline-block;width:67%}.login-input-row>.form-check{padding-left:calc(33% + 35px);width:100%}.floating-message{left:10%;position:absolute;right:10%;top:5%}.floating-message div{-ms-flex-align:center;-ms-flex-pack:center;-webkit-box-align:center;-webkit-box-pack:center;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;justify-content:center;min-height:4rem}.floating-message div i{padding-right:20px}.ls-login-footer{height:40px;width:100%}"]
            }] }
];
/** @nocollapse */
ResetPasswordComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
ResetPasswordComponent.propDecorators = {
    passwordInputName: [{ type: Input }],
    passwordConfirmInputName: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    ResetPasswordComponent.prototype.samePassword;
    /** @type {?} */
    ResetPasswordComponent.prototype.newpassword;
    /** @type {?} */
    ResetPasswordComponent.prototype.newpasswordconfirm;
    /** @type {?} */
    ResetPasswordComponent.prototype.passwordInputName;
    /** @type {?} */
    ResetPasswordComponent.prototype.passwordConfirmInputName;
    /** @type {?} */
    ResetPasswordComponent.prototype._DomSanitizationService;
}
