// @libriciel/ls-composants/esm5/lib/pages/login/login/login.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/login/login.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
var LoginComponent = /** @class */ (function (_super) {
    tslib_1.__extends(LoginComponent, _super);
    function LoginComponent(_DomSanitizationService) {
        var _this = _super.call(this, _DomSanitizationService) || this;
        _this._DomSanitizationService = _DomSanitizationService;
        _this.testMode = false;
        _this.currentPage = 'login';
        _this.loginFormAction = '';
        _this.loginShowError = false;
        _this.loginSuccessMessage = '';
        _this.loginIsFormAjax = false;
        _this.loginPasswordReset = false;
        _this.loginShowRememberMe = false;
        _this.loginUsernameInputName = 'username';
        _this.loginPasswordInputName = 'password';
        _this.loginRememberMeInputName = 'rememberme';
        _this.forgotFormAction = '';
        _this.forgotIsFormAjax = false;
        _this.forgotOneFieldForgot = false;
        _this.forgotUsernameInputName = 'username';
        _this.forgotEmailInputName = 'email';
        _this.forgotUsernameormailInputName = 'usernameormail';
        _this.resetFormAction = '';
        _this.resetIsFormAjax = false;
        _this.resetPasswordInputName = 'password';
        _this.resetPasswordConfirmInputName = 'passwordconfirm';
        _this.loginSubmit = new EventEmitter();
        _this.forgotSubmit = new EventEmitter();
        _this.resetSubmit = new EventEmitter();
        return _this;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    LoginComponent.prototype.login = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.loginSubmit.emit(event);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    LoginComponent.prototype.forgot = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.forgotSubmit.emit(event);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    LoginComponent.prototype.reset = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.resetSubmit.emit(event);
    };
    LoginComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-login',
                    template: "<div *ngIf=\"testMode\" class=\"testmode-container\">\n  <span>Mode test</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'login'\">Page de login</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'forgot'\">Mot de passe oubli\u00E9</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'forgot-success'\">Mot de passe oubli\u00E9 demand\u00E9</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'reset'\">Remise \u00E0 z\u00E9ro</span>\n</div>\n\n<ls-login-form\n  *ngIf=\"currentPage === 'login'\"\n  [visualConfiguration]=\"_visualConfiguration\"\n  [logo]=\"logo\"\n\n  [formAction]=\"loginFormAction\"\n  [showError]=\"loginShowError\"\n  [successMessage]=\"loginSuccessMessage\"\n  [passwordReset]=\"loginPasswordReset\"\n  [isFormAjax]=\"loginIsFormAjax\"\n  [showRememberMe]=\"loginShowRememberMe\"\n  [passwordInputName]=\"loginPasswordInputName\"\n  [usernameInputName]=\"loginUsernameInputName\"\n  [rememberMeInputName]=\"loginRememberMeInputName\"\n\n  (forgotPassword)=\"currentPage = 'forgot'\"\n  (submit)=\"login($event)\"\n>\n  <ng-content select=\".login-form-addons\"></ng-content>\n</ls-login-form>\n\n<ls-forgot-password\n  *ngIf=\"currentPage === 'forgot'\"\n  [visualConfiguration]=\"_visualConfiguration\"\n  [logo]=\"logo\"\n\n  [formAction]=\"forgotFormAction\"\n  [isFormAjax]=\"forgotIsFormAjax\"\n  [oneFieldForgot]=\"forgotOneFieldForgot\"\n  [usernameInputName]=\"forgotUsernameInputName\"\n  [emailInputName]=\"forgotEmailInputName\"\n  [usernameormailInputName]=\"forgotUsernameormailInputName\"\n\n  (backToLogin)=\"currentPage = 'login'\"\n  (submit)=\"forgot($event)\"\n>\n  <ng-content select=\".forgot-form-addons\"></ng-content>\n</ls-forgot-password>\n\n<ls-forgot-password-success\n  *ngIf=\"currentPage === 'forgot-success'\"\n  [logo]=\"logo\"\n  [visualConfiguration]=\"_visualConfiguration\"\n\n  (backToLogin)=\"currentPage = 'login'\"\n></ls-forgot-password-success>\n\n<ls-reset-password\n  *ngIf=\"currentPage === 'reset'\"\n  [logo]=\"logo\"\n  [visualConfiguration]=\"_visualConfiguration\"\n\n  [formAction]=\"resetFormAction\"\n  [isFormAjax]=\"resetIsFormAjax\"\n  [passwordInputName]=\"resetPasswordInputName\"\n  [passwordConfirmInputName]=\"resetPasswordConfirmInputName\"\n\n  (submit)=\"reset($event)\"\n>\n  <ng-content select=\".reset-form-addons\"></ng-content>\n</ls-reset-password>\n",
                    styles: [".testmode-container{zoom:.7;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;position:absolute;top:5px;left:5px;z-index:1}.testmode-container *{cursor:pointer;margin-bottom:5px}"]
                }] }
    ];
    /** @nocollapse */
    LoginComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    LoginComponent.propDecorators = {
        testMode: [{ type: Input }],
        currentPage: [{ type: Input }],
        loginFormAction: [{ type: Input }],
        loginShowError: [{ type: Input }],
        loginSuccessMessage: [{ type: Input }],
        loginIsFormAjax: [{ type: Input }],
        loginPasswordReset: [{ type: Input }],
        loginShowRememberMe: [{ type: Input }],
        loginUsernameInputName: [{ type: Input }],
        loginPasswordInputName: [{ type: Input }],
        loginRememberMeInputName: [{ type: Input }],
        forgotFormAction: [{ type: Input }],
        forgotIsFormAjax: [{ type: Input }],
        forgotOneFieldForgot: [{ type: Input }],
        forgotUsernameInputName: [{ type: Input }],
        forgotEmailInputName: [{ type: Input }],
        forgotUsernameormailInputName: [{ type: Input }],
        resetFormAction: [{ type: Input }],
        resetIsFormAjax: [{ type: Input }],
        resetPasswordInputName: [{ type: Input }],
        resetPasswordConfirmInputName: [{ type: Input }],
        loginSubmit: [{ type: Output }],
        forgotSubmit: [{ type: Output }],
        resetSubmit: [{ type: Output }]
    };
    return LoginComponent;
}(LoginPageModelClass));
export { LoginComponent };
if (false) {
    /** @type {?} */
    LoginComponent.prototype.testMode;
    /** @type {?} */
    LoginComponent.prototype.currentPage;
    /** @type {?} */
    LoginComponent.prototype.loginFormAction;
    /** @type {?} */
    LoginComponent.prototype.loginShowError;
    /** @type {?} */
    LoginComponent.prototype.loginSuccessMessage;
    /** @type {?} */
    LoginComponent.prototype.loginIsFormAjax;
    /** @type {?} */
    LoginComponent.prototype.loginPasswordReset;
    /** @type {?} */
    LoginComponent.prototype.loginShowRememberMe;
    /** @type {?} */
    LoginComponent.prototype.loginUsernameInputName;
    /** @type {?} */
    LoginComponent.prototype.loginPasswordInputName;
    /** @type {?} */
    LoginComponent.prototype.loginRememberMeInputName;
    /** @type {?} */
    LoginComponent.prototype.forgotFormAction;
    /** @type {?} */
    LoginComponent.prototype.forgotIsFormAjax;
    /** @type {?} */
    LoginComponent.prototype.forgotOneFieldForgot;
    /** @type {?} */
    LoginComponent.prototype.forgotUsernameInputName;
    /** @type {?} */
    LoginComponent.prototype.forgotEmailInputName;
    /** @type {?} */
    LoginComponent.prototype.forgotUsernameormailInputName;
    /** @type {?} */
    LoginComponent.prototype.resetFormAction;
    /** @type {?} */
    LoginComponent.prototype.resetIsFormAjax;
    /** @type {?} */
    LoginComponent.prototype.resetPasswordInputName;
    /** @type {?} */
    LoginComponent.prototype.resetPasswordConfirmInputName;
    /** @type {?} */
    LoginComponent.prototype.loginSubmit;
    /** @type {?} */
    LoginComponent.prototype.forgotSubmit;
    /** @type {?} */
    LoginComponent.prototype.resetSubmit;
    /** @type {?} */
    LoginComponent.prototype._DomSanitizationService;
}
