// @libriciel/ls-composants/esm2015/lib/pages/login/component/forgot-password-success/forgot-password-success.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/component/forgot-password-success/forgot-password-success.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
import { LoginMessages } from '../../i18n/login-messages';
export class ForgotPasswordSuccessComponent extends LoginPageModelClass {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        super(_DomSanitizationService);
        this._DomSanitizationService = _DomSanitizationService;
        this.backToLogin = new EventEmitter();
        this.loginMessages = LoginMessages;
    }
    /**
     * @return {?}
     */
    backToLoginPage() {
        this.backToLogin.emit();
    }
}
ForgotPasswordSuccessComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-forgot-password-success',
                template: "<ls-login-model [logo]=\"logo\" [visualConfiguration]=\"_visualConfiguration\">\n  <section class=\"forgot-success-page\">\n\n    <div class=\"alert alert-success\">\n      <i class=\"fa fa-lg fa-check\"></i>\n      Si vos informations sont correctes, un e-mail vous a \u00E9t\u00E9 envoy\u00E9.\n    </div>\n    <ls-button (click)=\"backToLoginPage()\" [title]=\"loginMessages.BACK_TO_CONNEXION_PAGE\"></ls-button>\n  </section>\n</ls-login-model>\n",
                styles: [".alert{margin-top:2rem;padding:1.5rem;text-align:center;width:600px}.forgot-success-page{-ms-flex-align:center;-ms-flex-direction:column;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:vertical;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column}.forgot-success-page>*{margin-bottom:2rem}"]
            }] }
];
/** @nocollapse */
ForgotPasswordSuccessComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
ForgotPasswordSuccessComponent.propDecorators = {
    backToLogin: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    ForgotPasswordSuccessComponent.prototype.backToLogin;
    /** @type {?} */
    ForgotPasswordSuccessComponent.prototype.loginMessages;
    /** @type {?} */
    ForgotPasswordSuccessComponent.prototype._DomSanitizationService;
}
