// @libriciel/ls-composants/esm2015/lib/ui-concepts/common-styles-constants.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/ui-concepts/common-styles-constants.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
export class CommonStylesConstants {
    /**
     * @param {?} style
     * @return {?}
     */
    static getStyleClass(style) {
        return `ls-${style.toString()}`;
    }
    /**
     * @param {?} style
     * @return {?}
     */
    static getButtonStyleClass(style) {
        return `btn-${style.toString()}`;
    }
    /**
     * @param {?} style
     * @return {?}
     */
    static getButtonOutlineStyleClass(style) {
        return `btn-outline-${style.toString()}`;
    }
    /**
     * @return {?}
     */
    static getButtonLinkStyleClass() {
        return `btn-link`;
    }
    /**
     * @return {?}
     */
    static getButtonMenuItemStyleClass() {
        return `menu-item ${CommonStylesConstants.getButtonLinkStyleClass()}`;
    }
    /**
     * @param {?} style
     * @return {?}
     */
    static getOnHoverStyleClass(style) {
        return this.getStyleClass(style) + '-on-hover';
    }
    // static getSuccessStyleClass(): string {
    //   return this.getStyleClass(Style.SUCCESS);
    // }
    // static getWarningStyleClass(): string {
    //   return this.getStyleClass(Style.WARNING);
    // }
    //
    // static getErrorStyleClass(): string {
    //   return this.getStyleClass(Style.DANGER);
    // }
    /**
     * @param {?} style
     * @return {?}
     */
    static getPanelStyleClass(style) {
        return 'ls-panel ' + this.getStyleClass(style) + '-panel';
    }
    /**
     * @return {?}
     */
    static getButtonSmallSizeClass() {
        return 'fa-sm';
    }
    /**
     * @return {?}
     */
    static getButtonLargeSizeClass() {
        return 'fa-lg';
    }
}
