// @libriciel/ls-composants/esm5/workflows/workflows/ls-workflow-instance-viewer/ls-workflow-instance-viewer.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflow-instance-viewer/ls-workflow-instance-viewer.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChild, Input, TemplateRef } from '@angular/core';
import { WorkflowInstance } from '../../shared/model/workflows/workflow-instance';
import { StepState } from '../../shared/model/workflows/step-state';
import { WorkflowInstanceLocalizedMessages, WorkflowInstanceOptions } from "../../shared/model/workflows/workflow-instance-option";
var LsWorkflowInstanceViewerComponent = /** @class */ (function () {
    function LsWorkflowInstanceViewerComponent() {
        this.messages = {
            FIRST_STEP_TXT: 'Start',
            LAST_STEP_TXT: 'End'
        };
        this.reducedMode = true;
        this.isOver = false;
        this.isRejected = false;
        this._options = {
            startIconClass: 'far fa-flag',
            endIconClass: 'fas fa-flag-checkered',
            validatedIconClass: 'fas fa-check',
            rejectedIconClass: 'fas fa-times',
            hideStart: false
        };
    }
    Object.defineProperty(LsWorkflowInstanceViewerComponent.prototype, "options", {
        get: /**
         * @return {?}
         */
        function () {
            return this._options;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this._options = Object.assign(this._options, value);
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LsWorkflowInstanceViewerComponent.prototype, "localizedMessages", {
        get: /**
         * @return {?}
         */
        function () {
            return this.messages;
        },
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.messages = Object.assign(this.messages, value);
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        console.warn('instanciating ls workflow instance viewer!');
        if (!this.workflowData) {
            console.error('Missing input circuit data');
            this.workflowData = new WorkflowInstance();
        }
        if (!this.messages) {
            throw new Error('Missing localised messages for LsWorkflowViewerComponent!');
        }
        if (!this.getStepIconClassFct) {
            throw new Error('Missing getStepIconClassFct for LsWorkflowViewerComponent!');
        }
        if (!this.getStepTooltipFct) {
            throw new Error('Missing getStepTooltipFct for LsWorkflowViewerComponent!');
        }
    };
    /**
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.ngOnChanges = /**
     * @return {?}
     */
    function () {
        console.debug('LsWorkflowViewerComponent::ngOnChanges, ', this);
        this.activeStep = this.workflowData.currentStepIdx;
        this.isOver = this.workflowData.isOver;
        this.isRejected = this.workflowData.rejected;
    };
    /**
     * @param {?} step
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.getStepIconClass = /**
     * @param {?} step
     * @return {?}
     */
    function (step) {
        return this.getStepIconClassFct(step);
    };
    /**
     * @param {?} step
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.getStepTooltip = /**
     * @param {?} step
     * @return {?}
     */
    function (step) {
        return this.getStepTooltipFct(step);
    };
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.isActiveStep = /**
     * @param {?} stepIdx
     * @return {?}
     */
    function (stepIdx) {
        return !this.isRejected && !this.isOver && stepIdx === this.activeStep;
    };
    /**
     * @param {?} step
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.isPassed = /**
     * @param {?} step
     * @return {?}
     */
    function (step) {
        return (step.state === StepState.VALIDATED) || (step.state === StepState.REJECTED);
    };
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.isActualValidator = /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    function (step, validator) {
        return step.validatedBy.some((/**
         * @param {?} elem
         * @return {?}
         */
        function (elem) { return elem.id === validator.id; }));
    };
    /**
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.isCircuitOver = /**
     * @return {?}
     */
    function () {
        return this.isOver && !this.isRejected;
    };
    /**
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.getCircuitEndAdditionalClasses = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var classes = [];
        if (this.getCircuitEndAdditionalClassesFct) {
            classes.push(this.getCircuitEndAdditionalClassesFct());
        }
        else {
            classes.push(this._options.additionalFinalEndpointClasses || '');
        }
        if (this.reducedMode) {
            classes.push('reduced-end-step');
        }
        return classes;
    };
    /**
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.getAdditionalValidatorListClasses = /**
     * @return {?}
     */
    function () {
        //wa-circuit-instance
    };
    /**
     * @param {?} step
     * @param {?} stepIdx
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.getStepAdditionalClasses = /**
     * @param {?} step
     * @param {?} stepIdx
     * @return {?}
     */
    function (step, stepIdx) {
        // this.isActiveStep(stepIdx) ? 'wa-active-step wa-white-icn' : ''
        /** @type {?} */
        var classes = [];
        if (this.isActiveStep(stepIdx)) {
            classes.push('ls-active-step ls-white-icn');
        }
        if (this.reducedMode) {
            classes.push('ls-reduced-step');
        }
        if (!this.reducedMode) {
            if (step.state === StepState.SKIPPED) {
                classes.push('wa-circuit-skipped-step');
            }
        }
        if (step.state === StepState.VALIDATED) {
            classes.push('ls-validated-step');
        }
        return classes;
    };
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    LsWorkflowInstanceViewerComponent.prototype.getValidatorClasses = /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    function (step, validator) {
        if (this.isActualValidator(step, validator)) {
            if (step.state === StepState.REJECTED) {
                return this.options.rejectedIconClass;
            }
            else {
                return this.options.validatedIconClass;
            }
        }
        return '';
    };
    LsWorkflowInstanceViewerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-workflow-instance-viewer',
                    template: "<div class=\"ls-workflow-display\">\n  <div class=\"ls-workflow-canevas\">\n    <div class=\"circuit-start\">\n      <div *ngIf=\"!options.hideStart\" class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa fa-2x\" [ngClass]=\"options.startIconClass\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.FIRST_STEP_TXT\"></span>\n      </div>\n    </div>\n    <div class=\"steps-container\">\n      <div *ngFor=\"let step of workflowData.steps; index as i\">\n        <div class=\"ls-workflow-display-step-container\">\n          <div class=\"ls-workflow-step\" [ngClass]=\"getStepAdditionalClasses(step, i)\">\n            <div class=\"ls-step-top-line\">\n              <div class=\"step-number\">{{i + 1}}.</div>\n              <div class=\"step-name\">{{step.name}}</div>\n              <div [ngClass]=\"getStepIconClass(step)\" class=\"step-icn fa\" ngbTooltip=\"{{getStepTooltip(step)}}\"></div>\n            </div>\n            <div class=\"ls-display-step-separator\"></div>\n            <div class=\"ls-step-validator-list\" [ngClass]=\"getAdditionalValidatorListClasses\">\n              <ng-container *ngIf=\"isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"passedStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n              <ng-container *ngIf=\" ! isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\" class=\"ls-step-validator\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"pendingStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n            </div>\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"ls-workflow-end\" [ngClass]=\"getCircuitEndAdditionalClasses()\">\n      <div class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.endIconClass\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.LAST_STEP_TXT\"></span>\n      </div>\n    </div>\n  </div>\n</div>\n",
                    styles: ["body{background-color:var(--ls-grey-50)}.active-btn{color:#42a1e8}.ls-workflow-canevas{-ms-grid-column:1;grid-column:1;height:100%;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start}.ls-workflow-step{background-color:#fff;-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);box-shadow:0 0 .25rem rgba(0,0,0,.15);border:none;width:var(--ls-workflow-step-width,20rem);min-height:6rem}.ls-workflow-step .ls-step-top-line{padding:.5rem;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row;-webkit-box-pack:justify;-ms-flex-pack:justify;justify-content:space-between;-webkit-box-align:center;-ms-flex-align:center;align-items:center;font-weight:700}.ls-workflow-step .ls-display-step-separator{width:100%;border-bottom:1px solid #d3d3d3}.ls-workflow-step .ls-step-validator-list{padding:.5rem .5rem .5rem 1.5rem;color:#5b5b5b}.ls-workflow-step.selected{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.ls-workflow-step .step-icn{width:1.7rem;height:1.7rem}.ls-workflow-endpoint-content{margin:auto;text-align:center}.ls-workflow-endpoint-content .ls-step-bounds-txt{color:var(--primary);font-weight:700}.ls-workflow-endpoint-content .ls-workflow-endpoint-icn{color:var(--primary)}.ls-workflow-endpoint-content.not-active .ls-step-bounds-txt,.ls-workflow-endpoint-content.not-active .ls-workflow-endpoint-icn{color:#5b5b5b}.ls-workflow-endpoint-content .ls-step-validator-list{text-align:left}.ls-actual-validator{font-weight:700}.ls-actual-validator::before{content:'\\f00c';font-family:var(--ls-font-awesome-family, \"Font Awesome 5 Free\");color:var(--primary);font-size:1em;position:absolute;left:.5em}.ls-rejector{font-weight:700}.ls-rejector::before{content:'\\f00d';font-family:var(--ls-font-awesome-family, \"Font Awesome 5 Free\");color:var(--danger);font-size:1em;position:absolute;left:.5em}.wa-comment-icon{color:var(--primary);font-size:1em;margin-right:.5em}.ls-workflow-display .ls-workflow-step{padding-left:0;padding-right:0;margin:1rem auto}.ls-workflow-display .ls-workflow-step .ls-step-validator-list{padding:1rem 1rem 1rem .5rem}.ls-workflow-display .ls-workflow-step .ls-step-validator-list.ls-workflow-instance{padding-left:2rem}.ls-workflow-display .ls-workflow-step.ls-reduced-step{-webkit-box-shadow:none;box-shadow:none;border:none;background-color:rgba(0,0,0,0);width:var(--ls-workflow-reduced-step-width,16rem);min-height:unset;display:-ms-grid;display:grid;-ms-grid-columns:(1fr)[12];grid-template-columns:repeat(12,1fr);-ms-grid-rows:2em auto;grid-template-rows:2em auto;grid-row-gap:.5em}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line{grid-column:1/-1;-ms-grid-row:1;grid-row:1;display:-ms-grid;display:grid;-ms-grid-columns:1fr 2fr 9fr;grid-template-columns:1fr 2fr 9fr;-ms-grid-rows:1fr;grid-template-rows:1fr;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;padding:0}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-number{display:none}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-icn{-ms-grid-column:2;grid-column:2;-ms-grid-row:1;grid-row:1;font-size:24px;margin:auto;width:unset;height:unset}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-name{-ms-grid-column:3;grid-column:3;-ms-grid-row:1;grid-row:1}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-display-step-separator{-ms-grid-column:3;grid-column:3;-ms-grid-row:2;grid-row:2;border-left:1px solid #d3d3d3;border-bottom:0}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-validator-list{grid-column:4/-1;-ms-grid-row:2;grid-row:2;padding:.5em;font-size:.8rem;font-style:italic;color:#d3d3d3}.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line,.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line .step-icn.test-icon-class,.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line .step-icn.test-icon-class:before{color:#23a237}.ls-workflow-display .reduced-end-step .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:1em}.ls-workflow-display .ls-workflow-endpoint-content{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:horizontal;-webkit-box-direction:normal;-ms-flex-direction:row;flex-direction:row}.ls-workflow-display .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:.5rem}.ls-workflow-display .ls-step-top-line{-webkit-box-pack:start;-ms-flex-pack:start;justify-content:flex-start;padding-left:.5rem}.ls-workflow-display .ls-step-top-line .step-name{-webkit-box-flex:1;-ms-flex:1;flex:1;margin-left:.7rem}.ls-workflow-display .ls-workflow-display-step-container{position:relative}.ls-workflow-display .ls-active-step:not(.ls-reduced-step) .ls-step-top-line{background-color:var(--primary);color:#fff;margin:0}.ls-workflow-display .ls-active-step:not(.ls-reduced-step) .ls-step-top-line .step-icn{color:#fff}.ls-workflow-display .ls-active-step::before{content:'\\f061';font-family:var(--ls-font-awesome-family, \"Font Awesome 5 Free\");font-style:normal;font-variant:normal;text-rendering:auto;line-height:1;font-weight:900;color:var(--ls-workflow-arrow-color,var(--primary));font-size:2em;position:absolute;left:-1em}.ls-workflow-display .ls-active-step.end-step::before{font-size:1.2em}.ls-workflow-end{position:relative}"]
                }] }
    ];
    /** @nocollapse */
    LsWorkflowInstanceViewerComponent.ctorParameters = function () { return []; };
    LsWorkflowInstanceViewerComponent.propDecorators = {
        passedStepTemplateRef: [{ type: ContentChild, args: ['passedStepContent',] }],
        pendingStepTemplateRef: [{ type: ContentChild, args: ['pendingStepContent',] }],
        workflowData: [{ type: Input, args: ['src',] }],
        activeStep: [{ type: Input }],
        getStepIconClassFct: [{ type: Input }],
        getStepTooltipFct: [{ type: Input }],
        getCircuitEndAdditionalClassesFct: [{ type: Input }],
        reducedMode: [{ type: Input }],
        options: [{ type: Input }],
        localizedMessages: [{ type: Input }]
    };
    return LsWorkflowInstanceViewerComponent;
}());
export { LsWorkflowInstanceViewerComponent };
if (false) {
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.passedStepTemplateRef;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.pendingStepTemplateRef;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.workflowData;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.activeStep;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.messages;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getStepIconClassFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getStepTooltipFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getCircuitEndAdditionalClassesFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.reducedMode;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.isOver;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.isRejected;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype._options;
}
