// @libriciel/ls-composants/esm2015/lib/pages/login/login-model/login-model.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/login-model/login-model.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, HostListener, Input, ViewChild } from '@angular/core';
import { DomSanitizer } from '@angular/platform-browser';
import { defaultConfiguration } from '../login-config/login-config.model';
export class LoginModelComponent {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        this._DomSanitizationService = _DomSanitizationService;
        this._visualConfiguration = defaultConfiguration;
    }
    /**
     * @param {?} config
     * @return {?}
     */
    set visualConfiguration(config) {
        this._visualConfiguration = typeof config === 'string' ? JSON.parse(config) : config;
        this.getImageDimensions(this._visualConfiguration.imgData);
    }
    /**
     * @param {?} file
     * @return {?}
     */
    getImageDimensions(file) {
        /** @type {?} */
        const i = new Image();
        i.onload = (/**
         * @return {?}
         */
        () => {
            this.imgSize = { w: i.width, h: i.height };
            this.handleImgPositionning();
        });
        i.src = file;
    }
    /**
     * @return {?}
     */
    handleImgPositionning() {
        if (this.imgSize !== null) {
            /** @type {?} */
            const w = this.imgEl.nativeElement.offsetWidth;
            /** @type {?} */
            const h = this.imgEl.nativeElement.offsetHeight;
            /** @type {?} */
            let finalSize;
            /** @type {?} */
            let width = '90%';
            /** @type {?} */
            let height = '90%';
            if (this.imgSize.w > this.imgSize.h) {
                finalSize = (w * 0.9 * this.imgSize.h) / (h * this.imgSize.w) * 100;
                height = finalSize + '%';
                if (finalSize > 90) {
                    width = 'auto';
                    height = '90%';
                }
            }
            else {
                finalSize = (h * 0.9 * this.imgSize.w) / (w * this.imgSize.h) * 100;
                width = finalSize + '%';
                if (finalSize > 90) {
                    height = 'auto';
                    width = '90%';
                }
            }
            this.imgEl.nativeElement.style['background-size'] = width + ' ' + height;
        }
    }
}
LoginModelComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-login-model',
                template: "<main class=\"connection-page\">\n  <section class=\"main\">\n    <div [style.background-color]=\"_visualConfiguration.colorBackground\"\n         [style.color]=\"_visualConfiguration.colorText\"\n         [style.display]=\"_visualConfiguration.showCustom ? '': 'none'\"\n         class=\"coll-side\">\n      <div #imgcontainer class=\"img-container\" [ngStyle]=\"{'background-image': 'url(&quot;' + _visualConfiguration.imgData + '&quot;)'}\">\n\n      </div>\n      <div class=\"sub-txt\">\n        <span>{{_visualConfiguration.subText}}</span>\n      </div>\n    </div>\n    <div class=\"form-side\">\n\n      <div class=\"form-content\" [ngClass]=\"_visualConfiguration.showInformation && _visualConfiguration.informationText ? 'form-content-padding' : ''\">\n        <div class=\"soft-logo\">\n          <img id=\"libriciel-logo\" [src]=\"logo\" alt=\"Image du Libriciel\" src=\"\">\n        </div>\n\n        <ng-content></ng-content>\n\n      </div>\n    </div>\n  </section>\n  <section *ngIf=\"_visualConfiguration.showInformation && _visualConfiguration.informationText\" class=\"floating-message\">\n    <div [ngClass]=\"'alert-' + _visualConfiguration.informationType\" class=\"alert\">\n      <i [class.fa-exclamation-circle]=\"_visualConfiguration.informationType === 'warning'\"\n         [class.fa-exclamation-triangle]=\"_visualConfiguration.informationType === 'danger'\"\n         [class.fa-info-circle]=\"_visualConfiguration.informationType === 'info'\"\n         class=\"fa fa-lg\"\n      ></i>\n      {{_visualConfiguration.informationText}}</div>\n  </section>\n</main>\n\n\n\n\n",
                styles: [".hidden{display:none!important}.soft-logo{text-align:center;height:100%}.connection-page{height:calc(100vh - 40px);position:relative;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column}span.btn{cursor:pointer}.main{-webkit-box-flex:1;-ms-flex-positive:1;flex-grow:1;-ms-flex-negative:1;flex-shrink:1;width:100%;display:-webkit-box;display:-ms-flexbox;display:flex}.coll-side{width:30%;display:-ms-grid;display:grid;-ms-grid-rows:1fr 1rem 40%;-ms-grid-columns:1fr;grid-template:\"img\" 1fr \"text\" 40%/1fr;justify-items:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;grid-gap:1rem}.coll-side .img-container{-ms-grid-row:1;-ms-grid-column:1;grid-area:img;max-width:calc(100% - 2rem);width:100%;height:100%;text-align:center;background-position:bottom center;background-repeat:no-repeat}.coll-side .sub-txt{-ms-grid-row:3;-ms-grid-column:1;grid-area:text;max-width:100%;text-align:center;padding:1rem;-ms-flex-item-align:baseline;align-self:baseline}.form-side{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-flex:1;-ms-flex-positive:1;flex-grow:1;background-color:#fff}.form-side .form-content-padding{padding-top:100px!important}.form-side .form-content{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;width:500px;padding-top:50px;padding-bottom:50px}.form-side .form-content .page-title{font-size:1rem;font-weight:400;padding-top:3rem;padding-bottom:1.5rem;text-align:center}.floating-message{left:10%;right:10%;top:5%;position:absolute}.floating-message div{-webkit-box-align:center;-ms-flex-align:center;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;padding-top:1rem;padding-bottom:1rem}.floating-message div i{padding-right:20px}#libriciel-logo{height:auto;width:100%;max-width:100%}@media screen and (max-width:800px){.coll-side{display:none}}"]
            }] }
];
/** @nocollapse */
LoginModelComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
LoginModelComponent.propDecorators = {
    logo: [{ type: Input }],
    imgEl: [{ type: ViewChild, args: ['imgcontainer',] }],
    visualConfiguration: [{ type: Input }],
    handleImgPositionning: [{ type: HostListener, args: ['window:resize',] }]
};
if (false) {
    /** @type {?} */
    LoginModelComponent.prototype.logo;
    /** @type {?} */
    LoginModelComponent.prototype._visualConfiguration;
    /** @type {?} */
    LoginModelComponent.prototype.imgEl;
    /** @type {?} */
    LoginModelComponent.prototype.imgSize;
    /** @type {?} */
    LoginModelComponent.prototype._DomSanitizationService;
}
