// @libriciel/ls-composants/esm2015/lib/component/libersign/libersign.native.util.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.native.util.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { LibersignUtil } from './libersign.util';
export class LibersignNativeUtil extends LibersignUtil {
    /**
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    init(updateUrl, appletUrl) {
        this.status.certsLoading = true;
        try {
            LiberSign.setUpdateUrl(updateUrl);
        }
        catch (error) {
            console.log('error catched');
        }
        this.loadCertificates();
    }
    /**
     * @return {?}
     */
    loadCertificates() {
        LiberSign.getCertificates().then((/**
         * @param {?} cs
         * @return {?}
         */
        (cs) => {
            if (cs !== undefined) {
                this.certificates = cs;
                this.selectCertificate(this.favorite);
                this.commonLoadCert();
            }
        }), (/**
         * @param {?} err
         * @return {?}
         */
        err => {
            this.status.certsError = true;
        }));
    }
    /**
     * @param {?} dataToSign
     * @return {?}
     */
    doSign(dataToSign) {
        LiberSign.sign(this.currentCert.ID, { dataToSign: dataToSign }).then((/**
         * @param {?} sig
         * @return {?}
         */
        sig => {
            this.signatures.next(sig);
        }));
    }
    /**
     * @param {?} cert
     * @return {?}
     */
    selectCertificate(cert) {
        this.currentCert = cert;
    }
    /**
     * @return {?}
     */
    destroy() {
        // Nothing to do here
    }
}
