// @libriciel/ls-composants/esm2015/lib/component/libersign/libersign.util.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.util.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { LibersignStatus } from './libersign-status.model';
import { BehaviorSubject } from 'rxjs';
/**
 * @abstract
 */
export class LibersignUtil {
    constructor() {
        this.status = new LibersignStatus();
        this.FAVORITEKEY = 'libersign-fav';
        this._signatures = [];
        this.signatures = new BehaviorSubject(this._signatures);
    }
    /**
     * @param {?} cert
     * @return {?}
     */
    set favorite(cert) {
        localStorage.setItem(this.FAVORITEKEY, cert.ID);
        this._favorite = cert;
    }
    /**
     * @return {?}
     */
    get favorite() {
        if (this._favorite !== undefined) {
            return this._favorite;
        }
        else {
            /** @type {?} */
            const id = localStorage.getItem(this.FAVORITEKEY);
            for (const cert of this.certificates) {
                if (cert.ID === id) {
                    this._favorite = cert;
                    return cert;
                }
            }
        }
        return undefined;
    }
    /**
     * @param {?} dataToSign
     * @return {?}
     */
    sign(dataToSign) {
        this.status.dataToSignLoading = false;
        this.status.signLoading = true;
        this.doSign(dataToSign);
    }
    /**
     * @return {?}
     */
    commonLoadCert() {
        this.status.certsLoading = false;
        this.status.certsLoaded = true;
    }
}
if (false) {
    /** @type {?} */
    LibersignUtil.prototype.status;
    /** @type {?} */
    LibersignUtil.prototype.certificates;
    /** @type {?} */
    LibersignUtil.prototype.currentCert;
    /** @type {?} */
    LibersignUtil.prototype._favorite;
    /** @type {?} */
    LibersignUtil.prototype.FAVORITEKEY;
    /** @type {?} */
    LibersignUtil.prototype.signatures;
    /** @type {?} */
    LibersignUtil.prototype._signatures;
    /**
     * @abstract
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    LibersignUtil.prototype.init = function (updateUrl, appletUrl) { };
    /**
     * @abstract
     * @return {?}
     */
    LibersignUtil.prototype.loadCertificates = function () { };
    /**
     * @abstract
     * @param {?} cert
     * @return {?}
     */
    LibersignUtil.prototype.selectCertificate = function (cert) { };
    /**
     * @abstract
     * @param {?} dataToSign
     * @return {?}
     */
    LibersignUtil.prototype.doSign = function (dataToSign) { };
    /**
     * @abstract
     * @return {?}
     */
    LibersignUtil.prototype.destroy = function () { };
}
