// @libriciel/ls-composants/esm2015/lib/component/libersign/libersign.applet.util.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.applet.util.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { LibersignUtil } from './libersign.util';
export class LibersignAppletUtil extends LibersignUtil {
    /**
     * @param {?} zone
     */
    constructor(zone) {
        super();
        this.zone = zone;
    }
    /**
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    init(updateUrl, appletUrl) {
        this.status.appletLoading = true;
        this.appletElement = document.getElementById('appletLibersign');
        if (this.appletElement == null) {
            /** @type {?} */
            const appletElement = document.createElement('div');
            appletElement.id = 'appletLibersign';
            document.body.appendChild(appletElement);
            this.appletElement = document.getElementById('appletLibersign');
            // @ts-ignore
            window.appletIsLoaded = (/**
             * @return {?}
             */
            () => {
                this.zone.run((/**
                 * @return {?}
                 */
                () => {
                    this.status.appletLoading = false;
                    this.status.certsLoading = true;
                    this.loadCertificates();
                }));
                this.selectCertificate(this.favorite);
            });
            this.appletElement.innerHTML = `
<applet codebase="${appletUrl}" id="liberSignApplet" code="org/adullact/parapheur/applets/splittedsign/Main.class" archive="SplittedSignatureApplet.jar" name="Signature i-Parapheur" height="0" width="100%">
  <param name="permissions" value="all-permissions">
  <param name="codebase_lookup" value="true">
  <param name="display_cancel" value="true">
  <param name="cancel_mode" value="javascript">
</applet>`;
        }
        else {
            this.zone.run((/**
             * @return {?}
             */
            () => {
                this.status.appletLoading = false;
                this.status.certsLoading = true;
                this.loadCertificates();
            }));
            this.selectCertificate(this.favorite);
        }
    }
    /**
     * @return {?}
     */
    loadCertificates() {
        /** @type {?} */
        const appletElement = document.getElementById('liberSignApplet');
        if (appletElement != null) {
            // @ts-ignore
            /** @type {?} */
            const certStr = document.getElementById('liberSignApplet').getCertificates();
            this.certificates = JSON.parse(certStr)['certs'];
            this.commonLoadCert();
        }
    }
    /**
     * @param {?} cert
     * @return {?}
     */
    selectCertificate(cert) {
        this.zone.run((/**
         * @return {?}
         */
        () => {
            this.currentCert = cert;
        }));
    }
    /**
     * @param {?} dataToSign
     * @return {?}
     */
    doSign(dataToSign) {
        // @ts-ignore
        /** @type {?} */
        const signatureValue = document.getElementById('liberSignApplet').sign(this.currentCert.ID, dataToSign);
        this.zone.run((/**
         * @return {?}
         */
        () => {
            this.signatures.next(JSON.parse(signatureValue));
        }));
    }
    /**
     * @return {?}
     */
    destroy() {
        // @ts-ignore
    }
}
if (false) {
    /** @type {?} */
    LibersignAppletUtil.prototype.appletElement;
    /**
     * @type {?}
     * @private
     */
    LibersignAppletUtil.prototype.zone;
}
