// @libriciel/ls-composants/esm2015/lib/pages/login/component/login-config/login-config.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/component/login-config/login-config.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, HostListener, Input, Output, ViewChild } from '@angular/core';
import { defaultConfiguration } from './login-config.model';
import { DomSanitizer } from '@angular/platform-browser';
import { Weight } from '../../../../ui-concepts/weight';
import { CommonIcons } from '../../../../icons/common-icons';
import { CommonMessages } from '../../../../i18n/common-messages';
export class LoginConfigComponent {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        this._DomSanitizationService = _DomSanitizationService;
        this.save = new EventEmitter();
        this.back = new EventEmitter();
        this.previewChanges = false;
        this.saving = false;
        this.isSaved = false;
        this.weight = Weight;
        this.commonIcons = CommonIcons;
        this.commonMessages = CommonMessages;
        this._asyncSave = false;
        this._configuration = Object.assign({}, defaultConfiguration);
    }
    /**
     * @return {?}
     */
    onSaved() {
        this.isSaved = true;
        this.saving = false;
        clearTimeout(this._timeoutShow);
        this._timeoutShow = setTimeout((/**
         * @return {?}
         */
        () => {
            this.isSaved = false;
        }), 3000);
    }
    /**
     * @param {?} async
     * @return {?}
     */
    set asyncSave(async) {
        this._asyncSave = typeof async === 'string' ? async === 'true' : async;
    }
    /**
     * @param {?} config
     * @return {?}
     */
    set configuration(config) {
        this._configuration = typeof config === 'string' ? JSON.parse(config) : config;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    stopPropagation(event) {
        event.stopPropagation();
    }
    /**
     * @return {?}
     */
    onEscapeKeyPressed() {
        this.previewChanges = false;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    onFileChanged(event) {
        /** @type {?} */
        const file = event.target.files[0];
        /** @type {?} */
        const reader = new FileReader();
        reader.addEventListener('load', (/**
         * @return {?}
         */
        () => {
            this._configuration.imgName = file.name;
            // @ts-ignore
            this._configuration.imgData = reader.result;
        }), false);
        if (file) {
            reader.readAsDataURL(file);
        }
    }
    /**
     * @return {?}
     */
    removeCurrentFile() {
        this._configuration.imgData = undefined;
        this.fileInput.nativeElement.value = '';
    }
    /**
     * @return {?}
     */
    onReset() {
        this.removeCurrentFile();
        Object.assign(this._configuration, defaultConfiguration);
    }
    /**
     * @return {?}
     */
    onSave() {
        this.save.emit(this._configuration);
        if (this._asyncSave) {
            this.saving = true;
        }
        else {
            this.onSaved();
        }
    }
    /**
     * @return {?}
     */
    onBack() {
        this.back.emit();
    }
}
LoginConfigComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-login-config',
                template: "<h3>Personnalisation de la page de connexion</h3>\n\n<span class=\"alert alert-info\">\n  <i class=\"fa fa-info-circle\"></i>\n  Rappel : les champs renseign\u00E9s ci-dessous sont visibles de tous les utilisateurs\n</span>\n\n<div class=\"action-buttons\">\n  <div class=\"left-buttons\">\n\n    <ls-button-go-back (click)=\"onBack()\"></ls-button-go-back>\n    <ls-button (click)=\"previewChanges = true\" [icon]=\"commonIcons.PREVIEW_ICON\" [title]=\"commonMessages.PREVIEW\"\n               [weight]=\"weight.SECONDARY\">\n    </ls-button>\n    <ls-button (click)=\"onReset()\" [icon]=\"commonIcons.RE_INIT_ICON\" [title]=\"commonMessages.RE_INIT\"\n               [weight]=\"weight.SECONDARY\">\n    </ls-button>\n    <ls-button-save (click)=\"onSave()\" [processing]=\"saving\"></ls-button-save>\n\n    <span *ngIf=\"isSaved\" class=\"text text-success\"><i class=\"fa fa-check\"></i> Configuration enregistr\u00E9e</span>\n  </div>\n</div>\n\n<div class=\"content-div\">\n  <div>\n    <div>\n      <h3>Collectivit\u00E9 / structure</h3>\n      <p>Activer l'espace de personnalisation r\u00E9serv\u00E9 \u00E0 la collectivit\u00E9 / structure ?</p>\n      <div>\n        <input [(ngModel)]=\"_configuration.showCustom\" [value]=\"true\" checked class=\"btn-radio\" id=\"show-custom-yes\"\n               name=\"show-custom\"\n               type=\"radio\">\n        <label for=\"show-custom-yes\">\n          Activer\n        </label>\n        <input [(ngModel)]=\"_configuration.showCustom\" [value]=\"false\" class=\"btn-radio\" id=\"show-custom-no\"\n               name=\"show-custom\"\n               type=\"radio\">\n        <label class=\"margin-left-16\" for=\"show-custom-no\">\n          D\u00E9sactiver\n        </label>\n      </div>\n    </div>\n    <div [class.disabled]=\"!_configuration.showCustom\">\n      <h4>Logo de la collectivit\u00E9</h4>\n      <input\n        #fileInput\n        (change)=\"onFileChanged($event)\" style=\"display: none\"\n        type=\"file\">\n      <button (click)=\"fileInput.click()\" *ngIf=\"!_configuration.imgData\" class=\"btn btn-primary\">\n        <i class=\"fa fa-plus-circle\"></i>\n        Ajouter une image\n      </button>\n      <div *ngIf=\"_configuration.imgData\" class=\"image-informations\">\n        <div>\n          <i class=\"fa fa-image\"></i>\n          {{_configuration.imgName}}\n        </div>\n        <div>\n          <i (click)=\"removeCurrentFile()\" title=\"Supprimer le logo\" class=\"fa fa-trash fa-trash-alt\"></i>\n          <i (click)=\"fileInput.click()\" title=\"Modifier le logo\" class=\"fa fa-refresh fa-sync\"></i>\n        </div>\n      </div>\n    </div>\n    <div [class.disabled]=\"!_configuration.showCustom\">\n      <h4>\n        Couleurs\n      </h4>\n      <div class=\"d-flex\">\n        <div class=\"color-picker color-background\">\n          <label for=\"background-color\">Couleur de fond</label>\n          <ls-color-picker id=\"background-color\" [(ngModel)]=\"_configuration.colorBackground\"></ls-color-picker>\n        </div>\n        <div class=\"color-picker color-text\">\n          <label for=\"text-color\">Couleur du sous-titre</label>\n          <ls-color-picker id=\"text-color\" [(ngModel)]=\"_configuration.colorText\"></ls-color-picker>\n        </div>\n      </div>\n    </div>\n\n    <div [class.disabled]=\"!_configuration.showCustom\">\n      <label class=\"h4\" for=\"soustitre\">\n        Sous-titre\n      </label>\n      <ls-max-length-textarea [(ngModel)]=\"_configuration.subText\" id=\"soustitre\" maxlength=\"300\"\n                              name=\"soustitre\"></ls-max-length-textarea>\n    </div>\n  </div>\n  <div>\n    <div>\n      <h3>Message d'informations</h3>\n      <p>Afficher un message d'information ?</p>\n      <div>\n        <input [(ngModel)]=\"_configuration.showInformation\" [value]=\"true\" checked class=\"btn-radio\" id=\"show-info-yes\"\n               name=\"show-info\"\n               type=\"radio\">\n        <label for=\"show-info-yes\">\n          Afficher\n        </label>\n        <input [(ngModel)]=\"_configuration.showInformation\" [value]=\"false\" class=\"btn-radio\" id=\"show-info-no\"\n               name=\"show-info\"\n               type=\"radio\">\n        <label class=\"margin-left-16\" for=\"show-info-no\">\n          Ne pas afficher\n        </label>\n      </div>\n\n    </div>\n\n    <div [class.disabled]=\"!_configuration.showInformation\">\n      <label for=\"message-text\">Texte du message :</label>\n      <ls-max-length-textarea [(ngModel)]=\"_configuration.informationText\" id=\"message-text\" maxlength=\"300\"\n                              name=\"message-text\"></ls-max-length-textarea>\n    </div>\n\n    <div [class.disabled]=\"!_configuration.showInformation\">\n      <h4>Niveau d'alerte du message</h4>\n      <div class=\"\">\n        <input [(ngModel)]=\"_configuration.informationType\" id=\"exampleRadios1\" name=\"exampleRadios\" type=\"radio\"\n               value=\"info\">\n        <label class=\" alert alert-info margin-left-16\" for=\"exampleRadios1\">\n          <i class=\"fa fa-info-circle\">\n          </i>\n          Information\n        </label>\n      </div>\n      <div class=\"\">\n        <input [(ngModel)]=\"_configuration.informationType\" id=\"exampleRadios2\" name=\"exampleRadios\" type=\"radio\"\n               value=\"warning\">\n        <label class=\"alert alert-warning margin-left-16\" for=\"exampleRadios2\">\n          <i class=\"fa fa-exclamation-circle\"></i>\n          Alerte\n        </label>\n      </div>\n      <div class=\"\">\n        <input [(ngModel)]=\"_configuration.informationType \" id=\"exampleRadios3\" name=\"exampleRadios\" type=\"radio\"\n               value=\"danger\">\n        <label class=\"alert alert-danger margin-left-16\" for=\"exampleRadios3\">\n          <i class=\"fa fa-exclamation-triangle\"></i>\n          Danger\n        </label>\n      </div>\n    </div>\n  </div>\n</div>\n\n\n<div [class.show]=\"previewChanges\" class=\"preview-modal\">\n  <div (click)=\"previewChanges = false\" class=\"preview-background\">\n    <div (click)=\"stopPropagation($event)\" class=\"preview-body\">\n      <h3>Pr\u00E9visualisation de la page de connexion</h3>\n\n      <div class=\"login-previsu\">\n        <div [style.background-color]=\"_configuration.colorBackground\" class=\"login-previsu-sidebar\"\n             [style.opacity]=\"_configuration.showCustom ? 1 : 0\">\n          <img *ngIf=\"_configuration.imgData\"\n               [src]=\"_DomSanitizationService.bypassSecurityTrustUrl(_configuration.imgData)\" alt=\"Image preview...\"\n               src=\"\">\n          <div [style.color]=\"_configuration.colorText\">{{_configuration.subText}}</div>\n        </div>\n        <div class=\"login-previsu-content\">\n          <span *ngIf=\"_configuration.showInformation && _configuration.informationText\"\n                [ngSwitch]=\"_configuration.informationType\"\n                [ngClass]=\"'alert-' + _configuration.informationType\" class=\"alert\">\n            <i *ngSwitchCase=\"'info'\" class=\"fa fa-info-circle\"></i>\n            <i *ngSwitchCase=\"'warning'\" class=\"fa fa-exclamation-circle\"></i>\n            <i *ngSwitchCase=\"'danger'\" class=\"fa fa-exclamation-triangle\"></i>\n            {{_configuration.informationText}}\n          </span>\n        </div>\n        <div class=\"login-previsu-footer\">\n\n        </div>\n        <div class=\"login-previsu-footer\">\n\n        </div>\n      </div>\n      <i (click)=\"previewChanges = false\" class=\"fa fa-times close-icon\"></i>\n    </div>\n  </div>\n\n</div>\n",
                styles: [".alert{display:inline-block;margin-bottom:.6rem;margin-top:.6rem}.action-buttons{-ms-flex-pack:justify;-webkit-box-pack:justify;display:-webkit-box;display:-ms-flexbox;display:flex;justify-content:space-between}.action-buttons>:first-child *{margin-right:15px}textarea{min-height:110px}@media screen and (max-width:1280px){.content-div{-ms-flex-direction:column;-webkit-box-direction:normal;-webkit-box-orient:vertical;flex-direction:column}.content-div>*{margin:30px 40px}}@media screen and (min-width:1280px){.content-div>:first-child{border-right:1px solid rgba(0,0,0,.15);margin:30px 40px;padding-right:30px}.content-div>*{-ms-flex-direction:column;-webkit-box-direction:normal;-webkit-box-orient:vertical;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;margin:30px 40px 30px 0;width:50%}.content-div>*>:nth-child(n+2){margin-top:20px}}.content-div{-ms-flex-pack:start;-webkit-box-pack:start;-webkit-box-shadow:0 0 6px rgba(0,0,0,.15);background-color:#fff;box-shadow:0 0 6px rgba(0,0,0,.15);display:-webkit-box;display:-ms-flexbox;display:flex;font-family:Ubuntu,sans-serif;justify-content:flex-start;margin-top:20px}.content-div .h4,.content-div h4{font-size:15px;font-weight:700}.content-div .color-picker{-ms-flex-direction:column;-webkit-box-direction:normal;-webkit-box-orient:vertical;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column}.content-div .color-picker:first-child{margin-right:30px}.content-div .color-picker span{font-size:15px}.content-div textarea{margin-right:30px}.content-div .margin-left-16{margin-left:16px}.content-div .disabled{opacity:.25;pointer-events:none}.content-div>div>div:first-child{margin-bottom:20px}.preview-modal:not(.show){display:none}.preview-modal.show{height:100vh;left:0;position:fixed;top:0;width:100vw;z-index:10000}.preview-modal.show .preview-background{-ms-flex-align:center;-ms-flex-pack:center;-webkit-box-align:center;-webkit-box-pack:center;align-items:center;background-color:rgba(0,0,0,.4);display:-webkit-box;display:-ms-flexbox;display:flex;height:100%;justify-content:center;width:100%}.preview-modal.show .preview-body{-ms-grid-columns:1fr;-ms-grid-rows:50px 1fr;background-color:#fff;display:-ms-grid;display:grid;grid-template:\"title\" 50px \"previsu\" 1fr/1fr;height:70vh;padding:30px;position:relative;width:70vw}.preview-modal.show .preview-body>h3{-ms-grid-column:1;-ms-grid-row:1;grid-area:title}.preview-modal.show .close-icon{cursor:pointer;font-size:18px;position:absolute;right:25px;top:25px}.login-previsu{-ms-grid-column:1;-ms-grid-columns:35% 1fr;-ms-grid-row:2;-ms-grid-rows:1fr 20px;-webkit-box-shadow:0 0 6px rgba(0,0,0,.15);box-shadow:0 0 6px rgba(0,0,0,.15);display:-ms-grid;display:grid;grid-area:previsu;grid-template:\"sidebar  content\" 1fr \"footer   footer\" 20px/35% 1fr;height:95%;min-height:95%}.login-previsu .login-previsu-content{-ms-grid-column:2;-ms-grid-row:1;grid-area:content}.login-previsu .login-previsu-sidebar{-ms-flex-align:center;-ms-flex-direction:column;-ms-flex-pack:center;-ms-grid-column:1;-ms-grid-row:1;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:vertical;-webkit-box-pack:center;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;font-size:10px;grid-area:sidebar;justify-content:center;padding:30px;text-align:center}.login-previsu .login-previsu-sidebar div{padding-top:1rem}.login-previsu .login-previsu-footer{-ms-grid-column:1;-ms-grid-column-span:2;-ms-grid-row:2;background-color:#222;grid-area:footer}.login-previsu .alert{left:0;margin-left:auto;margin-right:auto;position:fixed;right:0;text-align:center;width:45vw}.login-previsu img{display:block;max-height:20vh;max-width:90%;width:auto}@media screen and (min-width:0 \\0){.login-previsu img{-ms-flex:1;-webkit-box-flex:1;flex:1;height:auto}.login-previsu div{max-width:100%}}.image-informations{display:-webkit-box;display:-ms-flexbox;display:flex}.image-informations div:first-child{background-color:#f7f7f7;max-width:50%;padding:19px;word-break:break-all}.image-informations div:nth-child(2){-ms-flex-direction:column;-webkit-box-direction:normal;-webkit-box-orient:vertical;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;padding-left:25px}.image-informations i{-webkit-transition:.1s;cursor:pointer;margin:auto;transition:.1s}.image-informations .fa-trash-alt:hover{color:var(--danger)}.image-informations .fa-sync:hover{color:var(--primary)}.color-picker>span{margin-bottom:5px}"]
            }] }
];
/** @nocollapse */
LoginConfigComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
LoginConfigComponent.propDecorators = {
    save: [{ type: Output }],
    back: [{ type: Output }],
    fileInput: [{ type: ViewChild, args: ['fileInput',] }],
    onSaved: [{ type: HostListener, args: ['saved',] }],
    asyncSave: [{ type: Input }],
    configuration: [{ type: Input }],
    onEscapeKeyPressed: [{ type: HostListener, args: ['document:keydown.escape',] }]
};
if (false) {
    /** @type {?} */
    LoginConfigComponent.prototype.save;
    /** @type {?} */
    LoginConfigComponent.prototype.back;
    /** @type {?} */
    LoginConfigComponent.prototype.fileInput;
    /** @type {?} */
    LoginConfigComponent.prototype.previewChanges;
    /** @type {?} */
    LoginConfigComponent.prototype.saving;
    /** @type {?} */
    LoginConfigComponent.prototype.isSaved;
    /** @type {?} */
    LoginConfigComponent.prototype.weight;
    /** @type {?} */
    LoginConfigComponent.prototype.commonIcons;
    /** @type {?} */
    LoginConfigComponent.prototype.commonMessages;
    /** @type {?} */
    LoginConfigComponent.prototype._configuration;
    /** @type {?} */
    LoginConfigComponent.prototype._asyncSave;
    /** @type {?} */
    LoginConfigComponent.prototype._timeoutShow;
    /** @type {?} */
    LoginConfigComponent.prototype._DomSanitizationService;
}
