// @libriciel/ls-composants/esm2015/workflows/shared/model/workflows/workflow-instance.js
/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/workflow-instance.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { StepState } from "./step-state";
export class WorkflowInstance {
    // This constructor is only valid while we use stub services,
    // It should be replace with a constructor from "circuitRaw"
    // constructor(circuitModel: Circuit, stepInstances?: StepInstance[]) {
    /**
     * @param {?=} instData
     */
    constructor(instData) {
        this.currentStepIdx = 0;
        this.rejected = false;
        this.isOver = false;
        this.id = instData.id;
        this.name = instData.name;
        this.description = instData.description;
        this.rejected = instData.isRejected;
        this.isOver = instData.isOver;
        if (instData.steps) {
            this.steps = instData.steps;
            this.currentStepIdx = instData.steps.findIndex((/**
             * @param {?} step
             * @return {?}
             */
            step => step.state === StepState.CURRENT));
            if (this.currentStepIdx === -1) {
                this.currentStepIdx = 0;
            }
        }
    }
}
if (false) {
    /** @type {?} */
    WorkflowInstance.prototype.id;
    /** @type {?} */
    WorkflowInstance.prototype.name;
    /** @type {?} */
    WorkflowInstance.prototype.description;
    /** @type {?} */
    WorkflowInstance.prototype.currentStepIdx;
    /** @type {?} */
    WorkflowInstance.prototype.steps;
    /** @type {?} */
    WorkflowInstance.prototype.rejected;
    /** @type {?} */
    WorkflowInstance.prototype.isOver;
}
