// @libriciel/ls-composants/fesm2015/libriciel-ls-composants-workflows.js
import { Component, ContentChild, Input, EventEmitter, ViewChild, ViewChildren, Output, NgModule } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule } from '@angular/forms';
import { CommonModule } from '@angular/common';
import { NgbPopover, NgbModule } from '@ng-bootstrap/ng-bootstrap';
import { BrowserModule } from '@angular/platform-browser';
import { moveItemInArray, DragDropModule } from '@angular/cdk/drag-drop';

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/workflow-model.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class WorkflowModel {
    constructor() {
        this.steps = [];
    }
}
if (false) {
    /** @type {?} */
    WorkflowModel.prototype.id;
    /** @type {?} */
    WorkflowModel.prototype.name;
    /** @type {?} */
    WorkflowModel.prototype.description;
    /** @type {?} */
    WorkflowModel.prototype.active;
    /** @type {?} */
    WorkflowModel.prototype.steps;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/step-state.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @enum {string} */
const StepState = {
    PENDING: "pending",
    CURRENT: "current",
    VALIDATED: "approve",
    SKIPPED: "skipped",
    REJECTED: "rejected",
};

/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflow-model-viewer/ls-workflow-model-viewer.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class LsWorkflowModelViewerComponent {
    // messages = CircuitDisplayMessages;
    constructor() {
        this.isInstance = false;
        this.isOver = false;
        this.isRejected = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (!this.circuit) {
            console.error('Missing input circuit data');
            this.circuit = new WorkflowModel();
        }
        if (!this.messages) {
            throw new Error('Missing localised messages for LsWorkflowViewerComponent!');
        }
        if (!this.getStepIconClassFct) {
            throw new Error('Missing getStepIconClassFct for LsWorkflowViewerComponent!');
        }
        if (!this.getStepTooltipFct) {
            throw new Error('Missing getStepTooltipFct for LsWorkflowViewerComponent!');
        }
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        console.debug('LsWorkflowViewerComponent::ngOnChanges, ', this);
        if (this.isInstance) {
            console.debug('LsWorkflowViewerComponent::ngOnChanges - circuit is an instance ');
            /** @type {?} */
            const workflow = (/** @type {?} */ (this.circuit));
            this.activeStep = workflow.currentStepIdx;
            this.isOver = workflow.isOver;
            this.isRejected = workflow.rejected;
        }
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepIconClass(step) {
        return this.getStepIconClassFct(step);
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepTooltip(step) {
        return this.getStepTooltipFct(step);
    }
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    isActiveStep(stepIdx) {
        return !this.isRejected && !this.isOver && stepIdx === this.activeStep;
    }
    /**
     * @param {?} step
     * @return {?}
     */
    isPassed(step) {
        return this.isInstance && ((step.state === StepState.VALIDATED) || (step.state === StepState.REJECTED));
    }
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    isActualValidator(step, validator) {
        return this.isInstance && (step.validatedBy.some((/**
         * @param {?} elem
         * @return {?}
         */
        (elem) => elem.id === validator.id)));
    }
    /**
     * @return {?}
     */
    isCircuitOver() {
        return this.isOver && !this.isRejected;
    }
    /**
     * @return {?}
     */
    getCircuitEndAdditionalClasses() {
        if (this.getCircuitEndAdditionalClassesFct) {
            return this.getCircuitEndAdditionalClassesFct();
        }
        return '';
    }
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    getValidatorClasses(step, validator) {
        if (this.isActualValidator(step, validator)) {
            if (step.state === StepState.REJECTED) {
                return this.messages.REJECTED_BY_ICON_CLASS;
            }
            else {
                return this.messages.VALIDATED_BY_ICON_CLASS;
            }
        }
        return '';
    }
}
LsWorkflowModelViewerComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-workflow-model-viewer',
                template: "<div class=\"ls-workflow-display\">\n  <div class=\"ls-workflow-canevas\">\n    <div class=\"circuit-start\">\n      <div class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa fa-2x\" [ngClass]=\"messages.CIRCUIT_START_ICON\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.FIRST_STEP_TXT\"></span>\n      </div>\n    </div>\n    <div class=\"steps-container\">\n      <div *ngFor=\"let step of circuit.steps; index as i\" >\n        <div class=\"ls-workflow-display-step-container\">\n          <div  class=\"ls-workflow-step\" [ngClass]=\"isActiveStep(i) ? 'wa-active-step wa-white-icn' : ''\">\n            <div class=\"ls-step-top-line\">\n              <div class=\"step-number\">{{i + 1}}.</div>\n              <div class=\"step-name\">{{step.name}}</div>\n              <div [ngClass]=\"getStepIconClass(step)\" class=\"step-icn fa\" ngbTooltip=\"{{getStepTooltip(step)}}\"></div>\n            </div>\n            <div class=\"ls-display-step-separator\"></div>\n            <div class=\"ls-step-validator-list\" [ngClass]=\"isInstance ? 'wa-circuit-instance' : ''\">\n              <ng-container *ngIf=\"isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"passedStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n              <ng-container *ngIf=\" ! isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"pendingStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n            </div>\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"ls-workflow-end\">\n      <div class=\"ls-workflow-endpoint-content\" [ngClass]=\"getCircuitEndAdditionalClasses()\">\n        <span class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"messages.CIRCUIT_END_ICON\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.LAST_STEP_TXT\"></span>\n      </div>\n    </div>\n  </div>\n</div>\n",
                styles: ["@charset \"UTF-8\";body{background-color:var(--ls-grey-50)}.active-btn{color:#42a1e8}.ls-workflow-canevas{-ms-flex-align:center;-ms-flex-direction:column;-ms-flex-pack:start;-ms-grid-column:1;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:vertical;-webkit-box-pack:start;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;grid-column:1;height:100%;justify-content:flex-start}.ls-workflow-step{-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);background-color:#fff;border:none;box-shadow:0 0 .25rem rgba(0,0,0,.15);min-height:6rem;width:var(--ls-workflow-step-width,20rem)}.ls-workflow-step .ls-step-top-line{-ms-flex-align:center;-ms-flex-direction:row;-ms-flex-pack:justify;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:horizontal;-webkit-box-pack:justify;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;font-weight:700;justify-content:space-between;padding:.5rem}.ls-workflow-step .ls-display-step-separator{border-bottom:1px solid #d3d3d3;width:100%}.ls-workflow-step .ls-step-validator-list{color:#5b5b5b;padding:.5rem .5rem .5rem 1.5rem}.ls-workflow-step.selected{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.ls-workflow-step .step-icn{height:1.7rem;width:1.7rem}.ls-workflow-endpoint-content{margin:auto;text-align:center}.ls-workflow-endpoint-content .ls-step-bounds-txt{color:var(--primary);font-weight:700}.ls-workflow-endpoint-content .ls-workflow-endpoint-icn{color:var(--primary)}.ls-workflow-endpoint-content.not-active .ls-step-bounds-txt,.ls-workflow-endpoint-content.not-active .ls-workflow-endpoint-icn{color:#5b5b5b}.ls-workflow-endpoint-content .ls-step-validator-list{text-align:left}.ls-actual-validator{font-weight:700}.ls-actual-validator:before{color:var(--primary);content:\"\uF00C\";font-family:FontAwesome;font-size:1em;left:.5em;position:absolute}.ls-rejector{font-weight:700}.ls-rejector:before{color:var(--danger);content:\"\uF00D\";font-family:FontAwesome;font-size:1em;left:.5em;position:absolute}.wa-comment-icon{color:var(--primary);font-size:1em;margin-right:.5em}.ls-workflow-display .ls-workflow-step{margin:1rem auto;padding-left:0;padding-right:0}.ls-workflow-display .ls-workflow-step .ls-step-validator-list{padding:1rem 1rem 1rem .5rem}.ls-workflow-display .ls-workflow-step .ls-step-validator-list.ls-workflow-instance{padding-left:2rem}.ls-workflow-display .ls-workflow-endpoint-content{-ms-flex-direction:row;-webkit-box-direction:normal;-webkit-box-orient:horizontal;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row}.ls-workflow-display .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:.5rem}.ls-workflow-display .ls-step-top-line{-ms-flex-pack:start;-webkit-box-pack:start;justify-content:flex-start;padding-left:.5rem}.ls-workflow-display .ls-step-top-line .step-name{-ms-flex:1;-webkit-box-flex:1;flex:1;margin-left:.7rem}.ls-workflow-display .ls-display-step-separator{border-bottom:1px solid #d3d3d3;width:100%}.ls-workflow-display .ls-workflow-display-step-container{position:relative}.ls-workflow-display .ls-active-step .ls-step-top-line{background-color:var(--primary);color:#fff;margin:0}.ls-workflow-display .ls-active-step .ls-step-top-line .step-icn{color:#fff}.ls-workflow-display .ls-active-step:before{color:var(--primary);content:\"\uF061\";font-family:FontAwesome;font-size:2em;left:-1em;position:absolute}.ls-workflow-display .ls-active-step.end-step:before{font-size:1.2em}.ls-workflow-end{position:relative}"]
            }] }
];
/** @nocollapse */
LsWorkflowModelViewerComponent.ctorParameters = () => [];
LsWorkflowModelViewerComponent.propDecorators = {
    passedStepTemplateRef: [{ type: ContentChild, args: ['passedStepContent',] }],
    pendingStepTemplateRef: [{ type: ContentChild, args: ['pendingStepContent',] }],
    circuit: [{ type: Input }],
    activeStep: [{ type: Input }],
    messages: [{ type: Input }],
    getStepIconClassFct: [{ type: Input }],
    getStepTooltipFct: [{ type: Input }],
    getCircuitEndAdditionalClassesFct: [{ type: Input }],
    isInstance: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.passedStepTemplateRef;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.pendingStepTemplateRef;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.circuit;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.activeStep;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.messages;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.getStepIconClassFct;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.getStepTooltipFct;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.getCircuitEndAdditionalClassesFct;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.isInstance;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.isOver;
    /** @type {?} */
    LsWorkflowModelViewerComponent.prototype.isRejected;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/step-model.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class StepModel {
    constructor() {
        this.validators = [];
    }
    /**
     * @return {?}
     */
    clone() {
        console.log('StepModel::clone');
        /** @type {?} */
        const clone = new StepModel();
        clone.name = this.name;
        clone.type = this.type;
        clone.validators = [...this.validators];
        return clone;
    }
}
if (false) {
    /** @type {?} */
    StepModel.prototype.name;
    /** @type {?} */
    StepModel.prototype.type;
    /** @type {?} */
    StepModel.prototype.validators;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/workflow-model-options.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class WorkflowModelOptions {
}
if (false) {
    /** @type {?} */
    WorkflowModelOptions.prototype.startIconClass;
    /** @type {?} */
    WorkflowModelOptions.prototype.endIconClass;
    /** @type {?} */
    WorkflowModelOptions.prototype.editorGlobalLayoutClass;
    /** @type {?} */
    WorkflowModelOptions.prototype.additionalFinalEndpointClasses;
}
class WorkflowModelLocalizedMessages {
}
if (false) {
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.CREATE_STEP_TITLE;
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.EDIT_STEP_TITLE;
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.ADD_STEP_BTN_TXT;
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.FIRST_STEP_TXT;
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.LAST_STEP_TXT;
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.popCreateTitle;
    /** @type {?} */
    WorkflowModelLocalizedMessages.prototype.popEditTitle;
}

/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflow-editor/ls-workflow-editor.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class LsWorkflowEditorComponent {
    constructor() {
        this.messages = {
            CREATE_STEP_TITLE: 'Ajouter une étape',
            EDIT_STEP_TITLE: 'Modification de l\'étape',
            ADD_STEP_BTN_TXT: 'Ajouter une étape',
            FIRST_STEP_TXT: 'Début',
            LAST_STEP_TXT: 'Fin',
        };
        this.activeOptions = {
            startIconClass: 'far fa-flag',
            endIconClass: 'fas fa-flag-checkered',
            editorGlobalLayoutClass: 'ls-workflow-editor-default-layout',
            additionalFinalEndpointClasses: ''
        };
        this.endpointIsEditable = false;
        this.workflowChange = new EventEmitter();
    }
    /**
     * @return {?}
     */
    get localizedMessages() {
        return this.messages;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set localizedMessages(value) {
        this.messages = Object.assign(this.messages, value);
    }
    /**
     * @return {?}
     */
    get options() {
        return this.activeOptions;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set options(value) {
        this.activeOptions = Object.assign(this.activeOptions, value);
    }
    /**
     * @return {?}
     */
    get workflow() {
        return this.workflowData;
    }
    /**
     * @param {?} newValue
     * @return {?}
     */
    set workflow(newValue) {
        this.workflowData = newValue;
        this.workflowChange.emit(newValue);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (!this.workflowData) {
            console.error('Missing input workflow data');
            this.workflowData = new WorkflowModel();
        }
        if (!this.messages) {
            throw new Error('Missing localised messages for LsWorkflowViewerComponent!');
        }
    }
    /**
     * @return {?}
     */
    workflowDataChanged() {
        this.workflowChange.emit(this.workflowData);
    }
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    askForStepDeletion(stepIdx) {
        console.debug('delete step');
        if (this.confirmStepDeletionFct) {
            this.confirmStepDeletionFct(this.workflowData.steps[stepIdx]).then((/**
             * @param {?} result
             * @return {?}
             */
            result => {
                if (result) {
                    this.deleteStep(stepIdx);
                }
            }));
        }
        else {
            this.deleteStep(stepIdx);
        }
    }
    /**
     * @param {?} stepData
     * @param {?} idx
     * @return {?}
     */
    confirmStepCreation(stepData, idx) {
        this.addStep(stepData, idx);
        this.closeCurrentPopover();
    }
    /**
     * @param {?} stepData
     * @param {?} idx
     * @return {?}
     */
    confirmStepEdition(stepData, idx) {
        this.replaceStep(stepData, idx);
        this.closeCurrentPopover();
    }
    /**
     * @return {?}
     */
    confirmFinalStepEdition() {
        this.closeCurrentPopover();
    }
    /**
     * @return {?}
     */
    cancelStepEdition() {
        this.closeCurrentPopover();
    }
    /**
     * @param {?} idx
     * @return {?}
     */
    cancelStepCreation(idx) {
        this.closeCurrentPopover();
    }
    /**
     * @return {?}
     */
    isFirstStep() {
        return (!this.workflowData || this.workflowData.steps.length === 0);
    }
    /**
     * @param {?} idx
     * @return {?}
     */
    getAdditionalStepClass(idx) {
        if (this.currentEditedStepIdx !== null && this.currentEditedStepIdx === idx) {
            return 'selected';
        }
        return '';
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepIconClass(step) {
        if (this.getStepIconClassFct) {
            return this.getStepIconClassFct(step);
        }
        return '';
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepTooltip(step) {
        if (this.getStepTooltipFct) {
            return this.getStepTooltipFct(step);
        }
        return '';
    }
    /**
     * @param {?} src
     * @return {?}
     */
    cloneStep(src) {
        return src.clone();
    }
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    createStep(stepIdx) {
        /** @type {?} */
        const newStep = new StepModel();
        this.currentPopover = this.getCreationPopoverForIdx(stepIdx);
        this.currentPopover.hidden.subscribe((/**
         * @return {?}
         */
        () => this.cancelStepCreation(stepIdx)));
        this.currentPopover.open({
            stepData: newStep,
            idx: stepIdx,
            cancelStepCreation: this.cancelStepCreation.bind(this),
            confirmStepCreation: this.confirmStepCreation.bind(this)
        });
    }
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    editStep(stepIdx) {
        this.currentEditedStepIdx = stepIdx;
        /** @type {?} */
        const editingStep = this.cloneStep(this.workflowData.steps[stepIdx]);
        this.currentPopover = this.getEditionPopoverForIdx(stepIdx);
        this.currentPopover.hidden.subscribe((/**
         * @return {?}
         */
        () => this.cancelStepEdition()));
        this.currentPopover.open({
            stepData: editingStep,
            idx: stepIdx,
            cancelStepEdition: this.cancelStepEdition.bind(this),
            confirmStepEdition: this.confirmStepEdition.bind(this)
        });
    }
    /**
     * @return {?}
     */
    editFinalStep() {
        /** @type {?} */
        const editingStep = this.cloneStep(this.lastStepData);
        this.currentPopover = this.endpointPopoverElementRef;
        this.currentPopover.hidden.subscribe((/**
         * @return {?}
         */
        () => this.cancelStepEdition()));
        this.currentPopover.open({
            stepData: editingStep,
            cancelStepEdition: this.cancelStepEdition.bind(this),
            confirmStepEdition: this.confirmFinalStepEdition.bind(this)
        });
    }
    /**
     * @param {?} event
     * @return {?}
     */
    drop(event) {
        moveItemInArray(this.workflowData.steps, event.previousIndex, event.currentIndex);
        this.workflowDataChanged();
    }
    /**
     * @param {?} idx
     * @return {?}
     */
    dragDisabledForIdx(idx) {
        return (this.currentEditedStepIdx !== null && this.currentEditedStepIdx === idx);
    }
    /**
     * @private
     * @return {?}
     */
    closeCurrentPopover() {
        if (this.currentPopover && this.currentPopover.isOpen()) {
            this.currentPopover.close();
        }
        this.currentPopover = null;
        this.currentEditedStepIdx = null;
    }
    /**
     * @private
     * @param {?} step
     * @param {?} pos
     * @return {?}
     */
    addStep(step, pos) {
        this.workflowData.steps.splice(pos, 0, step);
        this.workflowDataChanged();
    }
    /**
     * @private
     * @param {?} newStep
     * @param {?} stepIdx
     * @return {?}
     */
    replaceStep(newStep, stepIdx) {
        this.workflowData.steps.splice(stepIdx, 1, newStep);
        this.workflowDataChanged();
    }
    /**
     * @private
     * @param {?} stepIdx
     * @return {?}
     */
    deleteStep(stepIdx) {
        this.workflowData.steps.splice(stepIdx, 1);
        this.workflowDataChanged();
    }
    /**
     * @private
     * @param {?} idx
     * @return {?}
     */
    getCreationPopoverForIdx(idx) {
        /** @type {?} */
        const filteredList = this.allCircuitEditionPopover.filter((/**
         * @param {?} item
         * @return {?}
         */
        (item) => {
            /** @type {?} */
            const elem = ((/** @type {?} */ (item)))._elementRef;
            if (elem && elem.nativeElement) {
                return elem.nativeElement.hasAttribute('create-btn');
            }
            return true;
        }));
        return filteredList[idx];
    }
    /**
     * @private
     * @param {?} idx
     * @return {?}
     */
    getEditionPopoverForIdx(idx) {
        /** @type {?} */
        const filteredList = this.allCircuitEditionPopover.filter((/**
         * @param {?} item
         * @return {?}
         */
        (item) => {
            /** @type {?} */
            const elem = ((/** @type {?} */ (item)))._elementRef;
            if (elem && elem.nativeElement) {
                return !(elem.nativeElement.hasAttribute('create-btn'));
            }
            return true;
        }));
        return filteredList[idx];
    }
}
LsWorkflowEditorComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-workflow-editor',
                template: "<div class=\"ls-workflow-editor\" [ngClass]=\"options.editorGlobalLayoutClass\">\n  <div class=\"ls-workflow-canevas\">\n    <div class=\"ls-workflow-start ls-workflow-endpoint\">\n      <div class=\"ls-workflow-endpoint-content\">\n        <div class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.startIconClass\"></div>\n        <div class=\"ls-step-bounds-txt\">{{messages.FIRST_STEP_TXT}}</div>\n      </div>\n    </div>\n    <div class=\"ls-workflow-stepList-container\" cdkDropList (cdkDropListDropped)=\"drop($event)\">\n      <div class=\"ls-workflow-link-container\">\n        <div class=\"ls-workflow-link\"></div>\n        <div [ngClass]=\"isFirstStep() ? 'firstStep' : ''\" class=\"ls-link-button-container\">\n          <div (click)=\"createStep(0)\" [autoClose]=\"'outside'\" [ngbPopover]=\"popCreateContent\"\n               [popoverTitle]=\"messages.popCreateTitle\" class=\"add-step-btn\" create-btn\n               placement=\"right\" popoverClass=\"ls-step-edition-popover\"\n               triggers=\"manual\"\n               autoClose=\"false\">\n            <span class=\"fa fa-plus-circle\"></span>\n            <span class=\"add-step-btn-txt\">{{messages.ADD_STEP_BTN_TXT}}</span>\n          </div>\n        </div>\n      </div>\n\n      <div *ngFor=\"let step of workflowData.steps; index as i\">\n        <div class=\"ls-workflow-step-container\" cdkDrag [cdkDragDisabled]=\"dragDisabledForIdx(i)\">\n          <div [ngClass]=\"getAdditionalStepClass(i)\" class=\"ls-workflow-step\">\n            <div class=\"ls-step-top-line\">\n              <div class=\"step-number\">{{i + 1}}.</div>\n              <div class=\"step-name\">{{step.name}}</div>\n              <div [ngClass]=\"getStepIconClass(step)\" class=\"step-icn fa\" ngbTooltip=\"{{getStepTooltip(step)}}\"></div>\n            </div>\n            <div class=\"ls-step-validator-list\">\n              <ng-template\n                [ngTemplateOutlet]=\"stepDisplayContentRef\"\n                [ngTemplateOutletContext]=\"{step: step}\">\n              </ng-template>\n            </div>\n          </div>\n          <div class=\"ls-workflow-step-external-buttons\">\n            <div (click)=\"editStep(i)\" [autoClose]=\"'outside'\" [ngbPopover]=\"popEditContent\"\n                 [popoverTitle]=\"messages.popEditTitle\" class=\"fa fa-pencil-alt fa-pencil edit-step-btn\"\n                 placement=\"right-top\"\n                 popoverClass=\"ls-step-edition-popover\"\n                 triggers=\"manual\"></div>\n            <div (click)=\"askForStepDeletion(i)\" class=\"fa fa-trash delete-step-btn\"></div>\n          </div>\n        </div>\n\n        <div class=\"ls-workflow-link-container\">\n          <div class=\"ls-workflow-link\"></div>\n          <div class=\"ls-link-button-container\">\n            <div (click)=\"createStep(i + 1)\" [autoClose]=\"'outside'\" [ngbPopover]=\"popCreateContent\"\n                 [popoverTitle]=\"messages.popCreateTitle\" class=\"add-step-btn\" create-btn\n                 placement=\"right\" popoverClass=\"ls-step-edition-popover\"\n                 triggers=\"manual\"\n                 autoClose=\"false\">\n              <span class=\"fa fa-plus-circle\"></span>\n              <span class=\"add-step-btn-txt\">{{messages.ADD_STEP_BTN_TXT}}</span>\n            </div>\n          </div>\n        </div>\n      </div>\n\n    </div>\n    <div class=\"ls-workflow-end ls-workflow-endpoint\" [ngClass]=\"options.additionalFinalEndpointClasses\">\n      <ng-container *ngIf=\"endpointIsEditable\">\n        <div\n          #endpointPopoverHandle\n          (click)=\"editFinalStep()\"\n          [autoClose]=\"'outside'\"\n          [ngbPopover]=\"popEditFinalStepContent\"\n          [popoverTitle]=\"messages.popEditTitle\"\n          class=\"ls-workflow-endpoint-overlay\"\n          placement=\"right\"\n          popoverClass=\"ls-step-edition-popover\"\n          triggers=\"manual\">\n\n        </div>\n      </ng-container>\n      <div class=\"ls-workflow-endpoint-content\">\n        <div class=\"ls-workflow-endpoint-icn-group\">\n          <div class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.endIconClass\"></div>\n          <div class=\"ls-step-bounds-txt\">{{messages.LAST_STEP_TXT}}</div>\n        </div>\n        <div *ngIf=\"endpointIsEditable\" class=\"ls-step-validator-list\">\n          <ng-template [ngTemplateOutlet]=\"lastStepDisplayContentRef\">\n          </ng-template>\n        </div>\n      </div>\n\n    </div>\n  </div>\n</div>\n\n<ng-template\n  #popEditContent\n  let-idx=\"idx\"\n  let-stepData=\"stepData\"\n  let-cancelStepEdition=\"cancelStepEdition\"\n  let-confirmStepEdition=\"confirmStepEdition\">\n  <ng-template\n    [ngTemplateOutlet]=\"popEditContentTemplateRef\"\n    [ngTemplateOutletContext]=\"{idx:idx, stepData:stepData, cancelStepEdition:cancelStepEdition, confirmStepEdition:confirmStepEdition}\">\n  </ng-template>\n</ng-template>\n\n<ng-template\n  #popCreateContent\n  let-idx=\"idx\"\n  let-stepData=\"stepData\"\n  let-cancelStepCreation=\"cancelStepCreation\"\n  let-confirmStepCreation=\"confirmStepCreation\">\n  <ng-template\n    [ngTemplateOutlet]=\"popCreateContentTemplateRef\"\n    [ngTemplateOutletContext]=\"{idx:idx, stepData:stepData, cancelStepCreation:cancelStepCreation, confirmStepCreation:confirmStepCreation}\">\n  </ng-template>\n</ng-template>\n\n\n<ng-template\n  #popEditFinalStepContent\n  let-stepData=\"stepData\"\n  let-cancelStepEdition=\"cancelStepEdition\"\n  let-confirmStepEdition=\"confirmStepEdition\">\n  <ng-template\n    [ngTemplateOutlet]=\"endpointPopoverEditContentRef\"\n    [ngTemplateOutletContext]=\"{stepData:stepData, cancelStepEdition:cancelStepEdition, confirmStepEdition:confirmStepEdition, isLastStep:true}\">\n  </ng-template>\n</ng-template>\n",
                styles: ["body{background-color:var(--ls-grey-50)}.active-btn{color:#42a1e8}.ls-workflow-canevas{-ms-flex-align:center;-ms-flex-direction:column;-ms-flex-pack:start;-ms-grid-column:1;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:vertical;-webkit-box-pack:start;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;grid-column:1;height:100%;justify-content:flex-start}.ls-workflow-endpoint.with-content,.ls-workflow-step{-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);background-color:#fff;border:none;box-shadow:0 0 .25rem rgba(0,0,0,.15);min-height:6rem;width:var(--ls-workflow-step-width,20rem)}.ls-workflow-endpoint.with-content .ls-step-top-line,.ls-workflow-step .ls-step-top-line{-ms-flex-align:center;-ms-flex-direction:row;-ms-flex-pack:justify;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:horizontal;-webkit-box-pack:justify;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;font-weight:700;justify-content:space-between;padding:.5rem}.ls-workflow-endpoint.with-content .ls-display-step-separator,.ls-workflow-step .ls-display-step-separator{border-bottom:1px solid #d3d3d3;width:100%}.ls-workflow-endpoint.with-content .ls-step-validator-list,.ls-workflow-step .ls-step-validator-list{color:#5b5b5b;padding:.5rem .5rem .5rem 1.5rem}.ls-workflow-step.selected,.selected.ls-workflow-endpoint.with-content{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.ls-workflow-endpoint.with-content .step-icn,.ls-workflow-step .step-icn{height:1.7rem;width:1.7rem}.ls-workflow-endpoint-content{margin:auto;text-align:center}.ls-workflow-endpoint-content .ls-step-bounds-txt{color:var(--primary);font-weight:700}.ls-workflow-endpoint-content .ls-workflow-endpoint-icn{color:var(--primary)}.ls-workflow-endpoint-content.not-active .ls-step-bounds-txt,.ls-workflow-endpoint-content.not-active .ls-workflow-endpoint-icn{color:#5b5b5b}.ls-workflow-endpoint-content .ls-step-validator-list{text-align:left}.ls-horizontal-separation-line{border-bottom:1px solid #5b5b5b;margin-top:2rem}.ls-workflow-endpoint{-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);background-color:#fff;border:none;border-radius:50%;box-shadow:0 0 .25rem rgba(0,0,0,.15);display:-ms-grid;display:grid;height:6rem;position:relative;width:6rem}.ls-workflow-endpoint.with-content{border-radius:0;display:block;width:var(--ls-workflow-step-width,20rem)}.ls-workflow-endpoint.with-content .ls-workflow-endpoint-content{height:100%;padding:.5em;width:100%}.ls-workflow-endpoint.with-content .ls-workflow-endpoint-content .ls-workflow-endpoint-icn-group{-ms-flex-align:baseline;-ms-flex-direction:row-reverse;-ms-flex-pack:justify;-webkit-box-align:baseline;-webkit-box-direction:reverse;-webkit-box-orient:horizontal;-webkit-box-pack:justify;align-items:baseline;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row-reverse;justify-content:space-between;padding:0 .5em;width:100%}.ls-workflow-endpoint.with-content .ls-workflow-endpoint-content .ls-workflow-endpoint-icn-group:after{content:\"\";display:block;width:2em}.ls-workflow-endpoint .ls-workflow-endpoint-overlay{bottom:0;cursor:pointer;height:100%;left:0;position:absolute;width:100%;z-index:2}.ls-workflow-link{border:1px solid #222;width:0}.cdk-drag-preview .ls-workflow-endpoint.with-content,.cdk-drag-preview .ls-workflow-step{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.cdk-drag-placeholder{min-height:6rem}.cdk-drag-placeholder .ls-workflow-endpoint.with-content,.cdk-drag-placeholder .ls-workflow-step,.cdk-drag-placeholder .ls-workflow-step-external-buttons{display:none}.cdk-drop-list-dragging .cdk-drag{-webkit-transition:-webkit-transform .25s cubic-bezier(0,0,.2,1);transition:-webkit-transform .25s cubic-bezier(0,0,.2,1);transition:transform .25s cubic-bezier(0,0,.2,1);transition:transform .25s cubic-bezier(0,0,.2,1),-webkit-transform .25s cubic-bezier(0,0,.2,1)}.cdk-drag-animating{-webkit-transition:-webkit-transform .3s cubic-bezier(0,0,.2,1);transition:-webkit-transform .3s cubic-bezier(0,0,.2,1);transition:transform .3s cubic-bezier(0,0,.2,1);transition:transform .3s cubic-bezier(0,0,.2,1),-webkit-transform .3s cubic-bezier(0,0,.2,1)}.ls-workflow-link-container{-ms-flex-direction:row;-webkit-box-direction:normal;-webkit-box-orient:horizontal;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;min-height:4rem;min-width:20rem}.ls-workflow-link-container:before{-ms-flex:0 0 calc(50% - 2px);-webkit-box-flex:0;content:\"\";flex:0 0 calc(50% - 2px)}.ls-workflow-link-container .ls-link-button-container{-ms-flex:0 0 calc(50% - 2px);-ms-flex-item-align:center;-ms-grid-row-align:center;-webkit-box-flex:0;align-self:center;flex:0 0 calc(50% - 2px);height:-webkit-min-content;height:-moz-min-content;height:min-content;padding-left:1rem;width:-webkit-min-content;width:-moz-min-content;width:min-content}.ls-workflow-stepList-container{width:100%}.ls-workflow-step-container{-ms-flex-direction:row;-ms-flex-item-align:center;-ms-flex-pack:center;-ms-grid-row-align:center;-webkit-box-direction:normal;-webkit-box-orient:horizontal;-webkit-box-pack:center;align-self:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;justify-content:center;margin-left:auto;margin-right:auto}.ls-workflow-step-container:before{-ms-flex:0 0 5rem;-webkit-box-flex:0;content:\"\";flex:0 0 5rem}.ls-workflow-step-container .ls-workflow-endpoint.with-content,.ls-workflow-step-container .ls-workflow-step{-ms-flex-item-align:center;-ms-grid-row-align:center;align-self:center}.ls-workflow-step-container .ls-workflow-step-external-buttons{-ms-flex:0 0 5rem;-webkit-box-flex:0;flex:0 0 5rem;height:-webkit-min-content;height:-moz-min-content;height:min-content;padding-left:1rem}.ls-workflow-step-external-buttons{-ms-flex-direction:row;-ms-flex-pack:start;-webkit-box-direction:normal;-webkit-box-orient:horizontal;-webkit-box-pack:start;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;justify-content:flex-start}.ls-workflow-step-external-buttons .edit-step-btn{cursor:pointer;margin-right:1rem}.ls-workflow-step-external-buttons .edit-step-btn:hover{color:var(--primary)}.ls-workflow-step-external-buttons .delete-step-btn{cursor:pointer}.ls-workflow-step-external-buttons .delete-step-btn:hover{color:var(--danger)}.ls-workflow-editor{display:-ms-grid;display:grid;height:100%}.ls-workflow-editor .ls-link-button-container .add-step-btn{cursor:pointer;width:-webkit-max-content;width:-moz-max-content;width:max-content}.ls-workflow-editor .ls-link-button-container .add-step-btn .add-step-btn-txt{-webkit-transition:opacity .2s ease-in-out;margin-left:.5em;opacity:0;transition:opacity .2s ease-in-out}.ls-workflow-editor .ls-link-button-container.firstStep .add-step-btn .add-step-btn-txt,.ls-workflow-editor .ls-link-button-container:hover .add-step-btn .add-step-btn-txt{opacity:1}.ls-workflow-editor-default-layout{-ms-grid-columns:66% auto;-ms-grid-rows:100%;grid-template-columns:66% auto;grid-template-rows:100%}@media (min-width:1400px){.container{max-width:1340px}}@media (min-width:1600px){.container{max-width:1540px}}"]
            }] }
];
/** @nocollapse */
LsWorkflowEditorComponent.ctorParameters = () => [];
LsWorkflowEditorComponent.propDecorators = {
    endpointPopoverElementRef: [{ type: ViewChild, args: ['endpointPopoverHandle', { read: NgbPopover },] }],
    allCircuitEditionPopover: [{ type: ViewChildren, args: [NgbPopover,] }],
    popEditContentTemplateRef: [{ type: ContentChild, args: ['popEditContent',] }],
    popCreateContentTemplateRef: [{ type: ContentChild, args: ['popCreateContent',] }],
    endpointPopoverEditContentRef: [{ type: ContentChild, args: ['endpointPopoverEditContent',] }],
    stepDisplayContentRef: [{ type: ContentChild, args: ['stepDisplayContent',] }],
    lastStepDisplayContentRef: [{ type: ContentChild, args: ['lastStepDisplayContent',] }],
    endpointIsEditable: [{ type: Input }],
    getStepIconClassFct: [{ type: Input }],
    getStepTooltipFct: [{ type: Input }],
    confirmStepDeletionFct: [{ type: Input }],
    lastStepData: [{ type: Input }],
    workflowChange: [{ type: Output }],
    localizedMessages: [{ type: Input }],
    options: [{ type: Input }],
    workflow: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.currentEditedStepIdx;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.endpointPopoverElementRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.allCircuitEditionPopover;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.currentPopover;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.popEditContentTemplateRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.popCreateContentTemplateRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.endpointPopoverEditContentRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.stepDisplayContentRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.lastStepDisplayContentRef;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.messages;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.activeOptions;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.endpointIsEditable;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.getStepIconClassFct;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.getStepTooltipFct;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.confirmStepDeletionFct;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.lastStepData;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.workflowData;
    /** @type {?} */
    LsWorkflowEditorComponent.prototype.workflowChange;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/workflow-instance.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class WorkflowInstance {
    // This constructor is only valid while we use stub services,
    // It should be replace with a constructor from "circuitRaw"
    // constructor(circuitModel: Circuit, stepInstances?: StepInstance[]) {
    /**
     * @param {?=} instData
     */
    constructor(instData) {
        this.currentStepIdx = 0;
        this.rejected = false;
        this.isOver = false;
        this.id = instData.id;
        this.name = instData.name;
        this.description = instData.description;
        this.rejected = instData.isRejected;
        this.isOver = instData.isOver;
        if (instData.steps) {
            this.steps = instData.steps;
            this.currentStepIdx = instData.steps.findIndex((/**
             * @param {?} step
             * @return {?}
             */
            step => step.state === StepState.CURRENT));
            if (this.currentStepIdx === -1) {
                this.currentStepIdx = 0;
            }
        }
    }
}
if (false) {
    /** @type {?} */
    WorkflowInstance.prototype.id;
    /** @type {?} */
    WorkflowInstance.prototype.name;
    /** @type {?} */
    WorkflowInstance.prototype.description;
    /** @type {?} */
    WorkflowInstance.prototype.currentStepIdx;
    /** @type {?} */
    WorkflowInstance.prototype.steps;
    /** @type {?} */
    WorkflowInstance.prototype.rejected;
    /** @type {?} */
    WorkflowInstance.prototype.isOver;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/workflow-instance-option.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class WorkflowInstanceOptions extends WorkflowModelOptions {
}
if (false) {
    /** @type {?} */
    WorkflowInstanceOptions.prototype.validatedIconClass;
    /** @type {?} */
    WorkflowInstanceOptions.prototype.rejectedIconClass;
    /** @type {?} */
    WorkflowInstanceOptions.prototype.hideStart;
}
class WorkflowInstanceLocalizedMessages {
}
if (false) {
    /** @type {?} */
    WorkflowInstanceLocalizedMessages.prototype.FIRST_STEP_TXT;
    /** @type {?} */
    WorkflowInstanceLocalizedMessages.prototype.LAST_STEP_TXT;
}

/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflow-instance-viewer/ls-workflow-instance-viewer.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class LsWorkflowInstanceViewerComponent {
    constructor() {
        this.messages = {
            FIRST_STEP_TXT: 'Start',
            LAST_STEP_TXT: 'End'
        };
        this.reducedMode = true;
        this.isOver = false;
        this.isRejected = false;
        this._options = {
            startIconClass: 'far fa-flag',
            endIconClass: 'fas fa-flag-checkered',
            validatedIconClass: 'fas fa-check',
            rejectedIconClass: 'fas fa-times',
            hideStart: false
        };
    }
    /**
     * @return {?}
     */
    get options() {
        return this._options;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set options(value) {
        this._options = Object.assign(this._options, value);
    }
    /**
     * @return {?}
     */
    get localizedMessages() {
        return this.messages;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set localizedMessages(value) {
        this.messages = Object.assign(this.messages, value);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        console.warn('instanciating ls workflow instance viewer!');
        if (!this.workflowData) {
            console.error('Missing input circuit data');
            this.workflowData = new WorkflowInstance();
        }
        if (!this.messages) {
            throw new Error('Missing localised messages for LsWorkflowViewerComponent!');
        }
        if (!this.getStepIconClassFct) {
            throw new Error('Missing getStepIconClassFct for LsWorkflowViewerComponent!');
        }
        if (!this.getStepTooltipFct) {
            throw new Error('Missing getStepTooltipFct for LsWorkflowViewerComponent!');
        }
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        console.debug('LsWorkflowViewerComponent::ngOnChanges, ', this);
        this.activeStep = this.workflowData.currentStepIdx;
        this.isOver = this.workflowData.isOver;
        this.isRejected = this.workflowData.rejected;
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepIconClass(step) {
        return this.getStepIconClassFct(step);
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepTooltip(step) {
        return this.getStepTooltipFct(step);
    }
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    isActiveStep(stepIdx) {
        return !this.isRejected && !this.isOver && stepIdx === this.activeStep;
    }
    /**
     * @param {?} step
     * @return {?}
     */
    isPassed(step) {
        return (step.state === StepState.VALIDATED) || (step.state === StepState.REJECTED);
    }
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    isActualValidator(step, validator) {
        return step.validatedBy.some((/**
         * @param {?} elem
         * @return {?}
         */
        (elem) => elem.id === validator.id));
    }
    /**
     * @return {?}
     */
    isCircuitOver() {
        return this.isOver && !this.isRejected;
    }
    /**
     * @return {?}
     */
    getCircuitEndAdditionalClasses() {
        /** @type {?} */
        const classes = [];
        if (this.getCircuitEndAdditionalClassesFct) {
            classes.push(this.getCircuitEndAdditionalClassesFct());
        }
        else {
            classes.push(this._options.additionalFinalEndpointClasses || '');
        }
        if (this.reducedMode) {
            classes.push('reduced-end-step');
        }
        return classes;
    }
    /**
     * @return {?}
     */
    getAdditionalValidatorListClasses() {
        //wa-circuit-instance
    }
    /**
     * @param {?} step
     * @param {?} stepIdx
     * @return {?}
     */
    getStepAdditionalClasses(step, stepIdx) {
        // this.isActiveStep(stepIdx) ? 'wa-active-step wa-white-icn' : ''
        /** @type {?} */
        const classes = [];
        if (this.isActiveStep(stepIdx)) {
            classes.push('ls-active-step ls-white-icn');
        }
        if (this.reducedMode) {
            classes.push('ls-reduced-step');
        }
        if (!this.reducedMode) {
            if (step.state === StepState.SKIPPED) {
                classes.push('wa-circuit-skipped-step');
            }
        }
        if (step.state === StepState.VALIDATED) {
            classes.push('ls-validated-step');
        }
        return classes;
    }
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    getValidatorClasses(step, validator) {
        if (this.isActualValidator(step, validator)) {
            if (step.state === StepState.REJECTED) {
                return this.options.rejectedIconClass;
            }
            else {
                return this.options.validatedIconClass;
            }
        }
        return '';
    }
}
LsWorkflowInstanceViewerComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-workflow-instance-viewer',
                template: "<div class=\"ls-workflow-display\">\n  <div class=\"ls-workflow-canevas\">\n    <div class=\"circuit-start\">\n      <div *ngIf=\"!options.hideStart\" class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa fa-2x\" [ngClass]=\"options.startIconClass\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.FIRST_STEP_TXT\"></span>\n      </div>\n    </div>\n    <div class=\"steps-container\">\n      <div *ngFor=\"let step of workflowData.steps; index as i\">\n        <div class=\"ls-workflow-display-step-container\">\n          <div class=\"ls-workflow-step\" [ngClass]=\"getStepAdditionalClasses(step, i)\">\n            <div class=\"ls-step-top-line\">\n              <div class=\"step-number\">{{i + 1}}.</div>\n              <div class=\"step-name\">{{step.name}}</div>\n              <div [ngClass]=\"getStepIconClass(step)\" class=\"step-icn fa\" ngbTooltip=\"{{getStepTooltip(step)}}\"></div>\n            </div>\n            <div class=\"ls-display-step-separator\"></div>\n            <div class=\"ls-step-validator-list\" [ngClass]=\"getAdditionalValidatorListClasses\">\n              <ng-container *ngIf=\"isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"passedStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n              <ng-container *ngIf=\" ! isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\" class=\"ls-step-validator\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"pendingStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n            </div>\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"ls-workflow-end\" [ngClass]=\"getCircuitEndAdditionalClasses()\">\n      <div class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.endIconClass\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.LAST_STEP_TXT\"></span>\n      </div>\n    </div>\n  </div>\n</div>\n",
                styles: ["@charset \"UTF-8\";body{background-color:var(--ls-grey-50)}.active-btn{color:#42a1e8}.ls-workflow-canevas{-ms-flex-align:center;-ms-flex-direction:column;-ms-flex-pack:start;-ms-grid-column:1;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:vertical;-webkit-box-pack:start;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;grid-column:1;height:100%;justify-content:flex-start}.ls-workflow-step{-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);background-color:#fff;border:none;box-shadow:0 0 .25rem rgba(0,0,0,.15);min-height:6rem;width:var(--ls-workflow-step-width,20rem)}.ls-workflow-step .ls-step-top-line{-ms-flex-align:center;-ms-flex-direction:row;-ms-flex-pack:justify;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:horizontal;-webkit-box-pack:justify;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;font-weight:700;justify-content:space-between;padding:.5rem}.ls-workflow-step .ls-display-step-separator{border-bottom:1px solid #d3d3d3;width:100%}.ls-workflow-step .ls-step-validator-list{color:#5b5b5b;padding:.5rem .5rem .5rem 1.5rem}.ls-workflow-step.selected{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.ls-workflow-step .step-icn{height:1.7rem;width:1.7rem}.ls-workflow-endpoint-content{margin:auto;text-align:center}.ls-workflow-endpoint-content .ls-step-bounds-txt{color:var(--primary);font-weight:700}.ls-workflow-endpoint-content .ls-workflow-endpoint-icn{color:var(--primary)}.ls-workflow-endpoint-content.not-active .ls-step-bounds-txt,.ls-workflow-endpoint-content.not-active .ls-workflow-endpoint-icn{color:#5b5b5b}.ls-workflow-endpoint-content .ls-step-validator-list{text-align:left}.ls-actual-validator{font-weight:700}.ls-actual-validator:before{color:var(--primary);content:\"\uF00C\";font-family:var(--ls-font-awesome-family,\"Font Awesome 5 Free\");font-size:1em;left:.5em;position:absolute}.ls-rejector{font-weight:700}.ls-rejector:before{color:var(--danger);content:\"\uF00D\";font-family:var(--ls-font-awesome-family,\"Font Awesome 5 Free\");font-size:1em;left:.5em;position:absolute}.wa-comment-icon{color:var(--primary);font-size:1em;margin-right:.5em}.ls-workflow-display .ls-workflow-step{margin:1rem auto;padding-left:0;padding-right:0}.ls-workflow-display .ls-workflow-step .ls-step-validator-list{padding:1rem 1rem 1rem .5rem}.ls-workflow-display .ls-workflow-step .ls-step-validator-list.ls-workflow-instance{padding-left:2rem}.ls-workflow-display .ls-workflow-step.ls-reduced-step{-ms-grid-columns:(1fr)[12];-ms-grid-rows:2em auto;-webkit-box-shadow:none;background-color:transparent;border:none;box-shadow:none;display:-ms-grid;display:grid;grid-row-gap:.5em;grid-template-columns:repeat(12,1fr);grid-template-rows:2em auto;min-height:unset;width:var(--ls-workflow-reduced-step-width,16rem)}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line{align-items:center;display:grid;grid-column:1/-1;grid-row:1;grid-template-columns:1fr 2fr 9fr;grid-template-rows:1fr;justify-content:center;padding:0}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-number{display:none}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-icn{-ms-grid-column:2;-ms-grid-row:1;font-size:24px;grid-column:2;grid-row:1;height:unset;margin:auto;width:unset}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-name{-ms-grid-column:3;-ms-grid-row:1;grid-column:3;grid-row:1}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-display-step-separator{-ms-grid-column:3;-ms-grid-row:2;border-bottom:0;border-left:1px solid #d3d3d3;grid-column:3;grid-row:2}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-validator-list{-ms-grid-row:2;color:#d3d3d3;font-size:.8rem;font-style:italic;grid-column:4/-1;grid-row:2;padding:.5em}.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line,.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line .step-icn.test-icon-class,.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line .step-icn.test-icon-class:before{color:#23a237}.ls-workflow-display .reduced-end-step .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:1em}.ls-workflow-display .ls-workflow-endpoint-content{-ms-flex-direction:row;-webkit-box-direction:normal;-webkit-box-orient:horizontal;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row}.ls-workflow-display .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:.5rem}.ls-workflow-display .ls-step-top-line{-ms-flex-pack:start;-webkit-box-pack:start;justify-content:flex-start;padding-left:.5rem}.ls-workflow-display .ls-step-top-line .step-name{-ms-flex:1;-webkit-box-flex:1;flex:1;margin-left:.7rem}.ls-workflow-display .ls-workflow-display-step-container{position:relative}.ls-workflow-display .ls-active-step:not(.ls-reduced-step) .ls-step-top-line{background-color:var(--primary);color:#fff;margin:0}.ls-workflow-display .ls-active-step:not(.ls-reduced-step) .ls-step-top-line .step-icn{color:#fff}.ls-workflow-display .ls-active-step:before{color:var(--ls-workflow-arrow-color,var(--primary));content:\"\uF061\";font-family:var(--ls-font-awesome-family,\"Font Awesome 5 Free\");font-size:2em;font-style:normal;font-variant:normal;font-weight:900;left:-1em;line-height:1;position:absolute;text-rendering:auto}.ls-workflow-display .ls-active-step.end-step:before{font-size:1.2em}.ls-workflow-end{position:relative}"]
            }] }
];
/** @nocollapse */
LsWorkflowInstanceViewerComponent.ctorParameters = () => [];
LsWorkflowInstanceViewerComponent.propDecorators = {
    passedStepTemplateRef: [{ type: ContentChild, args: ['passedStepContent',] }],
    pendingStepTemplateRef: [{ type: ContentChild, args: ['pendingStepContent',] }],
    workflowData: [{ type: Input, args: ['src',] }],
    activeStep: [{ type: Input }],
    getStepIconClassFct: [{ type: Input }],
    getStepTooltipFct: [{ type: Input }],
    getCircuitEndAdditionalClassesFct: [{ type: Input }],
    reducedMode: [{ type: Input }],
    options: [{ type: Input }],
    localizedMessages: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.passedStepTemplateRef;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.pendingStepTemplateRef;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.workflowData;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.activeStep;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.messages;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getStepIconClassFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getStepTooltipFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getCircuitEndAdditionalClassesFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.reducedMode;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.isOver;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.isRejected;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype._options;
}

/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflows.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class LsWorkflowsModule {
    // This is for Components with contents
    // public static angularComponents = {
    //   // components
    //   'ls-lib-back-top-top': BackToTopComponent,
    //   'ls-lib-max-length-textarea': MaxLengthTextAreaComponent,
    //   'ls-lib-color-picker': ColorPickerComponent,
    //   'ls-lib-timepicker': TimepickerComponent,
    //   'ls-lib-password': PasswordComponent,
    //
    // };
    constructor() {
    }
}
LsWorkflowsModule.decorators = [
    { type: NgModule, args: [{
                declarations: [
                    LsWorkflowEditorComponent,
                    LsWorkflowModelViewerComponent,
                    LsWorkflowInstanceViewerComponent,
                ],
                imports: [
                    BrowserModule,
                    CommonModule,
                    BrowserAnimationsModule,
                    FormsModule,
                    NgbModule,
                    DragDropModule
                ],
                exports: [
                    LsWorkflowEditorComponent,
                    LsWorkflowModelViewerComponent,
                    LsWorkflowInstanceViewerComponent,
                ]
            },] }
];
/** @nocollapse */
LsWorkflowsModule.ctorParameters = () => [];

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/step-instance.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class StepInstance {
    /**
     * @param {?=} stepInstData
     */
    constructor(stepInstData) {
        this.validatedBy = [];
        this.id = stepInstData.id;
        this.name = stepInstData.name;
        this.validators = stepInstData.validators ? [...stepInstData.validators] : [];
        this.state = stepInstData.state;
        this.validatedBy = stepInstData.actedUponBy ? [stepInstData.actedUponBy] : [];
        this.comment = stepInstData.comment;
        if (stepInstData.actionId) {
            this.id = stepInstData.actionId;
        }
    }
}
if (false) {
    /** @type {?} */
    StepInstance.prototype.id;
    /** @type {?} */
    StepInstance.prototype.name;
    /** @type {?} */
    StepInstance.prototype.type;
    /** @type {?} */
    StepInstance.prototype.validators;
    /** @type {?} */
    StepInstance.prototype.state;
    /** @type {?} */
    StepInstance.prototype.validatedBy;
    /** @type {?} */
    StepInstance.prototype.comment;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/workflow-actor.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class WorkflowActor {
}
if (false) {
    /** @type {?} */
    WorkflowActor.prototype.id;
    /** @type {?} */
    WorkflowActor.prototype.lastname;
    /** @type {?} */
    WorkflowActor.prototype.firstname;
    /** @type {?} */
    WorkflowActor.prototype.name;
}

/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/public_workflow_models.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: public_api.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * Generated from: libriciel-ls-composants-workflows.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { LsWorkflowEditorComponent, LsWorkflowInstanceViewerComponent, LsWorkflowModelViewerComponent, LsWorkflowsModule, StepInstance, StepModel, StepState, WorkflowActor, WorkflowInstance, WorkflowModel, WorkflowModelLocalizedMessages, WorkflowModelOptions };
