// @libriciel/ls-composants/esm2015/lib/component/action-bar/action-bar.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/action-bar/action-bar.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { Weight } from '../../ui-concepts/weight';
/**
 * @template T
 */
export class ActionBarComponent {
    constructor() {
        this.primaryActions = { actionItems: [] };
        this.secondaryActions = { actionItems: [] };
        this.otherActions = { actionItems: [] };
        this.needReload = new EventEmitter();
        this.actionResult = new EventEmitter();
        this.weight = Weight;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.secondaryActions.actionItems
            .forEach((/**
         * @param {?} item
         * @return {?}
         */
        item => item.initializeItem()));
        this.primaryActions.actionItems
            .forEach((/**
         * @param {?} item
         * @return {?}
         */
        item => item.initializeItem()));
        this.otherActions.actionItems
            .forEach((/**
         * @param {?} item
         * @return {?}
         */
        item => item.initializeItem()));
    }
    /**
     * @param {?} $event
     * @return {?}
     */
    sendResult($event) {
        this.actionResult.emit($event);
    }
}
ActionBarComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-action-bar',
                template: "<div\n  *ngIf=\"secondaryActions?.actionItems.length>0 || primaryActions?.actionItems.length>0 || otherActions?.actionItems.length>0\"\n  class=\"ls-action-bar\"\n  role=\"group\">\n  <div *ngIf=\"secondaryActions?.actionItems.length>0\" class=\"ls-action-bar\">\n    <ng-container *ngFor=\"let item of secondaryActions.actionItems; let index = index\">\n      <ls-button (done)=\"sendResult($event)\"\n                 *ngIf=\"item.enabled && item.actionValidator?.isActionValid([argument])\"\n                 [action]=\"item\"\n                 [argument]=\"[argument]\"\n                 [icon]=\"item.icon\"\n                 [style]=\"item.style\"\n                 [title]=\"item.longName\"\n                 [weight]=\"weight.SECONDARY\"\n                 id=\"{{item.name | toSnakeCase}}-secondary\"\n                 lsActionable>\n      </ls-button>\n    </ng-container>\n  </div>\n  <div *ngIf=\"primaryActions?.actionItems.length>0\" class=\"ls-action-bar\">\n    <ng-container *ngFor=\"let item of primaryActions.actionItems; let index = index\">\n      <ls-button (done)=\"sendResult($event)\"\n                 *ngIf=\"item.enabled && item.actionValidator?.isActionValid([argument])\"\n                 [action]=\"item\"\n                 [argument]=\"[argument]\"\n                 [icon]=\"item.icon\"\n                 [style]=\"item.style\"\n                 [title]=\"item.longName\"\n                 [weight]=\"weight.PRIMARY\"\n                 id=\"{{item.name | toSnakeCase}}-primary\"\n                 lsActionable>\n      </ls-button>\n    </ng-container>\n  </div>\n  <div *ngIf=\"otherActions?.actionItems.length>0\" class=\"ls-action-bar\">\n    <ng-container *ngFor=\"let item of otherActions.actionItems; let index = index\">\n      <ls-button (done)=\"sendResult($event)\"\n                 *ngIf=\"item.enabled && item.actionValidator?.isActionValid([argument])\"\n                 [action]=\"item\"\n                 [argument]=\"[argument]\"\n                 [icon]=\"item.icon\"\n                 [style]=\"item.style\"\n                 [title]=\"item.longName\"\n                 [weight]=\"weight.SECONDARY\"\n                 id=\"{{item.name | toSnakeCase}}-other\"\n                 lsActionable>\n      </ls-button>\n    </ng-container>\n  </div>\n</div>\n",
                styles: [".ls-action-bar{display:-webkit-box;display:-ms-flexbox;display:flex}.ls-action-bar>:not(:last-child){margin-right:.5rem}"]
            }] }
];
/** @nocollapse */
ActionBarComponent.ctorParameters = () => [];
ActionBarComponent.propDecorators = {
    primaryActions: [{ type: Input }],
    secondaryActions: [{ type: Input }],
    otherActions: [{ type: Input }],
    argument: [{ type: Input }],
    needReload: [{ type: Output }],
    actionResult: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    ActionBarComponent.prototype.primaryActions;
    /** @type {?} */
    ActionBarComponent.prototype.secondaryActions;
    /** @type {?} */
    ActionBarComponent.prototype.otherActions;
    /** @type {?} */
    ActionBarComponent.prototype.argument;
    /** @type {?} */
    ActionBarComponent.prototype.needReload;
    /** @type {?} */
    ActionBarComponent.prototype.actionResult;
    /** @type {?} */
    ActionBarComponent.prototype.weight;
}
