// @libriciel/ls-composants/esm5/lib/component/ui-components.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/ui-components.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BackToTopComponent } from './back-to-top/back-to-top.component';
import { ColorPickerComponent } from './color-picker/color-picker.component';
import { MaxLengthTextAreaComponent } from './max-length-textarea/max-length-textarea.component';
import { PasswordComponent } from './password/password.component';
import { TimepickerComponent } from './timepicker/timepicker.component';
import { CommonModule } from '@angular/common';
import { BrowserModule } from '@angular/platform-browser';
import { LibersignComponent } from './libersign/libersign.component';
import { HttpClientModule } from '@angular/common/http';
var LsUiComponentsModule = /** @class */ (function () {
    function LsUiComponentsModule() {
    }
    // This is for Components with contents
    LsUiComponentsModule.angularComponents = {
        // components
        'ls-lib-back-top-top': BackToTopComponent,
        'ls-lib-max-length-textarea': MaxLengthTextAreaComponent,
        'ls-lib-color-picker': ColorPickerComponent,
        'ls-lib-timepicker': TimepickerComponent,
        'ls-lib-password': PasswordComponent,
        'ls-lib-libersign': LibersignComponent,
    };
    LsUiComponentsModule.decorators = [
        { type: NgModule, args: [{
                    schemas: [NO_ERRORS_SCHEMA],
                    entryComponents: [
                        BackToTopComponent,
                        ColorPickerComponent,
                        MaxLengthTextAreaComponent,
                        PasswordComponent,
                        TimepickerComponent,
                        LibersignComponent
                    ],
                    declarations: [
                        BackToTopComponent,
                        ColorPickerComponent,
                        MaxLengthTextAreaComponent,
                        PasswordComponent,
                        TimepickerComponent,
                        LibersignComponent
                    ],
                    imports: [
                        BrowserModule,
                        CommonModule,
                        BrowserAnimationsModule,
                        FormsModule,
                        ReactiveFormsModule,
                        HttpClientModule
                    ],
                    exports: [
                        BackToTopComponent,
                        ColorPickerComponent,
                        MaxLengthTextAreaComponent,
                        PasswordComponent,
                        TimepickerComponent,
                        LibersignComponent,
                    ]
                },] }
    ];
    /** @nocollapse */
    LsUiComponentsModule.ctorParameters = function () { return []; };
    return LsUiComponentsModule;
}());
export { LsUiComponentsModule };
if (false) {
    /** @type {?} */
    LsUiComponentsModule.angularComponents;
}
