// @libriciel/ls-composants/esm2015/workflows/workflows/ls-workflow-instance-viewer/ls-workflow-instance-viewer.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: workflows/ls-workflow-instance-viewer/ls-workflow-instance-viewer.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ContentChild, Input } from '@angular/core';
import { WorkflowInstance } from '../../shared/model/workflows/workflow-instance';
import { StepState } from '../../shared/model/workflows/step-state';
import { WorkflowInstanceLocalizedMessages, WorkflowInstanceOptions } from '../../shared/model/workflows/workflow-instance-option';
export class LsWorkflowInstanceViewerComponent {
    constructor() {
        this.messages = {
            FIRST_STEP_TXT: 'Start',
            LAST_STEP_TXT: 'End'
        };
        this.reducedMode = true;
        this.isOver = false;
        this.isRejected = false;
        this._options = {
            startIconClass: 'far fa-flag',
            endIconClass: 'fas fa-flag-checkered',
            validatedIconClass: 'fas fa-check',
            rejectedIconClass: 'fas fa-times',
            hideStart: false
        };
    }
    /**
     * @return {?}
     */
    get options() {
        return this._options;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set options(value) {
        this._options = Object.assign(this._options, value);
    }
    /**
     * @return {?}
     */
    get localizedMessages() {
        return this.messages;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set localizedMessages(value) {
        this.messages = Object.assign(this.messages, value);
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        console.warn('instanciating ls workflow instance viewer!');
        if (!this.workflowData) {
            console.error('Missing input circuit data');
            this.workflowData = new WorkflowInstance();
        }
        if (!this.messages) {
            throw new Error('Missing localised messages for LsWorkflowViewerComponent!');
        }
        if (!this.getStepIconClassFct) {
            throw new Error('Missing getStepIconClassFct for LsWorkflowViewerComponent!');
        }
        if (!this.getStepTooltipFct) {
            throw new Error('Missing getStepTooltipFct for LsWorkflowViewerComponent!');
        }
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        console.debug('LsWorkflowViewerComponent::ngOnChanges, ', this);
        this.activeStep = this.workflowData.currentStepIdx;
        this.isOver = this.workflowData.isOver;
        this.isRejected = this.workflowData.rejected;
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepIconClass(step) {
        return this.getStepIconClassFct(step);
    }
    /**
     * @param {?} step
     * @return {?}
     */
    getStepTooltip(step) {
        return this.getStepTooltipFct(step);
    }
    /**
     * @param {?} stepIdx
     * @return {?}
     */
    isActiveStep(stepIdx) {
        return !this.isRejected && !this.isOver && stepIdx === this.activeStep;
    }
    /**
     * @param {?} step
     * @return {?}
     */
    isPassed(step) {
        return (step.state === StepState.VALIDATED) || (step.state === StepState.REJECTED);
    }
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    isActualValidator(step, validator) {
        return step.validatedBy.some((/**
         * @param {?} elem
         * @return {?}
         */
        (elem) => elem.id === validator.id));
    }
    /**
     * @return {?}
     */
    isCircuitOver() {
        return this.isOver && !this.isRejected;
    }
    /**
     * @return {?}
     */
    getCircuitEndAdditionalClasses() {
        /** @type {?} */
        const classes = [];
        if (this.getCircuitEndAdditionalClassesFct) {
            classes.push(this.getCircuitEndAdditionalClassesFct());
        }
        else {
            classes.push(this._options.additionalFinalEndpointClasses || '');
        }
        if (this.reducedMode) {
            classes.push('reduced-end-step');
        }
        return classes;
    }
    /**
     * @return {?}
     */
    getAdditionalValidatorListClasses() {
        //wa-circuit-instance
    }
    /**
     * @param {?} step
     * @param {?} stepIdx
     * @return {?}
     */
    getStepAdditionalClasses(step, stepIdx) {
        // this.isActiveStep(stepIdx) ? 'wa-active-step wa-white-icn' : ''
        /** @type {?} */
        const classes = [];
        if (this.isActiveStep(stepIdx)) {
            classes.push('ls-active-step ls-white-icn');
        }
        if (this.reducedMode) {
            classes.push('ls-reduced-step');
        }
        if (!this.reducedMode) {
            if (step.state === StepState.SKIPPED) {
                classes.push('wa-circuit-skipped-step');
            }
        }
        if (step.state === StepState.VALIDATED) {
            classes.push('ls-validated-step');
        }
        return classes;
    }
    /**
     * @param {?} step
     * @param {?} validator
     * @return {?}
     */
    getValidatorClasses(step, validator) {
        if (this.isActualValidator(step, validator)) {
            if (step.state === StepState.REJECTED) {
                return this.options.rejectedIconClass;
            }
            else {
                return this.options.validatedIconClass;
            }
        }
        return '';
    }
}
LsWorkflowInstanceViewerComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-workflow-instance-viewer',
                template: "<div class=\"ls-workflow-display\">\n  <div class=\"ls-workflow-canevas\">\n    <div class=\"circuit-start\">\n      <div *ngIf=\"!options.hideStart\" class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa fa-2x\" [ngClass]=\"options.startIconClass\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.FIRST_STEP_TXT\"></span>\n      </div>\n    </div>\n    <div class=\"steps-container\">\n      <div *ngFor=\"let step of workflowData.steps; index as i\">\n        <div class=\"ls-workflow-display-step-container\">\n          <div class=\"ls-workflow-step\" [ngClass]=\"getStepAdditionalClasses(step, i)\">\n            <div class=\"ls-step-top-line\">\n              <div class=\"step-number\">{{i + 1}}.</div>\n              <div class=\"step-name\">{{step.name}}</div>\n              <div [ngClass]=\"getStepIconClass(step)\" class=\"step-icn fa\" ngbTooltip=\"{{getStepTooltip(step)}}\"></div>\n            </div>\n            <div class=\"ls-display-step-separator\"></div>\n            <div class=\"ls-step-validator-list\" [ngClass]=\"getAdditionalValidatorListClasses\">\n              <ng-container *ngIf=\"isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"passedStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n              <ng-container *ngIf=\" ! isPassed(step)\">\n                <div *ngFor=\"let validator of step.validators\" class=\"ls-step-validator\">\n                  <ng-template\n                    [ngTemplateOutlet]=\"pendingStepTemplateRef\"\n                    [ngTemplateOutletContext]=\"{step: step, validator: validator}\">\n                  </ng-template>\n                </div>\n              </ng-container>\n            </div>\n          </div>\n        </div>\n      </div>\n    </div>\n    <div class=\"ls-workflow-end\" [ngClass]=\"getCircuitEndAdditionalClasses()\">\n      <div class=\"ls-workflow-endpoint-content\">\n        <span class=\"ls-workflow-endpoint-icn fa fa-2x\" [ngClass]=\"options.endIconClass\"></span>\n        <span class=\"ls-step-bounds-txt\" [innerText]=\"messages.LAST_STEP_TXT\"></span>\n      </div>\n    </div>\n  </div>\n</div>\n",
                styles: ["@charset \"UTF-8\";body{background-color:var(--ls-grey-50)}.active-btn{color:#42a1e8}.ls-workflow-canevas{-ms-flex-align:center;-ms-flex-direction:column;-ms-flex-pack:start;-ms-grid-column:1;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:vertical;-webkit-box-pack:start;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;grid-column:1;height:100%;justify-content:flex-start}.ls-workflow-step{-webkit-box-shadow:0 0 .25rem rgba(0,0,0,.15);background-color:#fff;border:none;box-shadow:0 0 .25rem rgba(0,0,0,.15);min-height:6rem;width:var(--ls-workflow-step-width,20rem)}.ls-workflow-step .ls-step-top-line{-ms-flex-align:center;-ms-flex-direction:row;-ms-flex-pack:justify;-webkit-box-align:center;-webkit-box-direction:normal;-webkit-box-orient:horizontal;-webkit-box-pack:justify;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row;font-weight:700;justify-content:space-between;padding:.5rem}.ls-workflow-step .ls-display-step-separator{border-bottom:1px solid #d3d3d3;width:100%}.ls-workflow-step .ls-step-validator-list{color:#5b5b5b;padding:.5rem .5rem .5rem 1.5rem}.ls-workflow-step.selected{-webkit-box-shadow:0 .5rem .25rem rgba(0,0,0,.3);box-shadow:0 .5rem .25rem rgba(0,0,0,.3)}.ls-workflow-step .step-icn{height:1.7rem;width:1.7rem}.ls-workflow-endpoint-content{margin:auto;text-align:center}.ls-workflow-endpoint-content .ls-step-bounds-txt{color:var(--primary);font-weight:700}.ls-workflow-endpoint-content .ls-workflow-endpoint-icn{color:var(--primary)}.ls-workflow-endpoint-content.not-active .ls-step-bounds-txt,.ls-workflow-endpoint-content.not-active .ls-workflow-endpoint-icn{color:#5b5b5b}.ls-workflow-endpoint-content .ls-step-validator-list{text-align:left}.ls-actual-validator{font-weight:700}.ls-actual-validator:before{color:var(--primary);content:\"\uF00C\";font-family:var(--ls-font-awesome-family,\"Font Awesome 5 Free\");font-size:1em;left:.5em;position:absolute}.ls-rejector{font-weight:700}.ls-rejector:before{color:var(--danger);content:\"\uF00D\";font-family:var(--ls-font-awesome-family,\"Font Awesome 5 Free\");font-size:1em;left:.5em;position:absolute}.wa-comment-icon{color:var(--primary);font-size:1em;margin-right:.5em}.ls-workflow-display .ls-workflow-step{margin:1rem auto;padding-left:0;padding-right:0}.ls-workflow-display .ls-workflow-step .ls-step-validator-list{padding:1rem 1rem 1rem .5rem}.ls-workflow-display .ls-workflow-step .ls-step-validator-list.ls-workflow-instance{padding-left:2rem}.ls-workflow-display .ls-workflow-step.ls-reduced-step{-ms-grid-columns:(1fr)[12];-ms-grid-rows:2em auto;-webkit-box-shadow:none;background-color:transparent;border:none;box-shadow:none;display:-ms-grid;display:grid;grid-row-gap:.5em;grid-template-columns:repeat(12,1fr);grid-template-rows:2em auto;min-height:unset;width:var(--ls-workflow-reduced-step-width,16rem)}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line{align-items:center;display:grid;grid-column:1/-1;grid-row:1;grid-template-columns:1fr 2fr 9fr;grid-template-rows:1fr;justify-content:center;padding:0}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-number{display:none}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-icn{-ms-grid-column:2;-ms-grid-row:1;font-size:24px;grid-column:2;grid-row:1;height:unset;margin:auto;width:unset}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-top-line .step-name{-ms-grid-column:3;-ms-grid-row:1;grid-column:3;grid-row:1}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-display-step-separator{-ms-grid-column:3;-ms-grid-row:2;border-bottom:0;border-left:1px solid #d3d3d3;grid-column:3;grid-row:2}.ls-workflow-display .ls-workflow-step.ls-reduced-step .ls-step-validator-list{-ms-grid-row:2;color:#d3d3d3;font-size:.8rem;font-style:italic;grid-column:4/-1;grid-row:2;padding:.5em}.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line,.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line .step-icn.test-icon-class,.ls-workflow-display .ls-workflow-step.ls-reduced-step.ls-validated-step .ls-step-top-line .step-icn.test-icon-class:before{color:#23a237}.ls-workflow-display .reduced-end-step .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:1em}.ls-workflow-display .ls-workflow-endpoint-content{-ms-flex-direction:row;-webkit-box-direction:normal;-webkit-box-orient:horizontal;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:row}.ls-workflow-display .ls-workflow-endpoint-content .ls-step-bounds-txt{margin-left:.5rem}.ls-workflow-display .ls-step-top-line{-ms-flex-pack:start;-webkit-box-pack:start;justify-content:flex-start;padding-left:.5rem}.ls-workflow-display .ls-step-top-line .step-name{-ms-flex:1;-webkit-box-flex:1;flex:1;margin-left:.7rem}.ls-workflow-display .ls-workflow-display-step-container{position:relative}.ls-workflow-display .ls-active-step:not(.ls-reduced-step) .ls-step-top-line{background-color:var(--primary);color:#fff;margin:0}.ls-workflow-display .ls-active-step:not(.ls-reduced-step) .ls-step-top-line .step-icn{color:#fff}.ls-workflow-display .ls-active-step:before{color:var(--ls-workflow-arrow-color,var(--primary));content:\"\uF061\";font-family:var(--ls-font-awesome-family,\"Font Awesome 5 Free\");font-size:2em;font-style:normal;font-variant:normal;font-weight:900;left:-1em;line-height:1;position:absolute;text-rendering:auto}.ls-workflow-display .ls-active-step.end-step:before{font-size:1.2em}.ls-workflow-end{position:relative}"]
            }] }
];
/** @nocollapse */
LsWorkflowInstanceViewerComponent.ctorParameters = () => [];
LsWorkflowInstanceViewerComponent.propDecorators = {
    passedStepTemplateRef: [{ type: ContentChild, args: ['passedStepContent',] }],
    pendingStepTemplateRef: [{ type: ContentChild, args: ['pendingStepContent',] }],
    workflowData: [{ type: Input, args: ['src',] }],
    activeStep: [{ type: Input }],
    getStepIconClassFct: [{ type: Input }],
    getStepTooltipFct: [{ type: Input }],
    getCircuitEndAdditionalClassesFct: [{ type: Input }],
    reducedMode: [{ type: Input }],
    options: [{ type: Input }],
    localizedMessages: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.passedStepTemplateRef;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.pendingStepTemplateRef;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.workflowData;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.activeStep;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.messages;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getStepIconClassFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getStepTooltipFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.getCircuitEndAdditionalClassesFct;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.reducedMode;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.isOver;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype.isRejected;
    /** @type {?} */
    LsWorkflowInstanceViewerComponent.prototype._options;
}
