// @libriciel/ls-composants/esm2015/lib/directive/actionable.directive.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/directive/actionable.directive.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Directive, EventEmitter, HostListener, Input, Output } from '@angular/core';
import { ActionItem } from '../model/action-item';
/**
 * @template T
 */
export class ActionableDirective {
    constructor() {
        this.done = new EventEmitter();
    }
    /**
     * @return {?}
     */
    onClick() {
        this.action.actuator.action(this.argument)
            .subscribe((/**
         * @param {?} actionResult
         * @return {?}
         */
        actionResult => this.done.emit(actionResult)));
    }
}
ActionableDirective.decorators = [
    { type: Directive, args: [{
                selector: '[lsActionable]'
            },] }
];
/** @nocollapse */
ActionableDirective.ctorParameters = () => [];
ActionableDirective.propDecorators = {
    action: [{ type: Input }],
    argument: [{ type: Input }],
    done: [{ type: Output }],
    onClick: [{ type: HostListener, args: ['click',] }]
};
if (false) {
    /** @type {?} */
    ActionableDirective.prototype.action;
    /** @type {?} */
    ActionableDirective.prototype.argument;
    /** @type {?} */
    ActionableDirective.prototype.done;
}
