// @libriciel/ls-composants/esm2015/lib/pages/login/component/login-model/LoginPageModel.class.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/component/login-model/LoginPageModel.class.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { EventEmitter, Input, Output } from '@angular/core';
import { defaultConfiguration } from '../login-config/login-config.model';
/**
 * @abstract
 */
export class LoginPageModelClass {
    /**
     * @protected
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        this._DomSanitizationService = _DomSanitizationService;
        this.submitted = false;
        this.formAction = '';
        this.isFormAjax = false;
        this.showError = false;
        this.submit = new EventEmitter();
        this._visualConfiguration = defaultConfiguration;
    }
    /**
     * @param {?} config
     * @return {?}
     */
    set visualConfiguration(config) {
        this._visualConfiguration = typeof config === 'string' ? JSON.parse(config) : config;
    }
    /**
     * @param {?} event
     * @return {?}
     */
    preventFormSubmit(event) {
        if (this.isFormAjax) {
            this.submitted = true;
            event.preventDefault();
        }
    }
}
LoginPageModelClass.propDecorators = {
    formAction: [{ type: Input }],
    logo: [{ type: Input }],
    isFormAjax: [{ type: Input }],
    showError: [{ type: Input }],
    submit: [{ type: Output }],
    visualConfiguration: [{ type: Input }]
};
if (false) {
    /** @type {?} */
    LoginPageModelClass.prototype.submitted;
    /** @type {?} */
    LoginPageModelClass.prototype.formAction;
    /** @type {?} */
    LoginPageModelClass.prototype.logo;
    /** @type {?} */
    LoginPageModelClass.prototype.isFormAjax;
    /** @type {?} */
    LoginPageModelClass.prototype.showError;
    /** @type {?} */
    LoginPageModelClass.prototype.submit;
    /** @type {?} */
    LoginPageModelClass.prototype._visualConfiguration;
    /** @type {?} */
    LoginPageModelClass.prototype._DomSanitizationService;
}
