// @libriciel/ls-composants/esm5/lib/component/timepicker/timepicker.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/timepicker/timepicker.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, forwardRef, HostBinding, Input, Renderer2, ViewChild } from '@angular/core';
import { animate, animateChild, group, query, state, style, transition, trigger } from '@angular/animations';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
var TimepickerComponent = /** @class */ (function () {
    function TimepickerComponent(el, renderer2) {
        this.el = el;
        this.renderer2 = renderer2;
        this.hoursElements = [
            'subrot270',
            'rot300',
            'rot330',
            'rot0',
            'rot30',
            'rot60',
            'rot90',
            'rot120',
            'rot150',
            'rot180',
            'rot210',
            'rot240',
            'rot270',
            'subrot300',
            'subrot330',
            'subrot0',
            'subrot30',
            'subrot60',
            'subrot90',
            'subrot120',
            'subrot150',
            'subrot180',
            'subrot210',
            'subrot240',
        ];
        this.minutesElements = [
            'rot270',
            'rot300',
            'rot330',
            'rot0',
            'rot30',
            'rot60',
            'rot90',
            'rot120',
            'rot150',
            'rot180',
            'rot210',
            'rot240',
        ];
        this.hour = 0;
        this.minute = 0;
        this.rot = 0;
        this.showWrapper = false;
        this.updateMode = false;
        this.hourMode = true;
        this.disabled = false;
        this.classInputGroup = true;
        this.classInputGroupEndIcon = true;
        // noinspection JSUnusedLocalSymbols
        this.onChange = (/**
         * @param {?} time
         * @return {?}
         */
        function (time) {
            // tslint:disable-next-line:semicolon
        });
        this.onTouched = (/**
         * @return {?}
         */
        function () {
            // tslint:disable-next-line:semicolon
        });
    }
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.ngAfterViewInit = /**
     * @return {?}
     */
    function () {
        this.moveAttributeToInput('id');
        this.moveAttributeToInput('tabindex');
        this.moveAttributeToInput('required', '');
        this.moveAttributeToInput('name', 'timepicker');
    };
    /**
     * @private
     * @param {?} attributeName
     * @param {?=} defaultValue
     * @return {?}
     */
    TimepickerComponent.prototype.moveAttributeToInput = /**
     * @private
     * @param {?} attributeName
     * @param {?=} defaultValue
     * @return {?}
     */
    function (attributeName, defaultValue) {
        /** @type {?} */
        var attribute = this.el.nativeElement.getAttribute(attributeName);
        if (attribute !== null) {
            this.renderer2.removeAttribute(this.el.nativeElement, attributeName);
        }
        if (attribute !== null || defaultValue) {
            this.renderer2.setAttribute(this.input.nativeElement, attributeName, attribute ? attribute : defaultValue);
        }
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.validate();
    };
    /**
     * @param {?} num
     * @return {?}
     */
    TimepickerComponent.prototype.pad = /**
     * @param {?} num
     * @return {?}
     */
    function (num) {
        return '00'.substring(0, 2 - ('' + num).length) + num;
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.show = /**
     * @return {?}
     */
    function () {
        var _this = this;
        this.showWrapper = true;
        setTimeout((/**
         * @return {?}
         */
        function () { return _this.setHour(_this.hour); }), 0);
    };
    /**
     * @param {?} nR
     * @return {?}
     */
    TimepickerComponent.prototype.rotateThis = /**
     * @param {?} nR
     * @return {?}
     */
    function (nR) {
        /** @type {?} */
        var aR;
        this.rot = this.rot || 0; // if rot undefined or 0, make 0, else rot
        aR = this.rot % 360;
        if (aR < 0) {
            aR += 360;
        }
        if (aR < 180 && (nR > (aR + 180))) {
            this.rot -= 360;
        }
        if (aR >= 180 && (nR <= (aR - 180))) {
            this.rot += 360;
        }
        this.rot += (nR - aR);
        return 'rotate( ' + this.rot + 'deg )';
    };
    /**
     * @param {?} hour
     * @return {?}
     */
    TimepickerComponent.prototype.setHour = /**
     * @param {?} hour
     * @return {?}
     */
    function (hour) {
        /** @type {?} */
        var transform = this.rotateThis((hour * 30) % 360);
        if (hour === 0 || hour > 12) {
            transform += ' scale(0.6)';
        }
        this.el.nativeElement.getElementsByClassName('time-picker-clock-arrow')[0].style.transform = transform;
        this.hour = hour % 24;
    };
    /**
     * @param {?} minute
     * @return {?}
     */
    TimepickerComponent.prototype.setMinute = /**
     * @param {?} minute
     * @return {?}
     */
    function (minute) {
        this.minute = minute % 60;
        this.el.nativeElement.getElementsByClassName('time-picker-clock-arrow')[0].style.transform = this.rotateThis(minute / 5 * 30);
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.mousedown = /**
     * @return {?}
     */
    function () {
        this.updateMode = true;
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.toHourMode = /**
     * @return {?}
     */
    function () {
        this.hourMode = true;
        this.setHour(this.hour);
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.toMinuteMode = /**
     * @return {?}
     */
    function () {
        this.hourMode = false;
        this.setMinute(this.minute);
    };
    /**
     * @param {?} event
     * @return {?}
     */
    TimepickerComponent.prototype.mouseup = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        this.move(event);
        this.updateMode = false;
        if (this.hourMode) {
            this.toMinuteMode();
        }
    };
    /**
     * @param {?} event
     * @return {?}
     */
    TimepickerComponent.prototype.move = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.updateMode && event.target instanceof HTMLButtonElement) {
            /** @type {?} */
            var btn = event.target;
            /** @type {?} */
            var nodes = Array.prototype.slice.call(btn.parentElement.children);
            /** @type {?} */
            var index = nodes.indexOf(btn);
            if (this.hourMode) {
                this.setHour(index);
            }
            else {
                this.setMinute(index * 5);
            }
        }
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.hide = /**
     * @return {?}
     */
    function () {
        this.showWrapper = false;
        this.updateMode = false;
        this.hourMode = true;
    };
    /**
     * @return {?}
     */
    TimepickerComponent.prototype.validate = /**
     * @return {?}
     */
    function () {
        this.writeValue(this.pad(this.hour) + ':' + this.pad(this.minute));
        this.el.nativeElement.getElementsByTagName('input')[0].value = this.pad(this.hour) + ':' + this.pad(this.minute);
        this.hide();
    };
    /**
     * @param {?} $event
     * @return {?}
     */
    TimepickerComponent.prototype.inputChange = /**
     * @param {?} $event
     * @return {?}
     */
    function ($event) {
        this.onChange($event.currentTarget.value);
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    TimepickerComponent.prototype.registerOnChange = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onChange = fn;
    };
    /**
     * @param {?} fn
     * @return {?}
     */
    TimepickerComponent.prototype.registerOnTouched = /**
     * @param {?} fn
     * @return {?}
     */
    function (fn) {
        this.onTouched = fn;
    };
    /**
     * @param {?} isDisabled
     * @return {?}
     */
    TimepickerComponent.prototype.setDisabledState = /**
     * @param {?} isDisabled
     * @return {?}
     */
    function (isDisabled) {
        this.disabled = isDisabled;
    };
    /**
     * @param {?} obj
     * @return {?}
     */
    TimepickerComponent.prototype.writeValue = /**
     * @param {?} obj
     * @return {?}
     */
    function (obj) {
        if (obj) {
            this.el.nativeElement.getElementsByTagName('input')[0].value = obj;
            if (typeof obj === 'string') {
                /** @type {?} */
                var time = obj.split(':');
                this.hour = +time[0];
                this.minute = +time[1];
            }
        }
        this.onChange(obj);
    };
    TimepickerComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-timepicker',
                    template: "<div class=\"wrapper\" *ngIf=\"showWrapper\" [@fadeInOut]>\n  <div class=\"shadow time-picker\" [@popOut]>\n    <div class=\"time-picker-header\">\n      <span [ngClass]=\"hourMode ? 'active' : ''\" class=\"pointer-event\"\n            (click)=\"this.toHourMode()\">{{pad(hour % 24)}}</span>\n      <span> : </span>\n      <span [ngClass]=\"hourMode ? '' : 'active'\" class=\"pointer-event\"\n            (click)=\"this.toMinuteMode()\">{{pad(minute % 60)}}</span>\n    </div>\n    <div class=\"time-picker-content\">\n      <div class=\"background\"></div>\n      <div class=\"clock\" (mousemove)=\"move($event)\" (mouseup)=\"mouseup($event)\" (mousedown)=\"mousedown()\">\n\n        <div class=\"time-picker-hour\" [ngClass]=\"hourMode ? 'active' : ''\">\n          <button *ngFor=\"let rot of hoursElements; let i = index\" class=\"{{rot}} hourglass-btn\"\n                  [ngClass]=\"i == hour ? 'active' : ''\">\n            {{i}}\n          </button>\n        </div>\n        <div class=\"time-picker-minute\" [ngClass]=\"hourMode ? '' : 'active'\">\n          <button *ngFor=\"let rot of minutesElements; let i = index\" class=\"{{rot}} hourglass-btn\"\n                  [ngClass]=\"i == (minute / 5) ? 'active' : ''\" (click)=\"setMinute(i * 5)\">\n            {{((i) * 5) % 60}}\n          </button>\n        </div>\n\n        <div class=\"time-picker-clock-origin\"></div>\n        <div class=\"time-picker-clock-arrow\"></div>\n      </div>\n    </div>\n    <div class=\"time-picker-footer\">\n      <button class=\"btn btn-outline-primary\" (click)=\"hide()\">Annuler</button>\n      <button class=\"btn btn-primary\" (click)=\"validate()\">OK</button>\n    </div>\n  </div>\n</div>\n\n<ng-container>\n  <input type=\"time\" #input class=\"ls-action-input form-control time-picker-input\"\n         (click)=\"show()\"\n         (change)=\"inputChange($event)\" (blur)=\"onTouched();\">\n  <span class=\"input-icon\" (click)=\"show()\">\n    <i class=\"far fa-clock\"></i>\n  </span>\n</ng-container>\n",
                    providers: [{
                            provide: NG_VALUE_ACCESSOR,
                            useExisting: forwardRef((/**
                             * @return {?}
                             */
                            function () { return TimepickerComponent; })),
                            multi: true
                        }],
                    animations: [
                        trigger('fadeInOut', [
                            transition('void => *', [
                                style({
                                    opacity: 0
                                }),
                                group([
                                    animate(300, style({
                                        opacity: 1
                                    })),
                                    query('@popOut', animateChild())
                                ])
                            ]),
                            transition('* => void', [
                                style({
                                    opacity: 1
                                }),
                                group([
                                    animate(300, style({
                                        opacity: 0
                                    })),
                                    query('@popOut', animateChild())
                                ])
                            ]),
                        ]),
                        trigger('popOut', [
                            // ...
                            state('void', style({
                                transform: 'scale(0.6)'
                            })),
                            transition('void <=> *', animate(300)),
                        ])
                    ],
                    styles: [".form-group{position:relative}.input-icon{position:absolute;top:0;right:0;background-color:transparent;color:var(--ls-grey-500);cursor:pointer;height:99%;width:40px;border-radius:5px 0 0 5px;display:-webkit-inline-box;display:-ms-inline-flexbox;display:inline-flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center}.input-icon:hover{color:var(--ls-grey-700)}.input-group-end-icon,:host.input-group-end-icon{position:relative;display:-webkit-box;display:-ms-flexbox;display:flex}.input-group-end-icon input,:host.input-group-end-icon input{padding-right:36px}input.ls-action-input{cursor:pointer}.input-group.input-group-end-icon .input-icon,:host.input-group.input-group-end-icon .input-icon{z-index:3}.hourglass-btn{position:absolute;background-color:transparent;-webkit-transition:color .2s,background-color .2s;transition:color .2s,background-color .2s;cursor:pointer;width:35px;height:35px;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;border-radius:50%;z-index:1;border:none}.hourglass-btn.active,.hourglass-btn:hover{background-color:var(--ls-primary-500);color:#fff}.rot0{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(0) translate(115px) rotate(0);transform:translate3d(127px,130px,0) rotate(0) translate(115px) rotate(0)}.subrot0{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(0) translate(70px) rotate(0);transform:translate3d(127px,130px,0) rotate(0) translate(70px) rotate(0)}.rot30{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(30deg) translate(115px) rotate(-30deg);transform:translate3d(127px,130px,0) rotate(30deg) translate(115px) rotate(-30deg)}.subrot30{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(30deg) translate(70px) rotate(-30deg);transform:translate3d(127px,130px,0) rotate(30deg) translate(70px) rotate(-30deg)}.rot60{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(60deg) translate(115px) rotate(-60deg);transform:translate3d(127px,130px,0) rotate(60deg) translate(115px) rotate(-60deg)}.subrot60{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(60deg) translate(70px) rotate(-60deg);transform:translate3d(127px,130px,0) rotate(60deg) translate(70px) rotate(-60deg)}.rot90{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(90deg) translate(115px) rotate(-90deg);transform:translate3d(127px,130px,0) rotate(90deg) translate(115px) rotate(-90deg)}.subrot90{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(90deg) translate(70px) rotate(-90deg);transform:translate3d(127px,130px,0) rotate(90deg) translate(70px) rotate(-90deg)}.rot120{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(120deg) translate(115px) rotate(-120deg);transform:translate3d(127px,130px,0) rotate(120deg) translate(115px) rotate(-120deg)}.subrot120{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(120deg) translate(70px) rotate(-120deg);transform:translate3d(127px,130px,0) rotate(120deg) translate(70px) rotate(-120deg)}.rot150{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(150deg) translate(115px) rotate(-150deg);transform:translate3d(127px,130px,0) rotate(150deg) translate(115px) rotate(-150deg)}.subrot150{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(150deg) translate(70px) rotate(-150deg);transform:translate3d(127px,130px,0) rotate(150deg) translate(70px) rotate(-150deg)}.rot180{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(180deg) translate(115px) rotate(-180deg);transform:translate3d(127px,130px,0) rotate(180deg) translate(115px) rotate(-180deg)}.subrot180{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(180deg) translate(70px) rotate(-180deg);transform:translate3d(127px,130px,0) rotate(180deg) translate(70px) rotate(-180deg)}.rot210{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(210deg) translate(115px) rotate(-210deg);transform:translate3d(127px,130px,0) rotate(210deg) translate(115px) rotate(-210deg)}.subrot210{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(210deg) translate(70px) rotate(-210deg);transform:translate3d(127px,130px,0) rotate(210deg) translate(70px) rotate(-210deg)}.rot240{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(240deg) translate(115px) rotate(-240deg);transform:translate3d(127px,130px,0) rotate(240deg) translate(115px) rotate(-240deg)}.subrot240{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(240deg) translate(70px) rotate(-240deg);transform:translate3d(127px,130px,0) rotate(240deg) translate(70px) rotate(-240deg)}.rot270{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(270deg) translate(115px) rotate(-270deg);transform:translate3d(127px,130px,0) rotate(270deg) translate(115px) rotate(-270deg)}.subrot270{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(270deg) translate(70px) rotate(-270deg);transform:translate3d(127px,130px,0) rotate(270deg) translate(70px) rotate(-270deg)}.rot300{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(300deg) translate(115px) rotate(-300deg);transform:translate3d(127px,130px,0) rotate(300deg) translate(115px) rotate(-300deg)}.subrot300{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(300deg) translate(70px) rotate(-300deg);transform:translate3d(127px,130px,0) rotate(300deg) translate(70px) rotate(-300deg)}.rot330{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(330deg) translate(115px) rotate(-330deg);transform:translate3d(127px,130px,0) rotate(330deg) translate(115px) rotate(-330deg)}.subrot330{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(330deg) translate(70px) rotate(-330deg);transform:translate3d(127px,130px,0) rotate(330deg) translate(70px) rotate(-330deg)}.rot360{color:rgba(0,0,0,.8);-webkit-transform:translate3d(127px,130px,0) rotate(360deg) translate(115px) rotate(-360deg);transform:translate3d(127px,130px,0) rotate(360deg) translate(115px) rotate(-360deg)}.subrot360{color:rgba(0,0,0,.6);font-size:.8rem;-webkit-transform:translate3d(127px,130px,0) rotate(360deg) translate(70px) rotate(-360deg);transform:translate3d(127px,130px,0) rotate(360deg) translate(70px) rotate(-360deg)}.time-picker{background-color:#fff;width:300px}.time-picker>*{min-height:88px;position:relative}.time-picker-clock-origin{width:6px;height:6px;border-radius:50%;position:absolute;left:50%;top:50%;background:var(--ls-primary-500);margin-left:-3px;margin-top:-3px}.time-picker-clock-arrow{width:2px;height:41%;position:absolute;left:0;top:27px;right:0;margin:auto;-webkit-transform-origin:bottom;-ms-transform-origin:bottom;transform-origin:bottom;background:var(--ls-primary-500);-webkit-transition:-webkit-transform .2s;transition:transform .2s;transition:transform .2s,-webkit-transform .2s}.time-picker-hour,.time-picker-minute{position:relative;z-index:1;opacity:0;-webkit-transform:translateY(130px) scale(0);-ms-transform:translateY(130px) scale(0);transform:translateY(130px) scale(0);-webkit-transition:opacity .2s,-webkit-transform .2s;transition:opacity .2s,transform .2s,-webkit-transform .2s}.time-picker-hour.active,.time-picker-minute.active{opacity:1;-webkit-transform:translateY(0) scale(1);-ms-transform:translateY(0) scale(1);transform:translateY(0) scale(1)}.time-picker-content{height:300px;width:290px;margin:10px 5px 5px}.time-picker-content .background{position:absolute;width:100%;height:100%;border-radius:50%;background:var(--ls-primary-100)}.time-picker-header{background-color:var(--ls-primary-500);color:#fff;font-size:3.2rem;display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center}.time-picker-header *{opacity:.4}.time-picker-header .active{opacity:1}.time-picker-header .pointer-event{cursor:pointer}.time-picker-footer{display:-webkit-box;display:-ms-flexbox;display:flex;-ms-flex-pack:distribute;justify-content:space-around;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.wrapper{position:fixed;top:0;bottom:0;left:0;right:0;font-family:Roboto,sans-serif;z-index:9999;background:rgba(0,0,0,.3);display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-align:center;-ms-flex-align:center;align-items:center;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center}"]
                }] }
    ];
    /** @nocollapse */
    TimepickerComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: Renderer2 }
    ]; };
    TimepickerComponent.propDecorators = {
        disabled: [{ type: Input }],
        input: [{ type: ViewChild, args: ['input',] }],
        classInputGroup: [{ type: HostBinding, args: ['class.input-group',] }],
        classInputGroupEndIcon: [{ type: HostBinding, args: ['class.input-group-end-icon',] }]
    };
    return TimepickerComponent;
}());
export { TimepickerComponent };
if (false) {
    /** @type {?} */
    TimepickerComponent.prototype.hoursElements;
    /** @type {?} */
    TimepickerComponent.prototype.minutesElements;
    /** @type {?} */
    TimepickerComponent.prototype.hour;
    /** @type {?} */
    TimepickerComponent.prototype.minute;
    /** @type {?} */
    TimepickerComponent.prototype.rot;
    /** @type {?} */
    TimepickerComponent.prototype.showWrapper;
    /** @type {?} */
    TimepickerComponent.prototype.updateMode;
    /** @type {?} */
    TimepickerComponent.prototype.hourMode;
    /** @type {?} */
    TimepickerComponent.prototype.disabled;
    /** @type {?} */
    TimepickerComponent.prototype.input;
    /** @type {?} */
    TimepickerComponent.prototype.classInputGroup;
    /** @type {?} */
    TimepickerComponent.prototype.classInputGroupEndIcon;
    /** @type {?} */
    TimepickerComponent.prototype.onChange;
    /** @type {?} */
    TimepickerComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    TimepickerComponent.prototype.el;
    /**
     * @type {?}
     * @private
     */
    TimepickerComponent.prototype.renderer2;
}
