// @libriciel/ls-composants/esm2015/lib/model/action-item.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/model/action-item.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { of } from 'rxjs';
import { Style } from '../ui-concepts/style';
/**
 * @template T
 */
export class ActionItem {
    /**
     * @param {?} object
     */
    constructor(object) {
        this.name = object.name;
        this.longName = object.longName ? object.longName : object.name;
        this.icon = object.icon;
        this.style = object.style ? object.style : Style.NORMAL;
        this.actuator = object.actuator;
        this.actionValidator = object.actionValidator;
        this.enabled = object.enabled;
    }
    /**
     * @return {?}
     */
    initializeItem() {
        /** @type {?} */
        const self = this;
        this.icon = this.icon && this.icon !== '' ? this.icon : 'fa fa-cogs';
        this.enabled = true;
        this.actionValidator = this.actionValidator
            ? this.actionValidator
            : new class {
                constructor() {
                    this.isActionValid = (/**
                     * @return {?}
                     */
                    () => true);
                }
            }();
        this.actuator = this.actuator
            ? this.actuator
            : new class {
                /**
                 * @param {?} elements
                 * @return {?}
                 */
                action(elements) {
                    // tslint:disable-next-line:no-console
                    console.debug(self.name + ' clicked on elements ' + elements.map((/**
                     * @param {?} value
                     * @return {?}
                     */
                    value => value ? value.id : '' + '')).join(','));
                    return of({});
                }
            }();
    }
}
if (false) {
    /** @type {?} */
    ActionItem.prototype.name;
    /** @type {?} */
    ActionItem.prototype.longName;
    /** @type {?} */
    ActionItem.prototype.icon;
    /** @type {?} */
    ActionItem.prototype.style;
    /** @type {?} */
    ActionItem.prototype.actuator;
    /** @type {?} */
    ActionItem.prototype.actionValidator;
    /** @type {?} */
    ActionItem.prototype.enabled;
}
