// @libriciel/ls-composants/esm2015/lib/component/buttons/save-button/save-button.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/buttons/save-button/save-button.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component } from '@angular/core';
import { ButtonComponent } from '../button/button.component';
import { CommonMessages } from '../../../i18n/common-messages';
import { CommonIcons } from '../../../icons/common-icons';
export class SaveButtonComponent extends ButtonComponent {
    /**
     * @return {?}
     */
    ngOnInit() {
        this.title = this.title ? this.title : CommonMessages.SAVE;
        this.icon = this.icon ? this.icon : CommonIcons.SAVE_ICON;
        super.ngOnInit();
    }
}
SaveButtonComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-button-save',
                template: "<button #buttonElement [disabled]=\"disabled || processing\"\n        (click)=\"btnClick.emit($event)\"\n        [ngClass]=\"getBtnClasses()\" class=\"btn btn-icon-label\" title=\"{{title}}\" type=\"button\">\n  <span *ngIf=\"iconOnLeft && icon\" [ngClass]=\"processing  ? processingIcon : icon + getIconClasses()\"></span>\n  <span *ngIf=\"title && !onlyIcon\" [innerText]=\"title\"></span>\n  <span *ngIf=\"!iconOnLeft && icon\" [ngClass]=\"processing  ? processingIcon : icon + getIconClasses()\"></span>\n</button>\n",
                styles: [":host .btn{cursor:pointer;white-space:nowrap}:host .btn.no-text{border-radius:0;height:100%;padding:0;width:100%}:host .btn.menu-item{border-radius:0;text-align:left;width:100%}:host .btn:disabled{cursor:not-allowed}:host .btn:not(.no-text-button) span.fa:not(:first-child),:host .btn:not(.no-text-button) span.far:not(:first-child),:host .btn:not(.no-text-button) span.fas:not(:first-child){margin-left:.5rem}:host .btn:not(.no-text-button) span.fa:not(:last-child),:host .btn:not(.no-text-button) span.far:not(:last-child),:host .btn:not(.no-text-button) span.fas:not(:last-child){margin-right:.5rem}"]
            }] }
];
