// @libriciel/ls-composants/esm2015/lib/structure/structure.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/structure/structure.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule } from '@angular/forms';
import { FooterComponent } from './footer/footer.component';
import { SidebarComponent } from './sidebar/sidebar.component';
import { ContentComponent } from './content/content.component';
import { SidebarService } from './sidebar/sidebar.service';
import { LsUiComponentsModule } from '../component/ui-components.module';
export class LsStructureModule {
    constructor() {
    }
}
LsStructureModule.angularComponents = {
    // structure
    'ls-lib-footer': FooterComponent,
    'ls-lib-sidebar': SidebarComponent,
    'ls-lib-content': ContentComponent,
};
LsStructureModule.decorators = [
    { type: NgModule, args: [{
                entryComponents: [
                    FooterComponent,
                    SidebarComponent,
                    ContentComponent,
                ],
                declarations: [
                    FooterComponent,
                    SidebarComponent,
                    ContentComponent,
                ],
                imports: [
                    BrowserModule,
                    BrowserAnimationsModule,
                    FormsModule,
                    LsUiComponentsModule
                ],
                exports: [
                    FooterComponent,
                    SidebarComponent,
                    ContentComponent,
                ],
                providers: [
                    SidebarService
                ]
            },] }
];
/** @nocollapse */
LsStructureModule.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    LsStructureModule.angularComponents;
}
