// @libriciel/ls-composants/esm5/lib/structure/content/content.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/structure/content/content.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef } from '@angular/core';
import { SidebarService } from '../sidebar/sidebar.service';
var ContentComponent = /** @class */ (function () {
    function ContentComponent(el, sidebarService) {
        var _this = this;
        this.el = el;
        this.sidebarService = sidebarService;
        this.sidebarService.collapse$
            .subscribe((/**
         * @param {?} isCollapsed
         * @return {?}
         */
        function (isCollapsed) { return _this.collapseEvent(isCollapsed); }));
    }
    /**
     * @return {?}
     */
    ContentComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () { };
    /**
     * @param {?} isCollapsed
     * @return {?}
     */
    ContentComponent.prototype.collapseEvent = /**
     * @param {?} isCollapsed
     * @return {?}
     */
    function (isCollapsed) {
        if (isCollapsed) {
            this.el.nativeElement.classList.add('sidebar-collapsed');
        }
        else {
            this.el.nativeElement.classList.remove('sidebar-collapsed');
        }
    };
    ContentComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-content',
                    template: "<main>\n  <ng-content></ng-content>\n  <ls-back-to-top></ls-back-to-top>\n</main>\n",
                    styles: [":host.sidebar-collapsed main{padding-left:7rem}:host:not(.sidebar-collapsed) main{padding-left:16rem}:host main{min-height:calc(100vh - 1.9rem);padding-right:4rem;padding-top:3.5rem;padding-bottom:3rem;-webkit-transition:padding-left .3s;transition:padding-left .3s}@media only screen and (max-width:768px){:host:not(.sidebar-collapsed) main{padding-left:4rem}:host.sidebar-collapsed main{padding-left:4rem}:host main{padding-right:.5rem}}"]
                }] }
    ];
    /** @nocollapse */
    ContentComponent.ctorParameters = function () { return [
        { type: ElementRef },
        { type: SidebarService }
    ]; };
    return ContentComponent;
}());
export { ContentComponent };
if (false) {
    /**
     * @type {?}
     * @private
     */
    ContentComponent.prototype.el;
    /**
     * @type {?}
     * @private
     */
    ContentComponent.prototype.sidebarService;
}
