// @libriciel/ls-composants/esm5/lib/pages/login/login-model/LoginPageModel.class.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/login-model/LoginPageModel.class.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { EventEmitter, Input, Output } from '@angular/core';
import { defaultConfiguration } from '../login-config/login-config.model';
var LoginPageModelClass = /** @class */ (function () {
    function LoginPageModelClass(_DomSanitizationService) {
        this._DomSanitizationService = _DomSanitizationService;
        this.submitted = false;
        this.formAction = '';
        this.isFormAjax = false;
        this.showError = false;
        this.submit = new EventEmitter();
        this._visualConfiguration = defaultConfiguration;
    }
    Object.defineProperty(LoginPageModelClass.prototype, "visualConfiguration", {
        set: /**
         * @param {?} config
         * @return {?}
         */
        function (config) {
            this._visualConfiguration = typeof config === 'string' ? JSON.parse(config) : config;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} event
     * @return {?}
     */
    LoginPageModelClass.prototype.preventFormSubmit = /**
     * @param {?} event
     * @return {?}
     */
    function (event) {
        if (this.isFormAjax) {
            this.submitted = true;
            event.preventDefault();
        }
    };
    LoginPageModelClass.propDecorators = {
        formAction: [{ type: Input }],
        logo: [{ type: Input }],
        isFormAjax: [{ type: Input }],
        showError: [{ type: Input }],
        submit: [{ type: Output }],
        visualConfiguration: [{ type: Input }]
    };
    return LoginPageModelClass;
}());
export { LoginPageModelClass };
if (false) {
    /** @type {?} */
    LoginPageModelClass.prototype.submitted;
    /** @type {?} */
    LoginPageModelClass.prototype.formAction;
    /** @type {?} */
    LoginPageModelClass.prototype.logo;
    /** @type {?} */
    LoginPageModelClass.prototype.isFormAjax;
    /** @type {?} */
    LoginPageModelClass.prototype.showError;
    /** @type {?} */
    LoginPageModelClass.prototype.submit;
    /** @type {?} */
    LoginPageModelClass.prototype._visualConfiguration;
    /** @type {?} */
    LoginPageModelClass.prototype._DomSanitizationService;
}
