// @libriciel/ls-composants/esm5/lib/component/libersign/libersign.applet.util.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.applet.util.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { LibersignUtil } from './libersign.util';
var LibersignAppletUtil = /** @class */ (function (_super) {
    tslib_1.__extends(LibersignAppletUtil, _super);
    function LibersignAppletUtil(zone) {
        var _this = _super.call(this) || this;
        _this.zone = zone;
        return _this;
    }
    /**
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    LibersignAppletUtil.prototype.init = /**
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    function (updateUrl, appletUrl) {
        var _this = this;
        this.status.appletLoading = true;
        this.appletElement = document.getElementById('appletLibersign');
        if (this.appletElement == null) {
            /** @type {?} */
            var appletElement = document.createElement('div');
            appletElement.id = 'appletLibersign';
            document.body.appendChild(appletElement);
            this.appletElement = document.getElementById('appletLibersign');
            // @ts-ignore
            window.appletIsLoaded = (/**
             * @return {?}
             */
            function () {
                _this.zone.run((/**
                 * @return {?}
                 */
                function () {
                    _this.status.appletLoading = false;
                    _this.status.certsLoading = true;
                    _this.loadCertificates();
                }));
                _this.selectCertificate(_this.favorite);
            });
            this.appletElement.innerHTML = "\n<applet codebase=\"" + appletUrl + "\" id=\"liberSignApplet\" code=\"org/adullact/parapheur/applets/splittedsign/Main.class\" archive=\"SplittedSignatureApplet.jar\" name=\"Signature i-Parapheur\" height=\"0\" width=\"100%\">\n  <param name=\"permissions\" value=\"all-permissions\">\n  <param name=\"codebase_lookup\" value=\"true\">\n  <param name=\"display_cancel\" value=\"true\">\n  <param name=\"cancel_mode\" value=\"javascript\">\n</applet>";
        }
        else {
            this.zone.run((/**
             * @return {?}
             */
            function () {
                _this.status.appletLoading = false;
                _this.status.certsLoading = true;
                _this.loadCertificates();
            }));
            this.selectCertificate(this.favorite);
        }
    };
    /**
     * @return {?}
     */
    LibersignAppletUtil.prototype.loadCertificates = /**
     * @return {?}
     */
    function () {
        /** @type {?} */
        var appletElement = document.getElementById('liberSignApplet');
        if (appletElement != null) {
            // @ts-ignore
            /** @type {?} */
            var certStr = document.getElementById('liberSignApplet').getCertificates();
            this.certificates = JSON.parse(certStr)['certs'];
            this.commonLoadCert();
        }
    };
    /**
     * @param {?} cert
     * @return {?}
     */
    LibersignAppletUtil.prototype.selectCertificate = /**
     * @param {?} cert
     * @return {?}
     */
    function (cert) {
        var _this = this;
        this.zone.run((/**
         * @return {?}
         */
        function () {
            _this.currentCert = cert;
        }));
    };
    /**
     * @param {?} dataToSign
     * @return {?}
     */
    LibersignAppletUtil.prototype.doSign = /**
     * @param {?} dataToSign
     * @return {?}
     */
    function (dataToSign) {
        var _this = this;
        // @ts-ignore
        /** @type {?} */
        var signatureValue = document.getElementById('liberSignApplet').sign(this.currentCert.ID, dataToSign);
        this.zone.run((/**
         * @return {?}
         */
        function () {
            _this.signatures.next(JSON.parse(signatureValue));
        }));
    };
    /**
     * @return {?}
     */
    LibersignAppletUtil.prototype.destroy = /**
     * @return {?}
     */
    function () {
        // @ts-ignore
    };
    return LibersignAppletUtil;
}(LibersignUtil));
export { LibersignAppletUtil };
if (false) {
    /** @type {?} */
    LibersignAppletUtil.prototype.appletElement;
    /**
     * @type {?}
     * @private
     */
    LibersignAppletUtil.prototype.zone;
}
