// @libriciel/ls-composants/esm5/lib/component/libersign/libersign.util.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.util.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { LibersignStatus } from './libersign-status.model';
import { BehaviorSubject } from 'rxjs';
/**
 * @abstract
 */
var /**
 * @abstract
 */
LibersignUtil = /** @class */ (function () {
    function LibersignUtil() {
        this.status = new LibersignStatus();
        this.FAVORITEKEY = 'libersign-fav';
        this._signatures = [];
        this.signatures = new BehaviorSubject(this._signatures);
    }
    Object.defineProperty(LibersignUtil.prototype, "favorite", {
        get: /**
         * @return {?}
         */
        function () {
            var e_1, _a;
            if (this._favorite !== undefined) {
                return this._favorite;
            }
            else {
                /** @type {?} */
                var id = localStorage.getItem(this.FAVORITEKEY);
                try {
                    for (var _b = tslib_1.__values(this.certificates), _c = _b.next(); !_c.done; _c = _b.next()) {
                        var cert = _c.value;
                        if (cert.ID === id) {
                            this._favorite = cert;
                            return cert;
                        }
                    }
                }
                catch (e_1_1) { e_1 = { error: e_1_1 }; }
                finally {
                    try {
                        if (_c && !_c.done && (_a = _b.return)) _a.call(_b);
                    }
                    finally { if (e_1) throw e_1.error; }
                }
            }
            return undefined;
        },
        set: /**
         * @param {?} cert
         * @return {?}
         */
        function (cert) {
            localStorage.setItem(this.FAVORITEKEY, cert.ID);
            this._favorite = cert;
        },
        enumerable: true,
        configurable: true
    });
    /**
     * @param {?} dataToSign
     * @return {?}
     */
    LibersignUtil.prototype.sign = /**
     * @param {?} dataToSign
     * @return {?}
     */
    function (dataToSign) {
        this.status.dataToSignLoading = false;
        this.status.signLoading = true;
        this.doSign(dataToSign);
    };
    /**
     * @return {?}
     */
    LibersignUtil.prototype.commonLoadCert = /**
     * @return {?}
     */
    function () {
        this.status.certsLoading = false;
        this.status.certsLoaded = true;
    };
    return LibersignUtil;
}());
/**
 * @abstract
 */
export { LibersignUtil };
if (false) {
    /** @type {?} */
    LibersignUtil.prototype.status;
    /** @type {?} */
    LibersignUtil.prototype.certificates;
    /** @type {?} */
    LibersignUtil.prototype.currentCert;
    /** @type {?} */
    LibersignUtil.prototype._favorite;
    /** @type {?} */
    LibersignUtil.prototype.FAVORITEKEY;
    /** @type {?} */
    LibersignUtil.prototype.signatures;
    /** @type {?} */
    LibersignUtil.prototype._signatures;
    /**
     * @abstract
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    LibersignUtil.prototype.init = function (updateUrl, appletUrl) { };
    /**
     * @abstract
     * @return {?}
     */
    LibersignUtil.prototype.loadCertificates = function () { };
    /**
     * @abstract
     * @param {?} cert
     * @return {?}
     */
    LibersignUtil.prototype.selectCertificate = function (cert) { };
    /**
     * @abstract
     * @param {?} dataToSign
     * @return {?}
     */
    LibersignUtil.prototype.doSign = function (dataToSign) { };
    /**
     * @abstract
     * @return {?}
     */
    LibersignUtil.prototype.destroy = function () { };
}
