// @libriciel/ls-composants/esm5/lib/component/libersign/libersign-status.model.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign-status.model.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
var LibersignStatus = /** @class */ (function () {
    function LibersignStatus() {
        this.installed = true;
        this.appletLoading = false;
        this.dataToSignLoading = false;
        this.signLoading = false;
        this.certsLoading = false;
        this.certsLoaded = false;
        this.certsError = false;
    }
    /**
     * @return {?}
     */
    LibersignStatus.prototype.isLoading = /**
     * @return {?}
     */
    function () {
        return this.appletLoading || this.dataToSignLoading || this.certsLoading || this.signLoading;
    };
    /**
     * @return {?}
     */
    LibersignStatus.prototype.showCerts = /**
     * @return {?}
     */
    function () {
        return this.certsLoaded && !this.dataToSignLoading && !this.signLoading;
    };
    return LibersignStatus;
}());
export { LibersignStatus };
if (false) {
    /** @type {?} */
    LibersignStatus.prototype.installed;
    /** @type {?} */
    LibersignStatus.prototype.appletLoading;
    /** @type {?} */
    LibersignStatus.prototype.dataToSignLoading;
    /** @type {?} */
    LibersignStatus.prototype.signLoading;
    /** @type {?} */
    LibersignStatus.prototype.certsLoading;
    /** @type {?} */
    LibersignStatus.prototype.certsLoaded;
    /** @type {?} */
    LibersignStatus.prototype.certsError;
}
