// @libriciel/ls-composants/esm5/lib/component/libersign/libersign.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, EventEmitter, Input, NgZone, Output } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { LibersignNativeUtil } from './libersign.native.util';
import { LibersignAppletUtil } from './libersign.applet.util';
var LibersignComponent = /** @class */ (function () {
    function LibersignComponent(httpClient, zone, element) {
        this.httpClient = httpClient;
        this.zone = zone;
        this.element = element;
        this.sign = new EventEmitter();
        this.cancel = new EventEmitter();
        this.signed = new EventEmitter();
        // @ts-ignore
        this.isIE = false || !!document.documentMode;
        this.isValid = true;
    }
    /**
     * @return {?}
     */
    LibersignComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        var _this = this;
        if (typeof LiberSign !== 'undefined') {
            this.libersignUtil = new LibersignNativeUtil();
        }
        else if (this.isIE) {
            this.libersignUtil = new LibersignAppletUtil(this.zone);
        }
        else {
            this.libersignUtil = new LibersignAppletUtil(this.zone);
        }
        this.libersignUtil.init(this.updateUrl, this.appletUrl);
        this.libersignUtil.signatures.subscribe((/**
         * @param {?} sig
         * @return {?}
         */
        function (sig) {
            _this.signed.emit(sig);
        }));
    };
    /**
     * @return {?}
     */
    LibersignComponent.prototype.onSign = /**
     * @return {?}
     */
    function () {
        this.libersignUtil.status.dataToSignLoading = true;
        this.sign.emit(this.libersignUtil.currentCert);
    };
    /**
     * @return {?}
     */
    LibersignComponent.prototype.onCancel = /**
     * @return {?}
     */
    function () {
        this.libersignUtil.destroy();
        this.cancel.emit();
    };
    Object.defineProperty(LibersignComponent.prototype, "dataToSign", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            // tslint:disable-next-line:triple-equals
            if (value != undefined && value.length > 0) {
                if (typeof value === 'string') {
                    try {
                        value = JSON.parse(value);
                        this.libersignUtil.status.signLoading = true;
                        this.libersignUtil.sign(value);
                    }
                    catch (error) {
                        // Do nothing
                    }
                }
                else {
                    this.libersignUtil.status.signLoading = true;
                    this.libersignUtil.sign(value);
                }
            }
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(LibersignComponent.prototype, "valid", {
        set: /**
         * @param {?} value
         * @return {?}
         */
        function (value) {
            this.isValid = value === 'true';
        },
        enumerable: true,
        configurable: true
    });
    LibersignComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-libersign',
                    template: "<div class=\"libersign-container\" [ngClass]=\"libersignUtil.status.certsLoaded ? '' : 'min-height'\">\n  <span class=\"text text-danger\" *ngIf=\"!libersignUtil.status.installed\">\n    <i class=\"fa fa-times\"></i> Merci d'installer l'extension LiberSign\n  </span>\n\n  <span class=\"text text-danger\" *ngIf=\"libersignUtil.status.certsError\">\n    <i class=\"fa fa-times\"></i> Erreur lors de la r\u00E9cup\u00E9ration des certificats\n  </span>\n\n  <div *ngIf=\"libersignUtil.status.isLoading()\" class=\"center\">\n    <div class=\"lds-roller\">\n      <div></div>\n      <div></div>\n      <div></div>\n      <div></div>\n      <div></div>\n      <div></div>\n      <div></div>\n      <div></div>\n    </div>\n\n    <div *ngIf=\"libersignUtil.status.certsLoading\" class=\"text text-info\">\n      Chargement des certificats disponibles en cours...\n    </div>\n\n    <div *ngIf=\"libersignUtil.status.appletLoading\" class=\"text text-info\">\n      Chargement de l'applet de signature en cours...\n    </div>\n\n    <div *ngIf=\"libersignUtil.status.dataToSignLoading\" class=\"text text-info\">\n      R\u00E9cup\u00E9ration des informations de signature en cours...\n    </div>\n\n    <div *ngIf=\"libersignUtil.status.signLoading\" class=\"text text-info\">\n      Signature en cours...\n    </div>\n  </div>\n\n  <div *ngIf=\"libersignUtil.status.showCerts()\" class=\"certs-loaded\">\n    <table class=\"table\">\n      <thead>\n      <tr>\n        <th></th>\n        <th>Nom</th>\n        <th>\u00C9metteur</th>\n        <th>Date d'expiration</th>\n      </tr>\n      </thead>\n\n      <tbody>\n      <tr *ngFor=\"let cert of libersignUtil.certificates\" (click)=\"libersignUtil.selectCertificate(cert)\" [class.active]=\"libersignUtil.currentCert && libersignUtil.currentCert.ID == cert.ID\">\n        <td>\n          <i class=\"fa-star text text-warning\" [ngClass]=\"libersignUtil.favorite === cert ? 'fa' : 'far fa fa-star-o'\" title=\"Certificat favori\" (click)=\"libersignUtil.favorite = cert\"></i>\n          <span [className]=\"'label label-' + cert.verifiedWith[0]\" *ngIf=\"cert.verifiedWith.length > 0\">{{cert.verifiedWith[0]}}</span>\n        </td>\n        <td>{{cert.CN}}</td>\n        <td>{{cert.ISSUERDN}}</td>\n        <td>{{cert.NOTAFTER | date:'dd/MM/yyyy'}}</td>\n      </tr>\n      </tbody>\n    </table>\n\n    <div class=\"float-right libersign-action-buttons\">\n      <button class=\"btn btn-warning\" (click)=\"onCancel()\">\n        <i class=\"fa fa-times-circle-o\"></i>\n        Annuler\n      </button>\n\n      <button class=\"btn btn-primary\" (click)=\"onSign()\" [disabled]=\"!libersignUtil.currentCert || !isValid\">\n        <i class=\"fa ls-signature\"></i>\n        Signer\n      </button>\n    </div>\n  </div>\n</div>\n\n\n",
                    styles: [".libersign-container{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.libersign-container.min-height{min-height:150px}.fa-star{margin-right:5px}.label-RGS{background-color:var(--ls-success-400);background-color:#5cb85c}.label-ADULLACT{background-color:var(--ls-info-400);background-color:#5bc0de}.label-LIBRICIEL{background-color:var(--ls-primary-500);background-color:#428bca}.label-EIDAS{background-color:var(--ls-grey-500);background-color:#999}.label-ETAT{background-color:var(--ls-warning-400);background-color:#f0ad4e}.certs-loaded{width:100%}table{text-align:left}.table td,.table th{padding:.3rem}tr{cursor:pointer}tr.active{background-color:var(--ls-primary-200);background-color:#b1dfff}tr.active>td{background-color:transparent!important}.center{text-align:center}#appletLibersign{height:0}.libersign-action-buttons>*{margin-left:1rem}.lds-roller{display:inline-block;position:relative;width:80px;height:80px}.lds-roller div{-webkit-animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-roller;animation:1.2s cubic-bezier(.5,0,.5,1) infinite lds-roller;-webkit-transform-origin:40px 40px;-ms-transform-origin:40px 40px;transform-origin:40px 40px}.lds-roller div:after{content:\" \";display:block;position:absolute;width:7px;height:7px;border-radius:50%;background:#000;margin:-4px 0 0 -4px}.lds-roller div:nth-child(1){-webkit-animation-delay:-36ms;animation-delay:-36ms}.lds-roller div:nth-child(1):after{top:63px;left:63px}.lds-roller div:nth-child(2){-webkit-animation-delay:-72ms;animation-delay:-72ms}.lds-roller div:nth-child(2):after{top:68px;left:56px}.lds-roller div:nth-child(3){-webkit-animation-delay:-108ms;animation-delay:-108ms}.lds-roller div:nth-child(3):after{top:71px;left:48px}.lds-roller div:nth-child(4){-webkit-animation-delay:-144ms;animation-delay:-144ms}.lds-roller div:nth-child(4):after{top:72px;left:40px}.lds-roller div:nth-child(5){-webkit-animation-delay:-.18s;animation-delay:-.18s}.lds-roller div:nth-child(5):after{top:71px;left:32px}.lds-roller div:nth-child(6){-webkit-animation-delay:-216ms;animation-delay:-216ms}.lds-roller div:nth-child(6):after{top:68px;left:24px}.lds-roller div:nth-child(7){-webkit-animation-delay:-252ms;animation-delay:-252ms}.lds-roller div:nth-child(7):after{top:63px;left:17px}.lds-roller div:nth-child(8){-webkit-animation-delay:-288ms;animation-delay:-288ms}.lds-roller div:nth-child(8):after{top:56px;left:12px}@-webkit-keyframes lds-roller{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}@keyframes lds-roller{0%{-webkit-transform:rotate(0);transform:rotate(0)}100%{-webkit-transform:rotate(360deg);transform:rotate(360deg)}}"]
                }] }
    ];
    /** @nocollapse */
    LibersignComponent.ctorParameters = function () { return [
        { type: HttpClient },
        { type: NgZone },
        { type: ElementRef }
    ]; };
    LibersignComponent.propDecorators = {
        updateUrl: [{ type: Input }],
        appletUrl: [{ type: Input }],
        sign: [{ type: Output }],
        cancel: [{ type: Output }],
        signed: [{ type: Output }],
        dataToSign: [{ type: Input }],
        valid: [{ type: Input }]
    };
    return LibersignComponent;
}());
export { LibersignComponent };
if (false) {
    /** @type {?} */
    LibersignComponent.prototype.updateUrl;
    /** @type {?} */
    LibersignComponent.prototype.appletUrl;
    /** @type {?} */
    LibersignComponent.prototype.sign;
    /** @type {?} */
    LibersignComponent.prototype.cancel;
    /** @type {?} */
    LibersignComponent.prototype.signed;
    /** @type {?} */
    LibersignComponent.prototype.isIE;
    /** @type {?} */
    LibersignComponent.prototype.isValid;
    /** @type {?} */
    LibersignComponent.prototype.libersignUtil;
    /**
     * @type {?}
     * @private
     */
    LibersignComponent.prototype.httpClient;
    /**
     * @type {?}
     * @private
     */
    LibersignComponent.prototype.zone;
    /**
     * @type {?}
     * @private
     */
    LibersignComponent.prototype.element;
}
