// @libriciel/ls-composants/esm5/lib/component/libersign/libersign.native.util.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/libersign/libersign.native.util.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { LibersignUtil } from './libersign.util';
var LibersignNativeUtil = /** @class */ (function (_super) {
    tslib_1.__extends(LibersignNativeUtil, _super);
    function LibersignNativeUtil() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    /**
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    LibersignNativeUtil.prototype.init = /**
     * @param {?} updateUrl
     * @param {?} appletUrl
     * @return {?}
     */
    function (updateUrl, appletUrl) {
        this.status.certsLoading = true;
        try {
            LiberSign.setUpdateUrl(updateUrl);
        }
        catch (error) {
            console.log('error catched');
        }
        this.loadCertificates();
    };
    /**
     * @return {?}
     */
    LibersignNativeUtil.prototype.loadCertificates = /**
     * @return {?}
     */
    function () {
        var _this = this;
        LiberSign.getCertificates().then((/**
         * @param {?} cs
         * @return {?}
         */
        function (cs) {
            if (cs !== undefined) {
                _this.certificates = cs;
                _this.selectCertificate(_this.favorite);
                _this.commonLoadCert();
            }
        }), (/**
         * @param {?} err
         * @return {?}
         */
        function (err) {
            _this.status.certsError = true;
        }));
    };
    /**
     * @param {?} dataToSign
     * @return {?}
     */
    LibersignNativeUtil.prototype.doSign = /**
     * @param {?} dataToSign
     * @return {?}
     */
    function (dataToSign) {
        var _this = this;
        LiberSign.sign(this.currentCert.ID, { dataToSign: dataToSign }).then((/**
         * @param {?} sig
         * @return {?}
         */
        function (sig) {
            _this.signatures.next(sig);
        }));
    };
    /**
     * @param {?} cert
     * @return {?}
     */
    LibersignNativeUtil.prototype.selectCertificate = /**
     * @param {?} cert
     * @return {?}
     */
    function (cert) {
        this.currentCert = cert;
    };
    /**
     * @return {?}
     */
    LibersignNativeUtil.prototype.destroy = /**
     * @return {?}
     */
    function () {
        // Nothing to do here
    };
    return LibersignNativeUtil;
}(LibersignUtil));
export { LibersignNativeUtil };
