// @libriciel/ls-composants/esm5/lib/component/back-to-top/back-to-top.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/back-to-top/back-to-top.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, HostListener } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
var BackToTopComponent = /** @class */ (function () {
    function BackToTopComponent() {
        this.showBackToTop = true;
        this.offset = 200;
    }
    /**
     * @return {?}
     */
    BackToTopComponent.prototype.checkScroll = /**
     * @return {?}
     */
    function () {
        this.showBackToTop = window.pageYOffset > this.offset;
    };
    /**
     * @return {?}
     */
    BackToTopComponent.prototype.ngOnInit = /**
     * @return {?}
     */
    function () {
        this.checkScroll();
    };
    /**
     * @return {?}
     */
    BackToTopComponent.prototype.backToTop = /**
     * @return {?}
     */
    function () {
        window.scrollTo({ left: 0, top: 0, behavior: 'smooth' });
    };
    BackToTopComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-back-to-top',
                    template: "<div class=\"backtotop\" *ngIf=\"showBackToTop\" [@fadeInOut]>\n  <a  (click)=\"backToTop()\" class=\"top-link\"><i class=\"fa fa-chevron-up\"></i></a>\n</div>\n",
                    animations: [
                        trigger('fadeInOut', [
                            // ...
                            state('void', style({
                                opacity: 0
                            })),
                            transition('void <=> *', animate(300)),
                        ]),
                    ],
                    styles: ["div.backtotop{position:fixed;z-index:9999;bottom:45px;right:20px;cursor:pointer}div.backtotop .top-link{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-pack:center;-ms-flex-pack:center;justify-content:center;-webkit-box-align:center;-ms-flex-align:center;align-items:center;text-decoration:none;text-align:center;color:#fff;font-size:2em;width:50px;height:50px;line-height:50px;border-radius:5px;background:#000;opacity:.7}"]
                }] }
    ];
    BackToTopComponent.propDecorators = {
        checkScroll: [{ type: HostListener, args: ['window:scroll',] }]
    };
    return BackToTopComponent;
}());
export { BackToTopComponent };
if (false) {
    /** @type {?} */
    BackToTopComponent.prototype.showBackToTop;
    /** @type {?} */
    BackToTopComponent.prototype.offset;
}
