// @libriciel/ls-composants/esm2015/lib/component/ui-components.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/ui-components.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule, NO_ERRORS_SCHEMA } from '@angular/core';
import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
import { FormsModule, ReactiveFormsModule } from '@angular/forms';
import { BackToTopComponent } from './back-to-top/back-to-top.component';
import { ColorPickerComponent } from './color-picker/color-picker.component';
import { MaxLengthTextAreaComponent } from './max-length-textarea/max-length-textarea.component';
import { PasswordComponent } from './password/password.component';
import { TimepickerComponent } from './timepicker/timepicker.component';
import { CommonModule } from '@angular/common';
import { BrowserModule } from '@angular/platform-browser';
import { LibersignComponent } from './libersign/libersign.component';
import { HttpClientModule } from '@angular/common/http';
import { ButtonComponent } from './buttons/button/button.component';
import { CancelButtonComponent } from './buttons/cancel-button/cancel-button.component';
import { GoBackButtonComponent } from './buttons/go-back-button/go-back-button.component';
import { SaveButtonComponent } from './buttons/save-button/save-button.component';
import { ValidateButtonComponent } from './buttons/validate-button/validate-button.component';
import { ActionBarComponent } from './action-bar/action-bar.component';
import { PipeModule } from '../pipe/pipe.module';
import { DirectiveModule } from '../directive/directive.module';
export class LsUiComponentsModule {
    constructor() {
    }
}
// This is for Components with contents
LsUiComponentsModule.angularComponents = {
    // components
    'ls-lib-back-top-top': BackToTopComponent,
    'ls-lib-max-length-textarea': MaxLengthTextAreaComponent,
    'ls-lib-color-picker': ColorPickerComponent,
    'ls-lib-timepicker': TimepickerComponent,
    'ls-lib-password': PasswordComponent,
    'ls-lib-libersign': LibersignComponent,
    'ls-lib-button': ButtonComponent,
    'ls-lib-button-cancel': CancelButtonComponent,
    'ls-lib-button-go-back': GoBackButtonComponent,
    'ls-lib-button-save': SaveButtonComponent,
    'ls-lib-button-validate': ValidateButtonComponent,
    'ls-lib-action-bar': ActionBarComponent
};
LsUiComponentsModule.decorators = [
    { type: NgModule, args: [{
                schemas: [NO_ERRORS_SCHEMA],
                entryComponents: [
                    BackToTopComponent,
                    ColorPickerComponent,
                    MaxLengthTextAreaComponent,
                    PasswordComponent,
                    TimepickerComponent,
                    LibersignComponent,
                    ButtonComponent,
                    CancelButtonComponent,
                    GoBackButtonComponent,
                    SaveButtonComponent,
                    ValidateButtonComponent,
                    ActionBarComponent
                ],
                declarations: [
                    BackToTopComponent,
                    ColorPickerComponent,
                    MaxLengthTextAreaComponent,
                    PasswordComponent,
                    TimepickerComponent,
                    LibersignComponent,
                    ButtonComponent,
                    CancelButtonComponent,
                    GoBackButtonComponent,
                    SaveButtonComponent,
                    ValidateButtonComponent,
                    ActionBarComponent
                ],
                imports: [
                    BrowserModule,
                    CommonModule,
                    BrowserAnimationsModule,
                    FormsModule,
                    ReactiveFormsModule,
                    HttpClientModule,
                    PipeModule,
                    DirectiveModule
                ],
                exports: [
                    BackToTopComponent,
                    ColorPickerComponent,
                    MaxLengthTextAreaComponent,
                    PasswordComponent,
                    TimepickerComponent,
                    LibersignComponent,
                    ButtonComponent,
                    CancelButtonComponent,
                    GoBackButtonComponent,
                    SaveButtonComponent,
                    ValidateButtonComponent,
                    ActionBarComponent
                ]
            },] }
];
/** @nocollapse */
LsUiComponentsModule.ctorParameters = () => [];
if (false) {
    /** @type {?} */
    LsUiComponentsModule.angularComponents;
}
