// @libriciel/ls-composants/esm2015/lib/pages/login/component/forgot-password/forgot-password.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/component/forgot-password/forgot-password.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
import { CommonIcons } from '../../../../icons/common-icons';
import { CommonMessages } from '../../../../i18n/common-messages';
import { LoginMessages } from '../../i18n/login-messages';
export class ForgotPasswordComponent extends LoginPageModelClass {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        super(_DomSanitizationService);
        this._DomSanitizationService = _DomSanitizationService;
        this.commonIcons = CommonIcons;
        this.commonMessages = CommonMessages;
        this.loginMessages = LoginMessages;
        this.oneFieldForgot = false;
        this.usernameormailInputName = 'usernameormail';
        this.usernameInputName = 'username';
        this.emailInputName = 'email';
        this.backToLogin = new EventEmitter();
    }
    /**
     * @return {?}
     */
    backToLoginPage() {
        this.backToLogin.emit();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    preventFormSubmit(event) {
        super.preventFormSubmit(event);
    }
    /**
     * @return {?}
     */
    submitForm() {
        this.submit.emit({
            username: this.username,
            email: this.email,
            usernameormail: this.usernameormail
        });
    }
}
ForgotPasswordComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-forgot-password',
                template: "<ls-login-model [logo]=\"logo\" [visualConfiguration]=\"_visualConfiguration\">\n  <section id=\"forgot-page\">\n    <h1 class=\"page-title\">Mot de passe oubli\u00E9 ?</h1>\n    <form [action]=\"formAction\" id=\"ls-forgot-form\" method=\"post\" ngNoForm (submit)=\"preventFormSubmit($event)\">\n\n      <div *ngIf=\"!oneFieldForgot\" class=\"usernameormail\">\n        <label [attr.for]=\"usernameInputName\">\n          Identifiant\n        </label>\n        <input [attr.name]=\"usernameInputName\" [attr.id]=\"usernameInputName\" [(ngModel)]=\"username\" class=\"form-control\"\n               placeholder=\"Mon identifiant\"\n               type=\"text\" required>\n      </div>\n      <div *ngIf=\"!oneFieldForgot\" class=\"usernameormail\">\n        <label [attr.for]=\"emailInputName\">\n          Adresse mail\n        </label>\n        <input [attr.id]=\"emailInputName\" [attr.name]=\"emailInputName\" [(ngModel)]=\"email\" class=\"form-control\"\n               placeholder=\"adresse@mail.com\" type=\"text\" required>\n      </div>\n\n\n      <div *ngIf=\"oneFieldForgot\" class=\"usernameormail\">\n        <label [attr.for]=\"usernameormailInputName\">\n          Saisissez votre nom d'utilisateur ou votre adresse mail\n        </label>\n        <input [attr.name]=\"usernameormailInputName\" [attr.id]=\"usernameormailInputName\" [(ngModel)]=\"usernameormail\"\n               class=\"form-control\"\n               placeholder=\"Nom d'utilisateur ou adresse mail\"\n               type=\"text\" required>\n      </div>\n\n      <div class=\"alert alert-info\">\n        <div class=\"fa fa-lg fa-info-circle\"></div>\n        <div>\n          Si vos informations sont correctes, un e-mail vous sera envoy\u00E9.<br>\n          Vous pourrez ainsi cr\u00E9er un nouveau mot de passe.\n        </div>\n      </div>\n\n      <ng-content></ng-content>\n\n      <div class=\"float-left\">\n        <ls-button-go-back (click)=\"backToLoginPage()\" [title]=\"loginMessages.BACK_TO_CONNEXION_PAGE\"></ls-button-go-back>\n      </div>\n\n      <div class=\"float-right\">\n        <ls-button (click)=\"submitForm()\" [icon]=\"commonIcons.SUBMIT_ICON\"\n                   [processing]=\"submitted\"\n                   [title]=\"commonMessages.SUBMIT\"\n                   form=\"ls-forgot-form\"\n                   type=\"submit\">\n        </ls-button>\n      </div>\n    </form>\n  </section>\n</ls-login-model>\n",
                styles: [".alert{-ms-flex-align:center;-webkit-box-align:center;align-items:center;display:-webkit-box;display:-ms-flexbox;display:flex}.alert .fa{width:2rem}#ls-forgot-form>*{margin-bottom:1rem}span.btn{cursor:pointer}.page-title{font-size:1rem;font-weight:400;padding-bottom:1.5rem;padding-top:3rem;text-align:center}"]
            }] }
];
/** @nocollapse */
ForgotPasswordComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
ForgotPasswordComponent.propDecorators = {
    oneFieldForgot: [{ type: Input }],
    usernameormailInputName: [{ type: Input }],
    usernameInputName: [{ type: Input }],
    emailInputName: [{ type: Input }],
    backToLogin: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    ForgotPasswordComponent.prototype.username;
    /** @type {?} */
    ForgotPasswordComponent.prototype.email;
    /** @type {?} */
    ForgotPasswordComponent.prototype.usernameormail;
    /** @type {?} */
    ForgotPasswordComponent.prototype.commonIcons;
    /** @type {?} */
    ForgotPasswordComponent.prototype.commonMessages;
    /** @type {?} */
    ForgotPasswordComponent.prototype.loginMessages;
    /** @type {?} */
    ForgotPasswordComponent.prototype.oneFieldForgot;
    /** @type {?} */
    ForgotPasswordComponent.prototype.usernameormailInputName;
    /** @type {?} */
    ForgotPasswordComponent.prototype.usernameInputName;
    /** @type {?} */
    ForgotPasswordComponent.prototype.emailInputName;
    /** @type {?} */
    ForgotPasswordComponent.prototype.backToLogin;
    /** @type {?} */
    ForgotPasswordComponent.prototype._DomSanitizationService;
}
