// @libriciel/ls-composants/esm2015/lib/component/max-length-textarea/max-length-textarea.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/max-length-textarea/max-length-textarea.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, ElementRef, forwardRef, Input, Renderer2, ViewChild } from '@angular/core';
import { NG_VALUE_ACCESSOR } from '@angular/forms';
import { MaxLengthTextAreaMessages } from './MaxLengthTextAreaMessages';
export class MaxLengthTextAreaComponent {
    /**
     * @param {?} element
     * @param {?} renderer2
     */
    constructor(element, renderer2) {
        this.element = element;
        this.renderer2 = renderer2;
        this.maxLengthTextAreaMessages = MaxLengthTextAreaMessages;
        this.onChange = (/**
         * @return {?}
         */
        () => {
            // tslint:disable-next-line:semicolon
        });
        this.onTouched = (/**
         * @return {?}
         */
        () => {
            // tslint:disable-next-line:semicolon
        });
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        /** @type {?} */
        const id = this.element.nativeElement.getAttribute('id');
        this.renderer2.removeAttribute(this.element.nativeElement, 'id');
        this.renderer2.setAttribute(this.input.nativeElement, 'id', id);
    }
    /**
     * @return {?}
     */
    get value() {
        return this._value;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set value(value) {
        this._value = value;
        this.onChange(value);
        this.onTouched();
    }
    /**
     * @return {?}
     */
    onChangeSelection() {
        this.onChange(this._value);
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnChange(fn) {
        this.onChange = fn;
    }
    /**
     * @param {?} fn
     * @return {?}
     */
    registerOnTouched(fn) {
        this.onTouched = fn;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    writeValue(value) {
        if (value) {
            this.value = value;
        }
    }
}
MaxLengthTextAreaComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-max-length-textarea',
                template: "<!--suppress HtmlFormInputWithoutLabel -->\n<textarea class=\"form-control\" #input\n  (ngModelChange)=\"onChangeSelection()\"\n  [(ngModel)]=\"value\" [required]=\"required\"\n  maxlength=\"{{maxlength}}\" [attr.name]=\"name\"></textarea>\n<span [innerText]=\"maxLengthTextAreaMessages.max_left(!value ? maxlength : maxlength - value.length)\"\n      class=\"information\"></span>\n\n",
                providers: [
                    {
                        provide: NG_VALUE_ACCESSOR,
                        useExisting: forwardRef((/**
                         * @return {?}
                         */
                        () => MaxLengthTextAreaComponent)),
                        multi: true
                    }
                ],
                styles: [":host{position:relative;display:block}:host textarea{min-height:3.5rem}:host span.information{color:var(--ls-grey-500);height:1.25rem;border-radius:0 0 4px 4px;font-size:.8rem;left:1px;right:1px;bottom:-20px;padding-right:6px;text-align:right;position:absolute;pointer-events:none}"]
            }] }
];
/** @nocollapse */
MaxLengthTextAreaComponent.ctorParameters = () => [
    { type: ElementRef },
    { type: Renderer2 }
];
MaxLengthTextAreaComponent.propDecorators = {
    maxlength: [{ type: Input }],
    minlength: [{ type: Input }],
    required: [{ type: Input }],
    _value: [{ type: Input }],
    name: [{ type: Input }],
    input: [{ type: ViewChild, args: ['input',] }]
};
if (false) {
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.maxlength;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.minlength;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.required;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.maxLengthTextAreaMessages;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype._value;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.name;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.input;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.onChange;
    /** @type {?} */
    MaxLengthTextAreaComponent.prototype.onTouched;
    /**
     * @type {?}
     * @private
     */
    MaxLengthTextAreaComponent.prototype.element;
    /**
     * @type {?}
     * @private
     */
    MaxLengthTextAreaComponent.prototype.renderer2;
}
