// @libriciel/ls-composants/esm5/workflows/shared/model/workflows/step-instance.js
/**
 * @fileoverview added by tsickle
 * Generated from: shared/model/workflows/step-instance.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
var StepInstance = /** @class */ (function () {
    function StepInstance(stepInstData) {
        this.validatedBy = [];
        this.id = stepInstData.id;
        this.name = stepInstData.name;
        this.validators = stepInstData.validators ? tslib_1.__spread(stepInstData.validators) : [];
        this.state = stepInstData.state;
        this.validatedBy = stepInstData.actedUponBy ? [stepInstData.actedUponBy] : [];
        this.comment = stepInstData.comment;
        if (stepInstData.actionId) {
            this.id = stepInstData.actionId;
        }
    }
    return StepInstance;
}());
export { StepInstance };
if (false) {
    /** @type {?} */
    StepInstance.prototype.id;
    /** @type {?} */
    StepInstance.prototype.name;
    /** @type {?} */
    StepInstance.prototype.type;
    /** @type {?} */
    StepInstance.prototype.validators;
    /** @type {?} */
    StepInstance.prototype.state;
    /** @type {?} */
    StepInstance.prototype.validatedBy;
    /** @type {?} */
    StepInstance.prototype.comment;
}
