// @libriciel/ls-composants/esm5/lib/pages/login/forgot-password-success/forgot-password-success.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/forgot-password-success/forgot-password-success.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as tslib_1 from "tslib";
import { Component, EventEmitter, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
var ForgotPasswordSuccessComponent = /** @class */ (function (_super) {
    tslib_1.__extends(ForgotPasswordSuccessComponent, _super);
    function ForgotPasswordSuccessComponent(_DomSanitizationService) {
        var _this = _super.call(this, _DomSanitizationService) || this;
        _this._DomSanitizationService = _DomSanitizationService;
        _this.backToLogin = new EventEmitter();
        return _this;
    }
    /**
     * @return {?}
     */
    ForgotPasswordSuccessComponent.prototype.backToLoginPage = /**
     * @return {?}
     */
    function () {
        this.backToLogin.emit();
    };
    ForgotPasswordSuccessComponent.decorators = [
        { type: Component, args: [{
                    selector: 'ls-forgot-password-success',
                    template: "<ls-login-model [logo]=\"logo\" [visualConfiguration]=\"_visualConfiguration\">\n  <section class=\"forgot-success-page\">\n\n    <div class=\"alert alert-success\">\n      <i class=\"fa fa-lg fa-check\"></i>\n      Si vos informations sont correctes, un e-mail vous a \u00E9t\u00E9 envoy\u00E9.\n    </div>\n    <button (click)=\"backToLoginPage()\" class=\"btn btn-primary\">Retourner \u00E0 la page de connexion</button>\n  </section>\n</ls-login-model>\n",
                    styles: [".alert{margin-top:2rem;padding:1.5rem;text-align:center;width:600px}.forgot-success-page{display:-webkit-box;display:-ms-flexbox;display:flex;-webkit-box-orient:vertical;-webkit-box-direction:normal;-ms-flex-direction:column;flex-direction:column;-webkit-box-align:center;-ms-flex-align:center;align-items:center}.forgot-success-page>*{margin-bottom:2rem}"]
                }] }
    ];
    /** @nocollapse */
    ForgotPasswordSuccessComponent.ctorParameters = function () { return [
        { type: DomSanitizer }
    ]; };
    ForgotPasswordSuccessComponent.propDecorators = {
        backToLogin: [{ type: Output }]
    };
    return ForgotPasswordSuccessComponent;
}(LoginPageModelClass));
export { ForgotPasswordSuccessComponent };
if (false) {
    /** @type {?} */
    ForgotPasswordSuccessComponent.prototype.backToLogin;
    /** @type {?} */
    ForgotPasswordSuccessComponent.prototype._DomSanitizationService;
}
