// @libriciel/ls-composants/esm2015/lib/component/back-to-top/back-to-top.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/component/back-to-top/back-to-top.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, HostListener } from '@angular/core';
import { trigger, state, style, transition, animate } from '@angular/animations';
export class BackToTopComponent {
    constructor() {
        this.showBackToTop = true;
        this.offset = 200;
    }
    /**
     * @return {?}
     */
    checkScroll() {
        this.showBackToTop = window.pageYOffset > this.offset;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.checkScroll();
    }
    /**
     * @return {?}
     */
    backToTop() {
        window.scrollTo({ left: 0, top: 0, behavior: 'smooth' });
    }
}
BackToTopComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-back-to-top',
                template: "<div class=\"backtotop\" *ngIf=\"showBackToTop\" [@fadeInOut]>\n  <a  (click)=\"backToTop()\" class=\"top-link\"><i class=\"fa fa-chevron-up\"></i></a>\n</div>\n",
                animations: [
                    trigger('fadeInOut', [
                        // ...
                        state('void', style({
                            opacity: 0
                        })),
                        transition('void <=> *', animate(300)),
                    ]),
                ],
                styles: ["div.backtotop{bottom:45px;cursor:pointer;position:fixed;right:20px;z-index:9999}div.backtotop .top-link{-ms-flex-align:center;-ms-flex-pack:center;-webkit-box-align:center;-webkit-box-pack:center;align-items:center;background:#000;border-radius:5px;color:#fff;display:-webkit-box;display:-ms-flexbox;display:flex;font-size:2em;height:50px;justify-content:center;line-height:50px;opacity:.7;text-align:center;text-decoration:none;width:50px}"]
            }] }
];
BackToTopComponent.propDecorators = {
    checkScroll: [{ type: HostListener, args: ['window:scroll',] }]
};
if (false) {
    /** @type {?} */
    BackToTopComponent.prototype.showBackToTop;
    /** @type {?} */
    BackToTopComponent.prototype.offset;
}
