// @libriciel/ls-composants/esm2015/lib/pages/login/component/login/login.component.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/pages/login/component/login/login.component.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { Component, EventEmitter, Input, Output } from '@angular/core';
import { LoginPageModelClass } from '../login-model/LoginPageModel.class';
import { DomSanitizer } from '@angular/platform-browser';
export class LoginComponent extends LoginPageModelClass {
    /**
     * @param {?} _DomSanitizationService
     */
    constructor(_DomSanitizationService) {
        super(_DomSanitizationService);
        this._DomSanitizationService = _DomSanitizationService;
        this.testMode = false;
        this.currentPage = 'login';
        this.loginFormAction = '';
        this.loginShowError = false;
        this.loginSuccessMessage = '';
        this.loginIsFormAjax = false;
        this.loginPasswordReset = false;
        this.loginShowRememberMe = false;
        this.loginUsernameInputName = 'username';
        this.loginPasswordInputName = 'password';
        this.loginRememberMeInputName = 'rememberme';
        this.forgotFormAction = '';
        this.forgotIsFormAjax = false;
        this.forgotOneFieldForgot = false;
        this.forgotUsernameInputName = 'username';
        this.forgotEmailInputName = 'email';
        this.forgotUsernameormailInputName = 'usernameormail';
        this.resetFormAction = '';
        this.resetIsFormAjax = false;
        this.resetPasswordInputName = 'password';
        this.resetPasswordConfirmInputName = 'passwordconfirm';
        this.loginSubmit = new EventEmitter();
        this.forgotSubmit = new EventEmitter();
        this.resetSubmit = new EventEmitter();
    }
    /**
     * @param {?} event
     * @return {?}
     */
    login(event) {
        this.loginSubmit.emit(event);
    }
    /**
     * @param {?} event
     * @return {?}
     */
    forgot(event) {
        this.forgotSubmit.emit(event);
    }
    /**
     * @param {?} event
     * @return {?}
     */
    reset(event) {
        this.resetSubmit.emit(event);
    }
}
LoginComponent.decorators = [
    { type: Component, args: [{
                selector: 'ls-login',
                template: "<div *ngIf=\"testMode\" class=\"testmode-container\">\n  <span>Mode test</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'login'\">Page de login</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'forgot'\">Mot de passe oubli\u00E9</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'forgot-success'\">Mot de passe oubli\u00E9 demand\u00E9</span>\n  <span class=\"btn btn-primary\" (click)=\"currentPage = 'reset'\">Remise \u00E0 z\u00E9ro</span>\n</div>\n\n<ls-login-form\n  *ngIf=\"currentPage === 'login'\"\n  [visualConfiguration]=\"_visualConfiguration\"\n  [logo]=\"logo\"\n\n  [formAction]=\"loginFormAction\"\n  [showError]=\"loginShowError\"\n  [successMessage]=\"loginSuccessMessage\"\n  [passwordReset]=\"loginPasswordReset\"\n  [isFormAjax]=\"loginIsFormAjax\"\n  [showRememberMe]=\"loginShowRememberMe\"\n  [passwordInputName]=\"loginPasswordInputName\"\n  [usernameInputName]=\"loginUsernameInputName\"\n  [rememberMeInputName]=\"loginRememberMeInputName\"\n\n  (forgotPassword)=\"currentPage = 'forgot'\"\n  (submit)=\"login($event)\">\n  <ng-content select=\".login-form-addons\"></ng-content>\n</ls-login-form>\n\n<ls-forgot-password\n  *ngIf=\"currentPage === 'forgot'\"\n  [visualConfiguration]=\"_visualConfiguration\"\n  [logo]=\"logo\"\n\n  [formAction]=\"forgotFormAction\"\n  [isFormAjax]=\"forgotIsFormAjax\"\n  [oneFieldForgot]=\"forgotOneFieldForgot\"\n  [usernameInputName]=\"forgotUsernameInputName\"\n  [emailInputName]=\"forgotEmailInputName\"\n  [usernameormailInputName]=\"forgotUsernameormailInputName\"\n\n  (backToLogin)=\"currentPage = 'login'\"\n  (submit)=\"forgot($event)\">\n  <ng-content select=\".forgot-form-addons\"></ng-content>\n</ls-forgot-password>\n\n<ls-forgot-password-success\n  *ngIf=\"currentPage === 'forgot-success'\"\n  [logo]=\"logo\"\n  [visualConfiguration]=\"_visualConfiguration\"\n\n  (backToLogin)=\"currentPage = 'login'\">\n</ls-forgot-password-success>\n\n<ls-reset-password\n  *ngIf=\"currentPage === 'reset'\"\n  [logo]=\"logo\"\n  [visualConfiguration]=\"_visualConfiguration\"\n\n  [formAction]=\"resetFormAction\"\n  [isFormAjax]=\"resetIsFormAjax\"\n  [passwordInputName]=\"resetPasswordInputName\"\n  [passwordConfirmInputName]=\"resetPasswordConfirmInputName\"\n\n  (submit)=\"reset($event)\">\n  <ng-content select=\".reset-form-addons\"></ng-content>\n</ls-reset-password>\n",
                styles: [".testmode-container{-ms-flex-direction:column;-webkit-box-direction:normal;-webkit-box-orient:vertical;display:-webkit-box;display:-ms-flexbox;display:flex;flex-direction:column;left:5px;position:absolute;top:5px;z-index:1;zoom:.7}.testmode-container *{cursor:pointer;margin-bottom:5px}"]
            }] }
];
/** @nocollapse */
LoginComponent.ctorParameters = () => [
    { type: DomSanitizer }
];
LoginComponent.propDecorators = {
    testMode: [{ type: Input }],
    currentPage: [{ type: Input }],
    loginFormAction: [{ type: Input }],
    loginShowError: [{ type: Input }],
    loginSuccessMessage: [{ type: Input }],
    loginIsFormAjax: [{ type: Input }],
    loginPasswordReset: [{ type: Input }],
    loginShowRememberMe: [{ type: Input }],
    loginUsernameInputName: [{ type: Input }],
    loginPasswordInputName: [{ type: Input }],
    loginRememberMeInputName: [{ type: Input }],
    forgotFormAction: [{ type: Input }],
    forgotIsFormAjax: [{ type: Input }],
    forgotOneFieldForgot: [{ type: Input }],
    forgotUsernameInputName: [{ type: Input }],
    forgotEmailInputName: [{ type: Input }],
    forgotUsernameormailInputName: [{ type: Input }],
    resetFormAction: [{ type: Input }],
    resetIsFormAjax: [{ type: Input }],
    resetPasswordInputName: [{ type: Input }],
    resetPasswordConfirmInputName: [{ type: Input }],
    loginSubmit: [{ type: Output }],
    forgotSubmit: [{ type: Output }],
    resetSubmit: [{ type: Output }]
};
if (false) {
    /** @type {?} */
    LoginComponent.prototype.testMode;
    /** @type {?} */
    LoginComponent.prototype.currentPage;
    /** @type {?} */
    LoginComponent.prototype.loginFormAction;
    /** @type {?} */
    LoginComponent.prototype.loginShowError;
    /** @type {?} */
    LoginComponent.prototype.loginSuccessMessage;
    /** @type {?} */
    LoginComponent.prototype.loginIsFormAjax;
    /** @type {?} */
    LoginComponent.prototype.loginPasswordReset;
    /** @type {?} */
    LoginComponent.prototype.loginShowRememberMe;
    /** @type {?} */
    LoginComponent.prototype.loginUsernameInputName;
    /** @type {?} */
    LoginComponent.prototype.loginPasswordInputName;
    /** @type {?} */
    LoginComponent.prototype.loginRememberMeInputName;
    /** @type {?} */
    LoginComponent.prototype.forgotFormAction;
    /** @type {?} */
    LoginComponent.prototype.forgotIsFormAjax;
    /** @type {?} */
    LoginComponent.prototype.forgotOneFieldForgot;
    /** @type {?} */
    LoginComponent.prototype.forgotUsernameInputName;
    /** @type {?} */
    LoginComponent.prototype.forgotEmailInputName;
    /** @type {?} */
    LoginComponent.prototype.forgotUsernameormailInputName;
    /** @type {?} */
    LoginComponent.prototype.resetFormAction;
    /** @type {?} */
    LoginComponent.prototype.resetIsFormAjax;
    /** @type {?} */
    LoginComponent.prototype.resetPasswordInputName;
    /** @type {?} */
    LoginComponent.prototype.resetPasswordConfirmInputName;
    /** @type {?} */
    LoginComponent.prototype.loginSubmit;
    /** @type {?} */
    LoginComponent.prototype.forgotSubmit;
    /** @type {?} */
    LoginComponent.prototype.resetSubmit;
    /** @type {?} */
    LoginComponent.prototype._DomSanitizationService;
}
